{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SH3GL2 - Single gene analysis in GP2 data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Description\n",
    "\n",
    "Using \"individual level data\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "### 0. Getting Started\n",
    "\n",
    "- Loading Python libraries\n",
    "- Defining functions\n",
    "- Installing packages\n",
    "\n",
    "### 1. Copy data from workspace to cloud environment\n",
    "\n",
    "### 2. Extract SH3GL2\n",
    "\n",
    "### 3. Annotate SH3GL2 variants\n",
    "\n",
    "### 4. Extract coding/non-syn variants\n",
    "\n",
    "### 5. Calculate frequency in cases versus controls\n",
    "\n",
    "### 6. Calculate frequency (homozygotes) in cases versus controls\n",
    "\n",
    "### 7. Save out results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting Started"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Loading Python libraries"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Defining functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "\n",
    "# Numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery\n",
    "\n",
    "#Import Sys\n",
    "import sys as sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "code_folding": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Billing and Workspace\n",
      "Workspace Name: Endophilin-A\n",
      "Billing Project: terra-18d8e41c\n",
      "Workspace Bucket, where you can upload and download data: gs://fc-f7a400c1-827e-48f8-b7b6-90c488a000a4\n",
      "\n",
      "GP2 v5.0\n",
      "Path to GP2 v5.0 Clinical Data: gs://gp2tier2/release5_11052023/clinical_data\n",
      "Path to GP2 v5.0 Raw Genotype Data: gs://gp2tier2/release5_11052023/raw_genotypes\n",
      "Path to GP2 v5.0 Imputed Genotype Data: gs://gp2tier2/release5_11052023/imputed_genotypes\n",
      "AMP-PD v3.0\n",
      "Path to AMP-PD v3.0 Clinical Data: gs://amp-pd-data/releases/2022_v3release_1115/clinical\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "\n",
    "## GP2 v5.0\n",
    "## Explicitly define release v5.0 path \n",
    "GP2_RELEASE_PATH = 'gs://gp2tier2/release5_11052023'\n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v5.0')\n",
    "print(f'Path to GP2 v5.0 Clinical Data: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v5.0 Raw Genotype Data: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v5.0 Imputed Genotype Data: {GP2_IMPUTED_GENO_PATH}')\n",
    "\n",
    "\n",
    "## AMP-PD v3.0\n",
    "## Explicitly define release v3.0 path \n",
    "AMP_RELEASE_PATH = 'gs://amp-pd-data/releases/2022_v3release_1115'\n",
    "AMP_CLINICAL_RELEASE_PATH = f'{AMP_RELEASE_PATH}/clinical'\n",
    "\n",
    "AMP_WGS_RELEASE_PATH = 'gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS'\n",
    "AMP_WGS_RELEASE_PLINK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'plink')\n",
    "AMP_WGS_RELEASE_PLINK_PFILES = os.path.join(AMP_WGS_RELEASE_PLINK_PATH, 'pfiles')\n",
    "\n",
    "print('AMP-PD v3.0')\n",
    "print(f'Path to AMP-PD v3.0 Clinical Data: {AMP_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PFILES}')\n",
    "print('')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install Plink 1.9 and Plink 2.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n",
      "Plink2 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi\n",
    "\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/alpha3/plink2_linux_avx2_20220603.zip\n",
    "unzip -o plink2_linux_avx2_20220603.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Remote restrictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 1.9 \n",
    "chmod u+x /home/jupyter/tools/plink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 2.0\n",
    "chmod u+x /home/jupyter/tools/plink2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install ANNOVAR: Download sources of annotation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGene.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneMrna.fa.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneVersion.txt.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotate_variation.pl\n",
      "coding_change.pl\n",
      "convert2annovar.pl\n",
      "example\n",
      "humandb\n",
      "retrieve_seq_from_fasta.pl\n",
      "table_annovar.pl\n",
      "variants_reduction.pl\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/annovar/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Copy data from GP2 bucket to workspace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Making a working directory\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: mkdir -p /home/jupyter/SH3GL2_GP2/\n"
     ]
    }
   ],
   "source": [
    "# Make a directory\n",
    "print(\"Making a working directory\")\n",
    "WORK_DIR = f'/home/jupyter/SH3GL2_GP2/'\n",
    "shell_do(f'mkdir -p {WORK_DIR}') # f' means f-string - contains expressions to execute the code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-18d8e41c ls gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr10_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr10_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr10_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr10_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr10_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr10_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr10_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr10_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr11_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr11_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr11_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr11_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr11_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr11_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr11_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr11_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr12_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr12_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr12_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr12_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr12_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr12_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr12_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr12_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr13_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr13_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr13_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr13_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr13_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr13_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr13_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr13_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr14_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr14_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr14_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr14_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr14_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr14_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr14_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr14_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr15_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr15_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr15_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr15_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr15_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr15_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr15_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr15_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr16_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr16_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr16_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr16_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr16_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr16_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr16_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr16_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr17_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr17_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr17_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr17_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr17_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr17_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr17_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr17_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr18_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr18_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr18_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr18_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr18_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr18_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr18_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr18_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr19_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr19_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr19_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr19_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr19_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr19_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr19_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr19_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr1_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr1_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr1_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr1_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr1_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr1_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr1_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr1_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr20_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr20_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr20_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr20_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr20_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr20_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr20_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr20_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr21_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr21_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr21_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr21_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr21_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr21_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr21_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr21_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr22_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr22_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr22_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr22_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr22_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr22_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr22_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr22_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr2_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr2_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr2_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr2_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr2_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr2_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr2_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr2_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr3_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr3_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr3_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr3_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr3_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr3_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr3_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr3_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr4_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr4_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr4_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr4_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr4_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr4_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr4_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr4_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr5_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr5_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr5_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr5_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr5_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr5_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr5_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr5_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr6_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr6_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr6_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr6_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr6_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr6_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr6_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr6_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr7_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr7_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr7_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr7_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr7_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr7_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr7_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr7_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr8_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr8_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr8_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr8_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr8_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr8_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr8_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr8_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr9_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr9_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr9_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr9_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr9_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr9_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr9_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr9_AAC_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chrX_AAC_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chrX_AAC_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chrX_AAC_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chrX_AAC_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chrX_AAC_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chrX_AAC_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chrX_AAC_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chrX_AAC_release5_no_mac_filter.pvar\n"
     ]
    }
   ],
   "source": [
    "# Check directory where GP2 data is\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_IMPUTED_GENO_PATH}/AAC/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-18d8e41c -m cp -r gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr9_AAC_release5_no_mac_filter* /home/jupyter/SH3GL2_GP2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr9_AAC_release5_no_mac_filter.pgen...\n",
      "Copying gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr9_AAC_release5_no_mac_filter.log...\n",
      "Copying gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr9_AAC_release5_no_mac_filter.psam...\n",
      "Copying gs://gp2tier2/release5_11052023/imputed_genotypes/AAC/chr9_AAC_release5_no_mac_filter.pvar...\n",
      "Download already complete for /home/jupyter/SH3GL2_GP2/chr9_AAC_release5_no_mac_filter.pgen component 0, skipping download but will run integrity checks.\n",
      "Resuming download for /home/jupyter/SH3GL2_GP2/chr9_AAC_release5_no_mac_filter.pgen component 2\n",
      "Download already complete for /home/jupyter/SH3GL2_GP2/chr9_AAC_release5_no_mac_filter.pgen component 3, skipping download but will run integrity checks.\n",
      "Download already complete for /home/jupyter/SH3GL2_GP2/chr9_AAC_release5_no_mac_filter.pgen component 1, skipping download but will run integrity checks.\n",
      "| [4/4 files][  1.2 GiB/  1.2 GiB] 100% Done                                    \n",
      "Operation completed over 4 objects/1.2 GiB.                                      \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {GP2_IMPUTED_GENO_PATH}/AAC/chr9_AAC_release5_no_mac_filter* {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create a covariate file with GP2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-18d8e41c ls gs://gp2tier2/release5_11052023/clinical_data\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://gp2tier2/release5_11052023/clinical_data/master_key_release5_final.csv\r\n",
      "gs://gp2tier2/release5_11052023/clinical_data/release5_data_dictionary.csv\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-18d8e41c -m cp -r gs://gp2tier2/release5_11052023/clinical_data/master_key_release5_final.csv /home/jupyter/SH3GL2_GP2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://gp2tier2/release5_11052023/clinical_data/master_key_release5_final.csv...\n",
      "/ [1/1 files][  2.5 MiB/  2.5 MiB] 100% Done                                    \n",
      "Operation completed over 1 objects/2.5 MiB.                                      \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_CLINICAL_RELEASE_PATH}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {GP2_CLINICAL_RELEASE_PATH}/master_key_release5_final.csv {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['GP2ID', 'GP2sampleID', 'manifest_id', 'phenotype', 'pheno_for_qc',\n",
       "       'other_pheno', 'sex_for_qc', 'age', 'age_of_onset', 'age_at_diagnosis',\n",
       "       'age_at_death', 'race_for_qc', 'family_history', 'region_for_qc',\n",
       "       'study', 'pruned', 'pruned_reason', 'label', 'related'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov = pd.read_csv(f'{WORK_DIR}/master_key_release5_final.csv')\n",
    "cov.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "cov_reduced = cov[['GP2sampleID','GP2sampleID','sex_for_qc', 'age', 'phenotype']]\n",
    "cov_reduced.columns = ['FID','IID', 'SEX','AGE','PHENO']\n",
    "#cov_reduced"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "conditions = [\n",
    "    (cov_reduced['PHENO'] == \"PD\"),\n",
    "    (cov_reduced['PHENO'] == \"Control\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "choices = [2,1]\n",
    "cov_reduced['PHENOTYPE'] = np.select(conditions, choices, default=-9).astype(np.int64)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.7/site-packages/pandas/core/frame.py:4913: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n"
     ]
    }
   ],
   "source": [
    "cov_reduced.reset_index(inplace=True)\n",
    "cov_reduced.drop(columns=[\"index\"], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "cov_reduced.drop(columns=['PHENO'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "sex = cov_reduced[['FID','IID','SEX']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "sex.to_csv(f'{WORK_DIR}/SEX.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno = cov_reduced[['FID','IID','PHENOTYPE']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno.to_csv(f'{WORK_DIR}/PHENO.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract SH3GL2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a3.3LM AVX2 Intel (3 Jun 2022)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_SH3GL2_AAC.log.\n",
      "Options in effect:\n",
      "  --chr 9\n",
      "  --from-bp 17579066\n",
      "  --make-bed\n",
      "  --out pheno_SH3GL2_AAC\n",
      "  --pfile chr9_AAC_release5_no_mac_filter\n",
      "  --to-bp 17797124\n",
      "\n",
      "Start time: Thu Nov 23 10:53:29 2023\n",
      "7450 MiB RAM detected; reserving 3725 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "1331 samples (782 females, 549 males; 1331 founders) loaded from\n",
      "chr9_AAC_release5_no_mac_filter.psam.\n",
      "7075019 variants loaded from chr9_AAC_release5_no_mac_filter.pvar.\n",
      "1 binary phenotype loaded (289 cases, 1031 controls).\n",
      "18694 variants remaining after main filters.\n",
      "Writing pheno_SH3GL2_AAC.fam ... done.\n",
      "Writing pheno_SH3GL2_AAC.bim ... done.\n",
      "Writing pheno_SH3GL2_AAC.bed ... 0%\b\b60%\b\b\bdone.\n",
      "End time: Thu Nov 23 10:53:30 2023\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# SH3GL2: gene positions on hg38 (from https://useast.ensembl.org/index.html)\n",
    "/home/jupyter/tools/plink2 \\\n",
    "--pfile chr9_AAC_release5_no_mac_filter \\\n",
    "--chr 9 \\\n",
    "--from-bp 17579066  \\\n",
    "--to-bp 17797124 \\\n",
    "--make-bed \\\n",
    "--out pheno_SH3GL2_AAC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualize plink files bim, fam and bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\tchr9:17579083:C:T\t0\t17579083\tT\tC\n",
      "9\tchr9:17579091:C:A\t0\t17579091\tA\tC\n",
      "9\tchr9:17579091:C:G\t0\t17579091\tG\tC\n",
      "9\tchr9:17579092:G:A\t0\t17579092\tA\tG\n",
      "9\tchr9:17579096:C:T\t0\t17579096\tT\tC\n",
      "9\tchr9:17579104:G:C\t0\t17579104\tC\tG\n",
      "9\tchr9:17579105:C:G\t0\t17579105\tG\tC\n",
      "9\tchr9:17579105:C:T\t0\t17579105\tT\tC\n",
      "9\tchr9:17579108:C:G\t0\t17579108\tG\tC\n",
      "9\tchr9:17579113:C:A\t0\t17579113\tA\tC\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize bim file\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_SH3GL2_AAC.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\tBCM_000096_s1\t0\t0\t1\t2\n",
      "0\tBCM_000142_s1\t0\t0\t1\t2\n",
      "0\tBCM_000150_s1\t0\t0\t2\t2\n",
      "0\tBCM_000204_s1\t0\t0\t2\t2\n",
      "0\tBCM_000215_s1\t0\t0\t1\t2\n",
      "0\tBCM_000230_s1\t0\t0\t1\t2\n",
      "0\tBCM_000270_s1\t0\t0\t2\t1\n",
      "0\tBCM_000277_s1\t0\t0\t2\t1\n",
      "0\tBCM_000375_s1\t0\t0\t2\t1\n",
      "0\tBCM_000418_s1\t0\t0\t1\t1\n",
      "1331 pheno_SH3GL2_AAC.fam\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize fam file\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_SH3GL2_AAC.fam\n",
    "\n",
    "#Store second column of the temp.fam file in a new IDs.txt file\n",
    "#awk '{print $2}' pheno_SH3GL2_AAC.fam > IDs.txt\n",
    "\n",
    "#Count lines temp.fam file\n",
    "wc -l pheno_SH3GL2_AAC.fam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_SH3GL2_AAC_nomacfilter.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_SH3GL2_AAC\n",
      "  --out pheno_SH3GL2_AAC_nomacfilter\n",
      "  --recode vcf-fid\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "18694 variants loaded from .bim file.\n",
      "1331 people (549 males, 782 females) loaded from .fam.\n",
      "1320 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1331 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999202.\n",
      "18694 variants and 1331 people pass filters and QC.\n",
      "Among remaining phenotypes, 289 are cases and 1031 are controls.  (11\n",
      "phenotypes are missing.)\n",
      "--recode vcf-fid to pheno_SH3GL2_AAC_nomacfilter.vcf ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# Turn binary files into VCF\n",
    "/home/jupyter/tools/plink \\\n",
    "--bfile pheno_SH3GL2_AAC \\\n",
    "--recode vcf-fid \\\n",
    "--out pheno_SH3GL2_AAC_nomacfilter"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate SH3GL2 variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "export PATH=$PATH:/home/jupyter/tools/rvtests/third/tabix-0.2.6/\n",
    "\n",
    "### Bgzip and Tabix\n",
    "bgzip pheno_SH3GL2_AAC_nomacfilter.vcf\n",
    "\n",
    "tabix -f -p vcf pheno_SH3GL2_AAC_nomacfilter.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 pheno_SH3GL2_AAC_nomacfilter.vcf.gz > pheno_SH3GL2_AAC.annovar.avinput>\n",
      "NOTICE: Finished reading 18701 lines from VCF file\n",
      "NOTICE: A total of 18694 locus in VCF file passed QC threshold, representing 17626 SNPs (10548 transitions and 7078 transversions) and 1068 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 23460206 SNP genotypes (14039388 transitions and 9420818 transversions) and 1421508 indels/substitutions for 1331 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl pheno_SH3GL2_AAC.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile pheno_SH3GL2_AAC.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile pheno_SH3GL2_AAC.annovar.refGene -exonsort -nofirstcodondel pheno_SH3GL2_AAC.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to pheno_SH3GL2_AAC.annovar.refGene.variant_function, pheno_SH3GL2_AAC.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 18694 unique variants in 18694 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile pheno_SH3GL2_AAC.annovar pheno_SH3GL2_AAC.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to pheno_SH3GL2_AAC.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to pheno_SH3GL2_AAC.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 18694 unique variants in 18694 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 1\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to pheno_SH3GL2_AAC.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from pheno_SH3GL2_AAC.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to pheno_SH3GL2_AAC.annovar.hg38_multianno.vcf\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "### Annotate variants using ANNOVAR: https://annovar.openbioinformatics.org/en/latest/ \n",
    "perl /home/jupyter/tools/annovar/table_annovar.pl pheno_SH3GL2_AAC_nomacfilter.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out pheno_SH3GL2_AAC.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract coding and non-syn variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Visualize multianno file\n",
    "SH3GL2_AAC = pd.read_csv(f'{WORK_DIR}/pheno_SH3GL2_AAC.annovar.hg38_multianno.txt', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "SH3GL2_AAC['Chr_bp'] = SH3GL2_AAC['Chr'].astype(str) + ':' + SH3GL2_AAC['Start'].astype(str) + ':' + SH3GL2_AAC['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "SH3GL2_AAC = SH3GL2_AAC.drop_duplicates(subset=['Chr_bp'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "SH3GL2_AAC = SH3GL2_AAC[SH3GL2_AAC['Gene.refGene']=='SH3GL2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr              18193\n",
       "Start            18193\n",
       "End              18193\n",
       "Ref              18193\n",
       "Alt              18193\n",
       "                 ...  \n",
       "Otherinfo1340    18193\n",
       "Otherinfo1341    18193\n",
       "Otherinfo1342    18193\n",
       "Otherinfo1343    18193\n",
       "Chr_bp           18193\n",
       "Length: 1355, dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_AAC.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr_bp</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9:17579083:17579083</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9:17579091:17579091</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9:17579092:17579092</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9:17579096:17579096</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>9:17579104:17579104</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18688</th>\n",
       "      <td>9:17797007:17797007</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18689</th>\n",
       "      <td>9:17797040:17797040</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18690</th>\n",
       "      <td>9:17797052:17797052</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18691</th>\n",
       "      <td>9:17797079:17797079</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18692</th>\n",
       "      <td>9:17797083:17797083</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18193 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Chr_bp Gene.refGene Func.refGene ExonicFunc.refGene\n",
       "0      9:17579083:17579083       SH3GL2         UTR5                  .\n",
       "1      9:17579091:17579091       SH3GL2         UTR5                  .\n",
       "3      9:17579092:17579092       SH3GL2         UTR5                  .\n",
       "4      9:17579096:17579096       SH3GL2         UTR5                  .\n",
       "5      9:17579104:17579104       SH3GL2         UTR5                  .\n",
       "...                    ...          ...          ...                ...\n",
       "18688  9:17797007:17797007       SH3GL2         UTR3                  .\n",
       "18689  9:17797040:17797040       SH3GL2         UTR3                  .\n",
       "18690  9:17797052:17797052       SH3GL2         UTR3                  .\n",
       "18691  9:17797079:17797079       SH3GL2         UTR3                  .\n",
       "18692  9:17797083:17797083       SH3GL2         UTR3                  .\n",
       "\n",
       "[18193 rows x 4 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reduced_SH3GL2_AAC = SH3GL2_AAC[[\"Chr_bp\", \"Gene.refGene\", 'Func.refGene','ExonicFunc.refGene']]\n",
    "reduced_SH3GL2_AAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "reduced_SH3GL2_AAC.to_csv(f'{WORKSPACE_BUCKET}/reduced_SH3GL2_AAC.txt', sep = '\\t', index = False, header= True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "intronic    18004\n",
       "UTR3          107\n",
       "exonic         46\n",
       "UTR5           36\n",
       "Name: Func.refGene, dtype: int64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_AAC['Func.refGene'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       ".                    18147\n",
       "nonsynonymous SNV       26\n",
       "synonymous SNV          20\n",
       "Name: ExonicFunc.refGene, dtype: int64"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_AAC['ExonicFunc.refGene'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr              46\n",
       "Start            46\n",
       "End              46\n",
       "Ref              46\n",
       "Alt              46\n",
       "                 ..\n",
       "Otherinfo1340    46\n",
       "Otherinfo1341    46\n",
       "Otherinfo1342    46\n",
       "Otherinfo1343    46\n",
       "Chr_bp           46\n",
       "Length: 1355, dtype: int64"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic variants\n",
    "coding_AAC = SH3GL2_AAC[SH3GL2_AAC['Func.refGene'] == 'exonic']\n",
    "coding_AAC.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr              20\n",
       "Start            20\n",
       "End              20\n",
       "Ref              20\n",
       "Alt              20\n",
       "                 ..\n",
       "Otherinfo1340    20\n",
       "Otherinfo1341    20\n",
       "Otherinfo1342    20\n",
       "Otherinfo1343    20\n",
       "Chr_bp           20\n",
       "Length: 1355, dtype: int64"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic and syn vars\n",
    "coding_synonymous_AAC = SH3GL2_AAC[(SH3GL2_AAC['Func.refGene'] == 'exonic') & (SH3GL2_AAC['ExonicFunc.refGene'] == 'synonymous SNV')]\n",
    "coding_synonymous_AAC.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo1335</th>\n",
       "      <th>Otherinfo1336</th>\n",
       "      <th>Otherinfo1337</th>\n",
       "      <th>Otherinfo1338</th>\n",
       "      <th>Otherinfo1339</th>\n",
       "      <th>Otherinfo1340</th>\n",
       "      <th>Otherinfo1341</th>\n",
       "      <th>Otherinfo1342</th>\n",
       "      <th>Otherinfo1343</th>\n",
       "      <th>Chr_bp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>9</td>\n",
       "      <td>17579258</td>\n",
       "      <td>17579258</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon1:c.C16T:p.L6F</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17579258:17579258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14606</th>\n",
       "      <td>9</td>\n",
       "      <td>17747115</td>\n",
       "      <td>17747115</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon2:c.A95C:p.D32A</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17747115:17747115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15842</th>\n",
       "      <td>9</td>\n",
       "      <td>17761500</td>\n",
       "      <td>17761500</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon3:c.C178G:p.P60A</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17761500:17761500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17864</th>\n",
       "      <td>9</td>\n",
       "      <td>17786465</td>\n",
       "      <td>17786465</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon4:c.C272T:p.A91V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17786465:17786465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17865</th>\n",
       "      <td>9</td>\n",
       "      <td>17786519</td>\n",
       "      <td>17786519</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon4:c.A326G:p.N109S</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17786519:17786519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17944</th>\n",
       "      <td>9</td>\n",
       "      <td>17787509</td>\n",
       "      <td>17787509</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon5:c.T461C:p.I154T</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17787509:17787509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18092</th>\n",
       "      <td>9</td>\n",
       "      <td>17789473</td>\n",
       "      <td>17789473</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.C547G:p.L183V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789473:17789473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18093</th>\n",
       "      <td>9</td>\n",
       "      <td>17789476</td>\n",
       "      <td>17789476</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.C550T:p.R184C</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789476:17789476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18094</th>\n",
       "      <td>9</td>\n",
       "      <td>17789477</td>\n",
       "      <td>17789477</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.G551A:p.R184H</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789477:17789477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18095</th>\n",
       "      <td>9</td>\n",
       "      <td>17789499</td>\n",
       "      <td>17789499</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.T573G:p.D191E</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789499:17789499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18096</th>\n",
       "      <td>9</td>\n",
       "      <td>17789545</td>\n",
       "      <td>17789545</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.A619G:p.M207V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789545:17789545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18242</th>\n",
       "      <td>9</td>\n",
       "      <td>17791272</td>\n",
       "      <td>17791272</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.G666C:p.Q222H</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791272:17791272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18243</th>\n",
       "      <td>9</td>\n",
       "      <td>17791276</td>\n",
       "      <td>17791276</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.G670C:p.E224Q</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791276:17791276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18245</th>\n",
       "      <td>9</td>\n",
       "      <td>17791308</td>\n",
       "      <td>17791308</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.G702C:p.Q234H</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791308:17791308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18247</th>\n",
       "      <td>9</td>\n",
       "      <td>17791316</td>\n",
       "      <td>17791316</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.C710T:p.T237M</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791316:17791316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18392</th>\n",
       "      <td>9</td>\n",
       "      <td>17793420</td>\n",
       "      <td>17793420</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.T782C:p.M261T</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793420:17793420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18393</th>\n",
       "      <td>9</td>\n",
       "      <td>17793428</td>\n",
       "      <td>17793428</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.G790C:p.E264Q</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793428:17793428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18394</th>\n",
       "      <td>9</td>\n",
       "      <td>17793437</td>\n",
       "      <td>17793437</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.A799G:p.T267A</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793437:17793437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18395</th>\n",
       "      <td>9</td>\n",
       "      <td>17793450</td>\n",
       "      <td>17793450</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.C812G:p.T271S</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793450:17793450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18396</th>\n",
       "      <td>9</td>\n",
       "      <td>17793465</td>\n",
       "      <td>17793465</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.G827T:p.G276V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793465:17793465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18397</th>\n",
       "      <td>9</td>\n",
       "      <td>17793471</td>\n",
       "      <td>17793471</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.C833T:p.S278F</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793471:17793471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18398</th>\n",
       "      <td>9</td>\n",
       "      <td>17793474</td>\n",
       "      <td>17793474</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.A836T:p.H279L</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793474:17793474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18574</th>\n",
       "      <td>9</td>\n",
       "      <td>17795560</td>\n",
       "      <td>17795560</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.G876T:p.Q292H</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795560:17795560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18577</th>\n",
       "      <td>9</td>\n",
       "      <td>17795624</td>\n",
       "      <td>17795624</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.G940A:p.G314S</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795624:17795624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18579</th>\n",
       "      <td>9</td>\n",
       "      <td>17795627</td>\n",
       "      <td>17795627</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.G943A:p.D315N</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795627:17795627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18580</th>\n",
       "      <td>9</td>\n",
       "      <td>17795649</td>\n",
       "      <td>17795649</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.A965G:p.Q322R</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795649:17795649</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26 rows × 1355 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chr     Start       End Ref Alt Func.refGene Gene.refGene  \\\n",
       "39       9  17579258  17579258   C   T       exonic       SH3GL2   \n",
       "14606    9  17747115  17747115   A   C       exonic       SH3GL2   \n",
       "15842    9  17761500  17761500   C   G       exonic       SH3GL2   \n",
       "17864    9  17786465  17786465   C   T       exonic       SH3GL2   \n",
       "17865    9  17786519  17786519   A   G       exonic       SH3GL2   \n",
       "17944    9  17787509  17787509   T   C       exonic       SH3GL2   \n",
       "18092    9  17789473  17789473   C   G       exonic       SH3GL2   \n",
       "18093    9  17789476  17789476   C   T       exonic       SH3GL2   \n",
       "18094    9  17789477  17789477   G   A       exonic       SH3GL2   \n",
       "18095    9  17789499  17789499   T   G       exonic       SH3GL2   \n",
       "18096    9  17789545  17789545   A   G       exonic       SH3GL2   \n",
       "18242    9  17791272  17791272   G   C       exonic       SH3GL2   \n",
       "18243    9  17791276  17791276   G   C       exonic       SH3GL2   \n",
       "18245    9  17791308  17791308   G   C       exonic       SH3GL2   \n",
       "18247    9  17791316  17791316   C   T       exonic       SH3GL2   \n",
       "18392    9  17793420  17793420   T   C       exonic       SH3GL2   \n",
       "18393    9  17793428  17793428   G   C       exonic       SH3GL2   \n",
       "18394    9  17793437  17793437   A   G       exonic       SH3GL2   \n",
       "18395    9  17793450  17793450   C   G       exonic       SH3GL2   \n",
       "18396    9  17793465  17793465   G   T       exonic       SH3GL2   \n",
       "18397    9  17793471  17793471   C   T       exonic       SH3GL2   \n",
       "18398    9  17793474  17793474   A   T       exonic       SH3GL2   \n",
       "18574    9  17795560  17795560   G   T       exonic       SH3GL2   \n",
       "18577    9  17795624  17795624   G   A       exonic       SH3GL2   \n",
       "18579    9  17795627  17795627   G   A       exonic       SH3GL2   \n",
       "18580    9  17795649  17795649   A   G       exonic       SH3GL2   \n",
       "\n",
       "      GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "39                     .  nonsynonymous SNV   \n",
       "14606                  .  nonsynonymous SNV   \n",
       "15842                  .  nonsynonymous SNV   \n",
       "17864                  .  nonsynonymous SNV   \n",
       "17865                  .  nonsynonymous SNV   \n",
       "17944                  .  nonsynonymous SNV   \n",
       "18092                  .  nonsynonymous SNV   \n",
       "18093                  .  nonsynonymous SNV   \n",
       "18094                  .  nonsynonymous SNV   \n",
       "18095                  .  nonsynonymous SNV   \n",
       "18096                  .  nonsynonymous SNV   \n",
       "18242                  .  nonsynonymous SNV   \n",
       "18243                  .  nonsynonymous SNV   \n",
       "18245                  .  nonsynonymous SNV   \n",
       "18247                  .  nonsynonymous SNV   \n",
       "18392                  .  nonsynonymous SNV   \n",
       "18393                  .  nonsynonymous SNV   \n",
       "18394                  .  nonsynonymous SNV   \n",
       "18395                  .  nonsynonymous SNV   \n",
       "18396                  .  nonsynonymous SNV   \n",
       "18397                  .  nonsynonymous SNV   \n",
       "18398                  .  nonsynonymous SNV   \n",
       "18574                  .  nonsynonymous SNV   \n",
       "18577                  .  nonsynonymous SNV   \n",
       "18579                  .  nonsynonymous SNV   \n",
       "18580                  .  nonsynonymous SNV   \n",
       "\n",
       "                             AAChange.refGene  ... Otherinfo1335  \\\n",
       "39        SH3GL2:NM_003026:exon1:c.C16T:p.L6F  ...           0/0   \n",
       "14606    SH3GL2:NM_003026:exon2:c.A95C:p.D32A  ...           0/0   \n",
       "15842   SH3GL2:NM_003026:exon3:c.C178G:p.P60A  ...           0/0   \n",
       "17864   SH3GL2:NM_003026:exon4:c.C272T:p.A91V  ...           0/0   \n",
       "17865  SH3GL2:NM_003026:exon4:c.A326G:p.N109S  ...           0/0   \n",
       "17944  SH3GL2:NM_003026:exon5:c.T461C:p.I154T  ...           0/0   \n",
       "18092  SH3GL2:NM_003026:exon6:c.C547G:p.L183V  ...           0/0   \n",
       "18093  SH3GL2:NM_003026:exon6:c.C550T:p.R184C  ...           0/0   \n",
       "18094  SH3GL2:NM_003026:exon6:c.G551A:p.R184H  ...           0/0   \n",
       "18095  SH3GL2:NM_003026:exon6:c.T573G:p.D191E  ...           0/0   \n",
       "18096  SH3GL2:NM_003026:exon6:c.A619G:p.M207V  ...           0/0   \n",
       "18242  SH3GL2:NM_003026:exon7:c.G666C:p.Q222H  ...           0/0   \n",
       "18243  SH3GL2:NM_003026:exon7:c.G670C:p.E224Q  ...           0/0   \n",
       "18245  SH3GL2:NM_003026:exon7:c.G702C:p.Q234H  ...           0/0   \n",
       "18247  SH3GL2:NM_003026:exon7:c.C710T:p.T237M  ...           0/0   \n",
       "18392  SH3GL2:NM_003026:exon8:c.T782C:p.M261T  ...           0/0   \n",
       "18393  SH3GL2:NM_003026:exon8:c.G790C:p.E264Q  ...           0/0   \n",
       "18394  SH3GL2:NM_003026:exon8:c.A799G:p.T267A  ...           0/0   \n",
       "18395  SH3GL2:NM_003026:exon8:c.C812G:p.T271S  ...           0/0   \n",
       "18396  SH3GL2:NM_003026:exon8:c.G827T:p.G276V  ...           0/0   \n",
       "18397  SH3GL2:NM_003026:exon8:c.C833T:p.S278F  ...           0/0   \n",
       "18398  SH3GL2:NM_003026:exon8:c.A836T:p.H279L  ...           0/0   \n",
       "18574  SH3GL2:NM_003026:exon9:c.G876T:p.Q292H  ...           0/0   \n",
       "18577  SH3GL2:NM_003026:exon9:c.G940A:p.G314S  ...           0/0   \n",
       "18579  SH3GL2:NM_003026:exon9:c.G943A:p.D315N  ...           0/0   \n",
       "18580  SH3GL2:NM_003026:exon9:c.A965G:p.Q322R  ...           0/0   \n",
       "\n",
       "       Otherinfo1336 Otherinfo1337 Otherinfo1338  Otherinfo1339  \\\n",
       "39               0/0           0/0           0/0            0/0   \n",
       "14606            0/0           0/0           0/0            0/0   \n",
       "15842            0/0           0/0           0/0            0/0   \n",
       "17864            0/0           0/0           0/0            0/0   \n",
       "17865            0/0           0/0           0/0            0/0   \n",
       "17944            0/0           0/0           0/0            0/0   \n",
       "18092            0/0           0/0           0/0            0/0   \n",
       "18093            0/0           0/0           0/0            0/0   \n",
       "18094            0/0           0/0           0/0            0/0   \n",
       "18095            0/0           0/0           0/0            0/0   \n",
       "18096            0/0           0/0           0/0            0/0   \n",
       "18242            0/0           0/0           0/0            0/0   \n",
       "18243            0/0           0/0           0/0            0/0   \n",
       "18245            0/0           0/0           0/0            0/0   \n",
       "18247            0/0           0/0           0/0            0/0   \n",
       "18392            0/0           0/0           0/0            0/0   \n",
       "18393            0/0           0/0           0/0            0/0   \n",
       "18394            0/0           0/0           0/0            0/0   \n",
       "18395            0/0           0/0           0/0            0/0   \n",
       "18396            0/0           0/0           0/0            0/0   \n",
       "18397            0/0           0/0           0/0            0/0   \n",
       "18398            0/0           0/0           0/0            0/0   \n",
       "18574            0/0           0/0           0/0            0/0   \n",
       "18577            0/0           0/0           0/0            0/0   \n",
       "18579            0/0           0/0           0/0            0/0   \n",
       "18580            0/0           0/0           0/0            0/0   \n",
       "\n",
       "       Otherinfo1340 Otherinfo1341 Otherinfo1342 Otherinfo1343  \\\n",
       "39               0/0           0/0           0/0           0/0   \n",
       "14606            0/0           0/0           0/0           0/0   \n",
       "15842            0/0           0/0           0/0           0/0   \n",
       "17864            0/0           0/0           0/0           0/0   \n",
       "17865            0/0           0/0           0/0           0/0   \n",
       "17944            0/0           0/0           0/0           0/0   \n",
       "18092            0/0           0/0           0/0           0/0   \n",
       "18093            0/0           0/0           0/0           0/0   \n",
       "18094            0/0           0/0           0/0           0/0   \n",
       "18095            0/0           0/0           0/0           0/0   \n",
       "18096            0/0           0/0           0/0           0/0   \n",
       "18242            0/0           0/0           0/0           0/0   \n",
       "18243            0/0           0/0           0/0           0/0   \n",
       "18245            0/0           0/0           0/0           0/0   \n",
       "18247            0/0           0/0           0/0           0/0   \n",
       "18392            0/0           0/0           0/0           0/0   \n",
       "18393            0/0           0/0           0/0           0/0   \n",
       "18394            0/0           0/0           0/0           0/0   \n",
       "18395            0/0           0/0           0/0           0/0   \n",
       "18396            0/0           0/0           0/0           0/0   \n",
       "18397            0/0           0/0           0/0           0/0   \n",
       "18398            0/0           0/0           0/0           0/0   \n",
       "18574            0/0           0/0           0/0           0/0   \n",
       "18577            0/0           0/0           0/0           0/0   \n",
       "18579            0/0           0/0           0/0           0/0   \n",
       "18580            0/0           0/0           0/0           0/0   \n",
       "\n",
       "                    Chr_bp  \n",
       "39     9:17579258:17579258  \n",
       "14606  9:17747115:17747115  \n",
       "15842  9:17761500:17761500  \n",
       "17864  9:17786465:17786465  \n",
       "17865  9:17786519:17786519  \n",
       "17944  9:17787509:17787509  \n",
       "18092  9:17789473:17789473  \n",
       "18093  9:17789476:17789476  \n",
       "18094  9:17789477:17789477  \n",
       "18095  9:17789499:17789499  \n",
       "18096  9:17789545:17789545  \n",
       "18242  9:17791272:17791272  \n",
       "18243  9:17791276:17791276  \n",
       "18245  9:17791308:17791308  \n",
       "18247  9:17791316:17791316  \n",
       "18392  9:17793420:17793420  \n",
       "18393  9:17793428:17793428  \n",
       "18394  9:17793437:17793437  \n",
       "18395  9:17793450:17793450  \n",
       "18396  9:17793465:17793465  \n",
       "18397  9:17793471:17793471  \n",
       "18398  9:17793474:17793474  \n",
       "18574  9:17795560:17795560  \n",
       "18577  9:17795624:17795624  \n",
       "18579  9:17795627:17795627  \n",
       "18580  9:17795649:17795649  \n",
       "\n",
       "[26 rows x 1355 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic and non-syn vars\n",
    "coding_nonsynonymous_AAC = SH3GL2_AAC[(SH3GL2_AAC['Func.refGene'] == 'exonic') & (SH3GL2_AAC['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_nonsynonymous_AAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr              18004\n",
       "Start            18004\n",
       "End              18004\n",
       "Ref              18004\n",
       "Alt              18004\n",
       "                 ...  \n",
       "Otherinfo1340    18004\n",
       "Otherinfo1341    18004\n",
       "Otherinfo1342    18004\n",
       "Otherinfo1343    18004\n",
       "Chr_bp           18004\n",
       "Length: 1355, dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter intronic variants\n",
    "intronic_AAC = SH3GL2_AAC[SH3GL2_AAC['Func.refGene'] == 'intronic']\n",
    "intronic_AAC.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr              107\n",
       "Start            107\n",
       "End              107\n",
       "Ref              107\n",
       "Alt              107\n",
       "                ... \n",
       "Otherinfo1340    107\n",
       "Otherinfo1341    107\n",
       "Otherinfo1342    107\n",
       "Otherinfo1343    107\n",
       "Chr_bp           107\n",
       "Length: 1355, dtype: int64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter UTR3 variants\n",
    "UTR3_AAC= SH3GL2_AAC[SH3GL2_AAC['Func.refGene'] == ('UTR3')]\n",
    "UTR3_AAC.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr              36\n",
       "Start            36\n",
       "End              36\n",
       "Ref              36\n",
       "Alt              36\n",
       "                 ..\n",
       "Otherinfo1340    36\n",
       "Otherinfo1341    36\n",
       "Otherinfo1342    36\n",
       "Otherinfo1343    36\n",
       "Chr_bp           36\n",
       "Length: 1355, dtype: int64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter UTR5 variants\n",
    "UTR5_AAC = SH3GL2_AAC[SH3GL2_AAC['Func.refGene'] == ('UTR5')]\n",
    "UTR5_AAC.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "reduced_coding_nonsynonymous_AAC = coding_nonsynonymous_AAC[[\"Chr\", \"Start\", \"End\", \"Gene.refGene\"]]\n",
    "reduced_coding_nonsynonymous_AAC.to_csv(f'{WORK_DIR}/reduced_coding_nonsynonymous_AAC.txt', sep = '\\t', index = False, header= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\t17579258\t17579258\tSH3GL2\n",
      "9\t17747115\t17747115\tSH3GL2\n",
      "9\t17761500\t17761500\tSH3GL2\n",
      "9\t17786465\t17786465\tSH3GL2\n",
      "9\t17786519\t17786519\tSH3GL2\n",
      "9\t17787509\t17787509\tSH3GL2\n",
      "9\t17789473\t17789473\tSH3GL2\n",
      "9\t17789476\t17789476\tSH3GL2\n",
      "9\t17789477\t17789477\tSH3GL2\n",
      "9\t17789499\t17789499\tSH3GL2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head reduced_coding_nonsynonymous_AAC.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_SH3GL2_AAC.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile pheno_SH3GL2_AAC\n",
      "  --ci 0.95\n",
      "  --extract range reduced_coding_nonsynonymous_AAC.txt\n",
      "  --out coding_nonsynonymous_pheno_SH3GL2_AAC\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "18694 variants loaded from .bim file.\n",
      "1331 people (549 males, 782 females) loaded from .fam.\n",
      "1320 phenotype values loaded from .fam.\n",
      "--extract range: 18668 variants excluded.\n",
      "--extract range: 26 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1331 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999942.\n",
      "26 variants and 1331 people pass filters and QC.\n",
      "Among remaining phenotypes, 289 are cases and 1031 are controls.  (11\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to coding_nonsynonymous_pheno_SH3GL2_AAC.assoc ...\n",
      "0%\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_SH3GL2_AAC --extract range reduced_coding_nonsynonymous_AAC.txt --assoc --ci 0.95 --out coding_nonsynonymous_pheno_SH3GL2_AAC --allow-no-sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17579258:C:T</td>\n",
       "      <td>17579258</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17747115:A:C</td>\n",
       "      <td>17747115</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17761500:C:G</td>\n",
       "      <td>17761500</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17786465:C:T</td>\n",
       "      <td>17786465</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17786519:A:G</td>\n",
       "      <td>17786519</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17787509:T:C</td>\n",
       "      <td>17787509</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789473:C:G</td>\n",
       "      <td>17789473</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789476:C:T</td>\n",
       "      <td>17789476</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789477:G:A</td>\n",
       "      <td>17789477</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789499:T:G</td>\n",
       "      <td>17789499</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000485</td>\n",
       "      <td>T</td>\n",
       "      <td>0.28040</td>\n",
       "      <td>0.5964</td>\n",
       "      <td>0.000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789545:A:G</td>\n",
       "      <td>17789545</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17791272:G:C</td>\n",
       "      <td>17791272</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000485</td>\n",
       "      <td>G</td>\n",
       "      <td>0.28040</td>\n",
       "      <td>0.5964</td>\n",
       "      <td>0.000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17791276:G:C</td>\n",
       "      <td>17791276</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17791308:G:C</td>\n",
       "      <td>17791308</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000485</td>\n",
       "      <td>G</td>\n",
       "      <td>0.28040</td>\n",
       "      <td>0.5964</td>\n",
       "      <td>0.000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17791316:C:T</td>\n",
       "      <td>17791316</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793420:T:C</td>\n",
       "      <td>17793420</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793428:G:C</td>\n",
       "      <td>17793428</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793437:A:G</td>\n",
       "      <td>17793437</td>\n",
       "      <td>G</td>\n",
       "      <td>0.001742</td>\n",
       "      <td>0.002425</td>\n",
       "      <td>A</td>\n",
       "      <td>0.09214</td>\n",
       "      <td>0.7615</td>\n",
       "      <td>0.718</td>\n",
       "      <td>1.096</td>\n",
       "      <td>0.08371</td>\n",
       "      <td>6.158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793450:C:G</td>\n",
       "      <td>17793450</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793465:G:T</td>\n",
       "      <td>17793465</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001455</td>\n",
       "      <td>G</td>\n",
       "      <td>0.84190</td>\n",
       "      <td>0.3589</td>\n",
       "      <td>0.000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793471:C:T</td>\n",
       "      <td>17793471</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000485</td>\n",
       "      <td>C</td>\n",
       "      <td>0.28040</td>\n",
       "      <td>0.5964</td>\n",
       "      <td>0.000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793474:A:T</td>\n",
       "      <td>17793474</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000485</td>\n",
       "      <td>A</td>\n",
       "      <td>0.28040</td>\n",
       "      <td>0.5964</td>\n",
       "      <td>0.000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795560:G:T</td>\n",
       "      <td>17795560</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795624:G:A</td>\n",
       "      <td>17795624</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795627:G:A</td>\n",
       "      <td>17795627</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795649:A:G</td>\n",
       "      <td>17795649</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHR                SNP        BP A1       F_A       F_U A2    CHISQ  \\\n",
       "0     9  chr9:17579258:C:T  17579258  T  0.000000  0.000000  C      NaN   \n",
       "1     9  chr9:17747115:A:C  17747115  C  0.000000  0.000000  A      NaN   \n",
       "2     9  chr9:17761500:C:G  17761500  G  0.000000  0.000000  C      NaN   \n",
       "3     9  chr9:17786465:C:T  17786465  T  0.000000  0.000000  C      NaN   \n",
       "4     9  chr9:17786519:A:G  17786519  G  0.000000  0.000000  A      NaN   \n",
       "5     9  chr9:17787509:T:C  17787509  C  0.000000  0.000000  T      NaN   \n",
       "6     9  chr9:17789473:C:G  17789473  G  0.000000  0.000000  C      NaN   \n",
       "7     9  chr9:17789476:C:T  17789476  T  0.000000  0.000000  C      NaN   \n",
       "8     9  chr9:17789477:G:A  17789477  A  0.000000  0.000000  G      NaN   \n",
       "9     9  chr9:17789499:T:G  17789499  G  0.000000  0.000485  T  0.28040   \n",
       "10    9  chr9:17789545:A:G  17789545  G  0.000000  0.000000  A      NaN   \n",
       "11    9  chr9:17791272:G:C  17791272  C  0.000000  0.000485  G  0.28040   \n",
       "12    9  chr9:17791276:G:C  17791276  C  0.000000  0.000000  G      NaN   \n",
       "13    9  chr9:17791308:G:C  17791308  C  0.000000  0.000485  G  0.28040   \n",
       "14    9  chr9:17791316:C:T  17791316  T  0.000000  0.000000  C      NaN   \n",
       "15    9  chr9:17793420:T:C  17793420  C  0.000000  0.000000  T      NaN   \n",
       "16    9  chr9:17793428:G:C  17793428  C  0.000000  0.000000  G      NaN   \n",
       "17    9  chr9:17793437:A:G  17793437  G  0.001742  0.002425  A  0.09214   \n",
       "18    9  chr9:17793450:C:G  17793450  G  0.000000  0.000000  C      NaN   \n",
       "19    9  chr9:17793465:G:T  17793465  T  0.000000  0.001455  G  0.84190   \n",
       "20    9  chr9:17793471:C:T  17793471  T  0.000000  0.000485  C  0.28040   \n",
       "21    9  chr9:17793474:A:T  17793474  T  0.000000  0.000485  A  0.28040   \n",
       "22    9  chr9:17795560:G:T  17795560  T  0.000000  0.000000  G      NaN   \n",
       "23    9  chr9:17795624:G:A  17795624  A  0.000000  0.000000  G      NaN   \n",
       "24    9  chr9:17795627:G:A  17795627  A  0.000000  0.000000  G      NaN   \n",
       "25    9  chr9:17795649:A:G  17795649  G  0.000000  0.000000  A      NaN   \n",
       "\n",
       "         P     OR     SE      L95    U95  \n",
       "0      NaN    NaN    NaN      NaN    NaN  \n",
       "1      NaN    NaN    NaN      NaN    NaN  \n",
       "2      NaN    NaN    NaN      NaN    NaN  \n",
       "3      NaN    NaN    NaN      NaN    NaN  \n",
       "4      NaN    NaN    NaN      NaN    NaN  \n",
       "5      NaN    NaN    NaN      NaN    NaN  \n",
       "6      NaN    NaN    NaN      NaN    NaN  \n",
       "7      NaN    NaN    NaN      NaN    NaN  \n",
       "8      NaN    NaN    NaN      NaN    NaN  \n",
       "9   0.5964  0.000    inf  0.00000    NaN  \n",
       "10     NaN    NaN    NaN      NaN    NaN  \n",
       "11  0.5964  0.000    inf  0.00000    NaN  \n",
       "12     NaN    NaN    NaN      NaN    NaN  \n",
       "13  0.5964  0.000    inf  0.00000    NaN  \n",
       "14     NaN    NaN    NaN      NaN    NaN  \n",
       "15     NaN    NaN    NaN      NaN    NaN  \n",
       "16     NaN    NaN    NaN      NaN    NaN  \n",
       "17  0.7615  0.718  1.096  0.08371  6.158  \n",
       "18     NaN    NaN    NaN      NaN    NaN  \n",
       "19  0.3589  0.000    inf  0.00000    NaN  \n",
       "20  0.5964  0.000    inf  0.00000    NaN  \n",
       "21  0.5964  0.000    inf  0.00000    NaN  \n",
       "22     NaN    NaN    NaN      NaN    NaN  \n",
       "23     NaN    NaN    NaN      NaN    NaN  \n",
       "24     NaN    NaN    NaN      NaN    NaN  \n",
       "25     NaN    NaN    NaN      NaN    NaN  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_AAC_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_SH3GL2_AAC.assoc', delim_whitespace=True)\n",
    "SH3GL2_AAC_freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_SH3GL2_AAC.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_SH3GL2_AAC\n",
      "  --extract range reduced_coding_nonsynonymous_AAC.txt\n",
      "  --out coding_nonsynonymous_pheno_SH3GL2_AAC\n",
      "  --recode A\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "18694 variants loaded from .bim file.\n",
      "1331 people (549 males, 782 females) loaded from .fam.\n",
      "1320 phenotype values loaded from .fam.\n",
      "--extract range: 18668 variants excluded.\n",
      "--extract range: 26 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1331 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999942.\n",
      "26 variants and 1331 people pass filters and QC.\n",
      "Among remaining phenotypes, 289 are cases and 1031 are controls.  (11\n",
      "phenotypes are missing.)\n",
      "--recode A to coding_nonsynonymous_pheno_SH3GL2_AAC.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_SH3GL2_AAC --extract range reduced_coding_nonsynonymous_AAC.txt --recode A --out coding_nonsynonymous_pheno_SH3GL2_AAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:17579258:C:T_T</th>\n",
       "      <th>chr9:17747115:A:C_C</th>\n",
       "      <th>chr9:17761500:C:G_G</th>\n",
       "      <th>chr9:17786465:C:T_T</th>\n",
       "      <th>...</th>\n",
       "      <th>chr9:17793428:G:C_C</th>\n",
       "      <th>chr9:17793437:A:G_G</th>\n",
       "      <th>chr9:17793450:C:G_G</th>\n",
       "      <th>chr9:17793465:G:T_T</th>\n",
       "      <th>chr9:17793471:C:T_T</th>\n",
       "      <th>chr9:17793474:A:T_T</th>\n",
       "      <th>chr9:17795560:G:T_T</th>\n",
       "      <th>chr9:17795624:G:A_A</th>\n",
       "      <th>chr9:17795627:G:A_A</th>\n",
       "      <th>chr9:17795649:A:G_G</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000096_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000142_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000150_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000204_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000215_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   FID            IID  PAT  MAT  SEX  PHENOTYPE  chr9:17579258:C:T_T  \\\n",
       "0    0  BCM_000096_s1    0    0    1          2                    0   \n",
       "1    0  BCM_000142_s1    0    0    1          2                    0   \n",
       "2    0  BCM_000150_s1    0    0    2          2                    0   \n",
       "3    0  BCM_000204_s1    0    0    2          2                    0   \n",
       "4    0  BCM_000215_s1    0    0    1          2                    0   \n",
       "\n",
       "   chr9:17747115:A:C_C  chr9:17761500:C:G_G  chr9:17786465:C:T_T  ...  \\\n",
       "0                    0                    0                    0  ...   \n",
       "1                    0                    0                    0  ...   \n",
       "2                    0                    0                    0  ...   \n",
       "3                    0                    0                    0  ...   \n",
       "4                    0                    0                    0  ...   \n",
       "\n",
       "   chr9:17793428:G:C_C  chr9:17793437:A:G_G  chr9:17793450:C:G_G  \\\n",
       "0                    0                  0.0                    0   \n",
       "1                    0                  0.0                    0   \n",
       "2                    0                  0.0                    0   \n",
       "3                    0                  0.0                    0   \n",
       "4                    0                  0.0                    0   \n",
       "\n",
       "   chr9:17793465:G:T_T  chr9:17793471:C:T_T  chr9:17793474:A:T_T  \\\n",
       "0                    0                    0                    0   \n",
       "1                    0                    0                    0   \n",
       "2                    0                    0                    0   \n",
       "3                    0                    0                    0   \n",
       "4                    0                    0                    0   \n",
       "\n",
       "   chr9:17795560:G:T_T  chr9:17795624:G:A_A  chr9:17795627:G:A_A  \\\n",
       "0                    0                    0                    0   \n",
       "1                    0                    0                    0   \n",
       "2                    0                    0                    0   \n",
       "3                    0                    0                    0   \n",
       "4                    0                    0                    0   \n",
       "\n",
       "   chr9:17795649:A:G_G  \n",
       "0                    0  \n",
       "1                    0  \n",
       "2                    0  \n",
       "3                    0  \n",
       "4                    0  \n",
       "\n",
       "[5 rows x 32 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_AAC_recode = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_SH3GL2_AAC.raw', delim_whitespace=True)\n",
    "SH3GL2_AAC_recode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:17579258:C:T_T</th>\n",
       "      <th>chr9:17747115:A:C_C</th>\n",
       "      <th>chr9:17761500:C:G_G</th>\n",
       "      <th>chr9:17786465:C:T_T</th>\n",
       "      <th>...</th>\n",
       "      <th>chr9:17793428:G:C_C</th>\n",
       "      <th>chr9:17793437:A:G_G</th>\n",
       "      <th>chr9:17793450:C:G_G</th>\n",
       "      <th>chr9:17793465:G:T_T</th>\n",
       "      <th>chr9:17793471:C:T_T</th>\n",
       "      <th>chr9:17793474:A:T_T</th>\n",
       "      <th>chr9:17795560:G:T_T</th>\n",
       "      <th>chr9:17795624:G:A_A</th>\n",
       "      <th>chr9:17795627:G:A_A</th>\n",
       "      <th>chr9:17795649:A:G_G</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [FID, IID, PAT, MAT, SEX, PHENOTYPE, chr9:17579258:C:T_T, chr9:17747115:A:C_C, chr9:17761500:C:G_G, chr9:17786465:C:T_T, chr9:17786519:A:G_G, chr9:17787509:T:C_C, chr9:17789473:C:G_G, chr9:17789476:C:T_T, chr9:17789477:G:A_A, chr9:17789499:T:G_G, chr9:17789545:A:G_G, chr9:17791272:G:C_C, chr9:17791276:G:C_C, chr9:17791308:G:C_C, chr9:17791316:C:T_T, chr9:17793420:T:C_C, chr9:17793428:G:C_C, chr9:17793437:A:G_G, chr9:17793450:C:G_G, chr9:17793465:G:T_T, chr9:17793471:C:T_T, chr9:17793474:A:T_T, chr9:17795560:G:T_T, chr9:17795624:G:A_A, chr9:17795627:G:A_A, chr9:17795649:A:G_G]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 32 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_AAC_recode[SH3GL2_AAC_recode[\"chr9:17793465:G:T_T\"]==2]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save out results..!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR} {WORKSPACE_BUCKET}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {
    "height": "248.972px",
    "width": "263.062px"
   },
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "268.372px"
   },
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
