{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SH3GL2 - Single gene analysis in GP2 data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Description\n",
    "\n",
    "Using \"individual level data\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 0. Getting Started\n",
    "\n",
    "- Loading Python libraries\n",
    "- Defining functions\n",
    "- Installing packages\n",
    "\n",
    "### 1. Copy data from workspace to cloud environment\n",
    "\n",
    "### 2. Extract SH3GL2\n",
    "\n",
    "### 3. Annotate SH3GL2 variants\n",
    "\n",
    "### 4. Extract coding/non-syn variants\n",
    "\n",
    "### 5. Calculate frequency in cases versus controls\n",
    "\n",
    "### 6. Calculate frequency (homozygotes) in cases versus controls\n",
    "\n",
    "### 7. Save out results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting Started"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Loading Python libraries"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Defining functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "\n",
    "# Numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery\n",
    "\n",
    "#Import Sys\n",
    "import sys as sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "code_folding": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Billing and Workspace\n",
      "Workspace Name: Endophilin-A\n",
      "Billing Project: terra-18d8e41c\n",
      "Workspace Bucket, where you can upload and download data: gs://fc-f7a400c1-827e-48f8-b7b6-90c488a000a4\n",
      "\n",
      "GP2 v5.0\n",
      "Path to GP2 v5.0 Clinical Data: gs://gp2tier2/release5_11052023/clinical_data\n",
      "Path to GP2 v5.0 Raw Genotype Data: gs://gp2tier2/release5_11052023/raw_genotypes\n",
      "Path to GP2 v5.0 Imputed Genotype Data: gs://gp2tier2/release5_11052023/imputed_genotypes\n",
      "AMP-PD v3.0\n",
      "Path to AMP-PD v3.0 Clinical Data: gs://amp-pd-data/releases/2022_v3release_1115/clinical\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "\n",
    "## GP2 v5.0\n",
    "## Explicitly define release v5.0 path \n",
    "GP2_RELEASE_PATH = 'gs://gp2tier2/release5_11052023'\n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_META_RELEASE_PATH = f'{GP2_RELEASE_PATH}/meta_data'\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v5.0')\n",
    "print(f'Path to GP2 v5.0 Clinical Data: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v5.0 Raw Genotype Data: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v5.0 Imputed Genotype Data: {GP2_IMPUTED_GENO_PATH}')\n",
    "\n",
    "\n",
    "## AMP-PD v3.0\n",
    "## Explicitly define release v3.0 path \n",
    "AMP_RELEASE_PATH = 'gs://amp-pd-data/releases/2022_v3release_1115'\n",
    "AMP_CLINICAL_RELEASE_PATH = f'{AMP_RELEASE_PATH}/clinical'\n",
    "\n",
    "AMP_WGS_RELEASE_PATH = 'gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS'\n",
    "AMP_WGS_RELEASE_PLINK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'plink')\n",
    "AMP_WGS_RELEASE_PLINK_PFILES = os.path.join(AMP_WGS_RELEASE_PLINK_PATH, 'pfiles')\n",
    "\n",
    "print('AMP-PD v3.0')\n",
    "print(f'Path to AMP-PD v3.0 Clinical Data: {AMP_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PFILES}')\n",
    "print('')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install Plink 1.9 and Plink 2.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n",
      "Plink2 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi\n",
    "\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/alpha3/plink2_linux_avx2_20220603.zip\n",
    "unzip -o plink2_linux_avx2_20220603.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Remote restrictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 1.9 \n",
    "chmod u+x /home/jupyter/tools/plink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 2.0\n",
    "chmod u+x /home/jupyter/tools/plink2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install ANNOVAR: Download sources of annotation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGene.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneMrna.fa.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneVersion.txt.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotate_variation.pl\n",
      "coding_change.pl\n",
      "convert2annovar.pl\n",
      "example\n",
      "humandb\n",
      "retrieve_seq_from_fasta.pl\n",
      "table_annovar.pl\n",
      "variants_reduction.pl\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/annovar/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## EUR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Copy data from GP2 bucket to workspace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Making a working directory\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: mkdir -p /home/jupyter/SH3GL2_GP2/\n"
     ]
    }
   ],
   "source": [
    "# Make a directory\n",
    "print(\"Making a working directory\")\n",
    "WORK_DIR = f'/home/jupyter/SH3GL2_GP2/'\n",
    "shell_do(f'mkdir -p {WORK_DIR}') # f' means f-string - contains expressions to execute the code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-18d8e41c ls gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr10_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr10_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr10_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr10_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr10_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr10_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr10_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr10_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr11_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr11_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr11_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr11_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr11_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr11_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr11_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr11_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr12_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr12_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr12_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr12_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr12_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr12_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr12_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr12_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr13_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr13_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr13_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr13_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr13_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr13_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr13_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr13_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr14_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr14_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr14_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr14_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr14_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr14_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr14_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr14_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr15_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr15_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr15_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr15_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr15_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr15_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr15_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr15_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr16_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr16_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr16_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr16_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr16_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr16_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr16_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr16_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr17_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr17_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr17_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr17_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr17_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr17_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr17_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr17_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr18_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr18_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr18_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr18_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr18_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr18_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr18_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr18_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr19_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr19_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr19_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr19_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr19_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr19_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr19_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr19_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr1_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr1_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr1_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr1_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr1_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr1_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr1_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr1_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr20_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr20_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr20_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr20_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr20_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr20_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr20_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr20_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr21_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr21_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr21_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr21_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr21_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr21_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr21_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr21_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr22_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr22_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr22_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr22_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr22_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr22_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr22_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr22_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr2_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr2_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr2_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr2_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr2_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr2_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr2_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr2_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr3_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr3_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr3_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr3_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr3_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr3_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr3_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr3_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr4_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr4_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr4_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr4_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr4_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr4_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr4_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr4_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr5_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr5_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr5_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr5_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr5_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr5_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr5_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr5_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr6_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr6_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr6_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr6_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr6_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr6_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr6_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr6_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr7_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr7_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr7_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr7_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr7_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr7_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr7_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr7_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr8_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr8_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr8_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr8_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr8_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr8_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr8_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr8_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr9_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr9_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr9_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr9_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr9_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr9_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr9_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chr9_EUR_release5_no_mac_filter.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chrX_EUR_release5.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chrX_EUR_release5.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chrX_EUR_release5.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chrX_EUR_release5.pvar\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chrX_EUR_release5_no_mac_filter.log\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chrX_EUR_release5_no_mac_filter.pgen\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chrX_EUR_release5_no_mac_filter.psam\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/EUR/chrX_EUR_release5_no_mac_filter.pvar\n"
     ]
    }
   ],
   "source": [
    "# Check directory where GP2 data is\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_IMPUTED_GENO_PATH}/EUR/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {GP2_IMPUTED_GENO_PATH}/EUR/chr9_EUR_release5_no_mac_filter* {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create a covariate file with GP2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_CLINICAL_RELEASE_PATH}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {GP2_CLINICAL_RELEASE_PATH}/master_key_release5_final.csv {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "cov = pd.read_csv(f'{WORK_DIR}/master_key_release5_final.csv')\n",
    "cov.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "cov_reduced = cov[['GP2sampleID','GP2sampleID','sex_for_qc', 'age', 'phenotype']]\n",
    "cov_reduced.columns = ['FID','IID', 'SEX','AGE','PHENO']\n",
    "#cov_reduced"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "conditions = [\n",
    "    (cov_reduced['PHENO'] == \"PD\"),\n",
    "    (cov_reduced['PHENO'] == \"Control\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "choices = [2,1]\n",
    "cov_reduced['PHENOTYPE'] = np.select(conditions, choices, default=-9).astype(np.int64)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "cov_reduced.reset_index(inplace=True)\n",
    "cov_reduced.drop(columns=[\"index\"], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cov_reduced.drop(columns=['PHENO'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sex = cov_reduced[['FID','IID','SEX']]\n",
    "sex.to_csv(f'{WORK_DIR}/SEX.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno = cov_reduced[['FID','IID','PHENOTYPE']]\n",
    "pheno.to_csv(f'{WORK_DIR}/PHENO.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract SH3GL2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# SH3GL2: gene positions on hg38 (from https://useast.ensembl.org/index.html)\n",
    "/home/jupyter/tools/plink2 \\\n",
    "--pfile chr9_EUR_release5_no_mac_filter \\\n",
    "--chr 9 \\\n",
    "--from-bp 17579066  \\\n",
    "--to-bp 17797124 \\\n",
    "--make-bed \\\n",
    "--out pheno_SH3GL2_EUR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualize plink files bim, fam and bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize bim file\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_SH3GL2_EUR.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize fam file\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_SH3GL2_EUR.fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# Turn binary files into VCF\n",
    "/home/jupyter/tools/plink \\\n",
    "--bfile pheno_SH3GL2_EUR \\\n",
    "--recode vcf-fid \\\n",
    "--out pheno_SH3GL2_EUR_nomacfilter"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate SH3GL2 variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "export PATH=$PATH:/home/jupyter/tools/rvtests/third/tabix-0.2.6/\n",
    "\n",
    "### Bgzip and Tabix\n",
    "bgzip pheno_SH3GL2_EUR_nomacfilter.vcf\n",
    "\n",
    "tabix -f -p vcf pheno_SH3GL2_EUR_nomacfilter.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "### Annotate variants using ANNOVAR: https://annovar.openbioinformatics.org/en/latest/ \n",
    "perl /home/jupyter/tools/annovar/table_annovar.pl pheno_SH3GL2_EUR_nomacfilter.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out pheno_SH3GL2_EUR.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract coding and non-syn variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Visualize multianno file\n",
    "SH3GL2_EUR = pd.read_csv(f'{WORK_DIR}/pheno_SH3GL2_EUR.annovar.hg38_multianno.txt', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "SH3GL2_EUR['Chr_bp'] = SH3GL2_EUR['Chr'].astype(str) + ':' + SH3GL2_EUR['Start'].astype(str) + ':' + SH3GL2_EUR['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "SH3GL2_EUR = SH3GL2_EUR.drop_duplicates(subset=['Chr_bp'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "SH3GL2_EUR = SH3GL2_EUR[SH3GL2_EUR['Gene.refGene']=='SH3GL2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "code_folding": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr               17368\n",
       "Start             17368\n",
       "End               17368\n",
       "Ref               17368\n",
       "Alt               17368\n",
       "                  ...  \n",
       "Otherinfo15365    17368\n",
       "Otherinfo15366    17368\n",
       "Otherinfo15367    17368\n",
       "Otherinfo15368    17368\n",
       "Chr_bp            17368\n",
       "Length: 15380, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Number of total varints \n",
    "SH3GL2_EUR.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr_bp</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9:17579077:17579077</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9:17579083:17579083</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9:17579085:17579085</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9:17579086:17579086</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9:17579090:17579090</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17864</th>\n",
       "      <td>9:17797008:17797008</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17865</th>\n",
       "      <td>9:17797011:17797011</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17866</th>\n",
       "      <td>9:17797047:17797047</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17867</th>\n",
       "      <td>9:17797079:17797079</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17868</th>\n",
       "      <td>9:17797083:17797083</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17368 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Chr_bp Gene.refGene Func.refGene ExonicFunc.refGene\n",
       "0      9:17579077:17579077       SH3GL2         UTR5                  .\n",
       "1      9:17579083:17579083       SH3GL2         UTR5                  .\n",
       "2      9:17579085:17579085       SH3GL2         UTR5                  .\n",
       "3      9:17579086:17579086       SH3GL2         UTR5                  .\n",
       "4      9:17579090:17579090       SH3GL2         UTR5                  .\n",
       "...                    ...          ...          ...                ...\n",
       "17864  9:17797008:17797008       SH3GL2         UTR3                  .\n",
       "17865  9:17797011:17797011       SH3GL2         UTR3                  .\n",
       "17866  9:17797047:17797047       SH3GL2         UTR3                  .\n",
       "17867  9:17797079:17797079       SH3GL2         UTR3                  .\n",
       "17868  9:17797083:17797083       SH3GL2         UTR3                  .\n",
       "\n",
       "[17368 rows x 4 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reduced_SH3GL2_EUR = SH3GL2_EUR[[\"Chr_bp\", \"Gene.refGene\", 'Func.refGene','ExonicFunc.refGene']]\n",
    "reduced_SH3GL2_EUR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "reduced_SH3GL2_EUR.to_csv(f'{WORKSPACE_BUCKET}/reduced_SH3GL2_EUR.txt', sep = '\\t', index = False, header= True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "intronic    17164\n",
       "UTR3          122\n",
       "exonic         48\n",
       "UTR5           34\n",
       "Name: Func.refGene, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_EUR['Func.refGene'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       ".                    17320\n",
       "nonsynonymous SNV       30\n",
       "synonymous SNV          18\n",
       "Name: ExonicFunc.refGene, dtype: int64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_EUR['ExonicFunc.refGene'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr               48\n",
       "Start             48\n",
       "End               48\n",
       "Ref               48\n",
       "Alt               48\n",
       "                  ..\n",
       "Otherinfo15365    48\n",
       "Otherinfo15366    48\n",
       "Otherinfo15367    48\n",
       "Otherinfo15368    48\n",
       "Chr_bp            48\n",
       "Length: 15380, dtype: int64"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic variants\n",
    "coding_EUR = SH3GL2_EUR[SH3GL2_EUR['Func.refGene'] == 'exonic']\n",
    "coding_EUR.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo15360</th>\n",
       "      <th>Otherinfo15361</th>\n",
       "      <th>Otherinfo15362</th>\n",
       "      <th>Otherinfo15363</th>\n",
       "      <th>Otherinfo15364</th>\n",
       "      <th>Otherinfo15365</th>\n",
       "      <th>Otherinfo15366</th>\n",
       "      <th>Otherinfo15367</th>\n",
       "      <th>Otherinfo15368</th>\n",
       "      <th>Chr_bp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13861</th>\n",
       "      <td>9</td>\n",
       "      <td>17747115</td>\n",
       "      <td>17747115</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon2:c.A95C:p.D32A</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17747115:17747115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15037</th>\n",
       "      <td>9</td>\n",
       "      <td>17761500</td>\n",
       "      <td>17761500</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon3:c.C178G:p.P60A</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17761500:17761500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17096</th>\n",
       "      <td>9</td>\n",
       "      <td>17787498</td>\n",
       "      <td>17787498</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon5:c.T450A:p.D150E</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17787498:17787498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17231</th>\n",
       "      <td>9</td>\n",
       "      <td>17789425</td>\n",
       "      <td>17789425</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.G499T:p.D167Y</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789425:17789425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17234</th>\n",
       "      <td>9</td>\n",
       "      <td>17789473</td>\n",
       "      <td>17789473</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.C547G:p.L183V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789473:17789473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17235</th>\n",
       "      <td>9</td>\n",
       "      <td>17789499</td>\n",
       "      <td>17789499</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.T573G:p.D191E</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789499:17789499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17236</th>\n",
       "      <td>9</td>\n",
       "      <td>17789545</td>\n",
       "      <td>17789545</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.A619G:p.M207V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789545:17789545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17386</th>\n",
       "      <td>9</td>\n",
       "      <td>17791272</td>\n",
       "      <td>17791272</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.G666C:p.Q222H</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791272:17791272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17387</th>\n",
       "      <td>9</td>\n",
       "      <td>17791276</td>\n",
       "      <td>17791276</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.G670C:p.E224Q</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791276:17791276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17388</th>\n",
       "      <td>9</td>\n",
       "      <td>17791290</td>\n",
       "      <td>17791290</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.G684C:p.Q228H</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791290:17791290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17390</th>\n",
       "      <td>9</td>\n",
       "      <td>17791308</td>\n",
       "      <td>17791308</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.G702C:p.Q234H</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791308:17791308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17392</th>\n",
       "      <td>9</td>\n",
       "      <td>17791316</td>\n",
       "      <td>17791316</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.C710T:p.T237M</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791316:17791316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17393</th>\n",
       "      <td>9</td>\n",
       "      <td>17791318</td>\n",
       "      <td>17791318</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.G712A:p.V238I</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791318:17791318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17557</th>\n",
       "      <td>9</td>\n",
       "      <td>17793386</td>\n",
       "      <td>17793386</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.C748G:p.Q250E</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793386:17793386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17558</th>\n",
       "      <td>9</td>\n",
       "      <td>17793389</td>\n",
       "      <td>17793389</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.C751G:p.P251A</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793389:17793389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17561</th>\n",
       "      <td>9</td>\n",
       "      <td>17793407</td>\n",
       "      <td>17793407</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.C769T:p.P257S</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793407:17793407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17562</th>\n",
       "      <td>9</td>\n",
       "      <td>17793417</td>\n",
       "      <td>17793417</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.G779A:p.R260Q</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793417:17793417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17563</th>\n",
       "      <td>9</td>\n",
       "      <td>17793437</td>\n",
       "      <td>17793437</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.A799G:p.T267A</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793437:17793437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17564</th>\n",
       "      <td>9</td>\n",
       "      <td>17793459</td>\n",
       "      <td>17793459</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.A821G:p.N274S</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793459:17793459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17565</th>\n",
       "      <td>9</td>\n",
       "      <td>17793465</td>\n",
       "      <td>17793465</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.G827T:p.G276V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793465:17793465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17566</th>\n",
       "      <td>9</td>\n",
       "      <td>17793471</td>\n",
       "      <td>17793471</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.C833T:p.S278F</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793471:17793471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17567</th>\n",
       "      <td>9</td>\n",
       "      <td>17793477</td>\n",
       "      <td>17793477</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.C839T:p.T280I</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793477:17793477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17735</th>\n",
       "      <td>9</td>\n",
       "      <td>17795552</td>\n",
       "      <td>17795552</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.A868G:p.M290V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795552:17795552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17736</th>\n",
       "      <td>9</td>\n",
       "      <td>17795571</td>\n",
       "      <td>17795571</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.G887A:p.R296Q</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795571:17795571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17737</th>\n",
       "      <td>9</td>\n",
       "      <td>17795582</td>\n",
       "      <td>17795582</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.G898A:p.D300N</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795582:17795582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17740</th>\n",
       "      <td>9</td>\n",
       "      <td>17795624</td>\n",
       "      <td>17795624</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.G940A:p.G314S</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795624:17795624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17742</th>\n",
       "      <td>9</td>\n",
       "      <td>17795649</td>\n",
       "      <td>17795649</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.A965G:p.Q322R</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795649:17795649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17743</th>\n",
       "      <td>9</td>\n",
       "      <td>17795682</td>\n",
       "      <td>17795682</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.A998G:p.H333R</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795682:17795682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17744</th>\n",
       "      <td>9</td>\n",
       "      <td>17795685</td>\n",
       "      <td>17795685</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.G1001A:p.G334D</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795685:17795685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17746</th>\n",
       "      <td>9</td>\n",
       "      <td>17795705</td>\n",
       "      <td>17795705</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.A1021G:p.I341V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795705:17795705</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30 rows Ã— 15380 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chr     Start       End Ref Alt Func.refGene Gene.refGene  \\\n",
       "13861    9  17747115  17747115   A   C       exonic       SH3GL2   \n",
       "15037    9  17761500  17761500   C   G       exonic       SH3GL2   \n",
       "17096    9  17787498  17787498   T   A       exonic       SH3GL2   \n",
       "17231    9  17789425  17789425   G   T       exonic       SH3GL2   \n",
       "17234    9  17789473  17789473   C   G       exonic       SH3GL2   \n",
       "17235    9  17789499  17789499   T   G       exonic       SH3GL2   \n",
       "17236    9  17789545  17789545   A   G       exonic       SH3GL2   \n",
       "17386    9  17791272  17791272   G   C       exonic       SH3GL2   \n",
       "17387    9  17791276  17791276   G   C       exonic       SH3GL2   \n",
       "17388    9  17791290  17791290   G   C       exonic       SH3GL2   \n",
       "17390    9  17791308  17791308   G   C       exonic       SH3GL2   \n",
       "17392    9  17791316  17791316   C   T       exonic       SH3GL2   \n",
       "17393    9  17791318  17791318   G   A       exonic       SH3GL2   \n",
       "17557    9  17793386  17793386   C   G       exonic       SH3GL2   \n",
       "17558    9  17793389  17793389   C   G       exonic       SH3GL2   \n",
       "17561    9  17793407  17793407   C   T       exonic       SH3GL2   \n",
       "17562    9  17793417  17793417   G   A       exonic       SH3GL2   \n",
       "17563    9  17793437  17793437   A   G       exonic       SH3GL2   \n",
       "17564    9  17793459  17793459   A   G       exonic       SH3GL2   \n",
       "17565    9  17793465  17793465   G   T       exonic       SH3GL2   \n",
       "17566    9  17793471  17793471   C   T       exonic       SH3GL2   \n",
       "17567    9  17793477  17793477   C   T       exonic       SH3GL2   \n",
       "17735    9  17795552  17795552   A   G       exonic       SH3GL2   \n",
       "17736    9  17795571  17795571   G   A       exonic       SH3GL2   \n",
       "17737    9  17795582  17795582   G   A       exonic       SH3GL2   \n",
       "17740    9  17795624  17795624   G   A       exonic       SH3GL2   \n",
       "17742    9  17795649  17795649   A   G       exonic       SH3GL2   \n",
       "17743    9  17795682  17795682   A   G       exonic       SH3GL2   \n",
       "17744    9  17795685  17795685   G   A       exonic       SH3GL2   \n",
       "17746    9  17795705  17795705   A   G       exonic       SH3GL2   \n",
       "\n",
       "      GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "13861                  .  nonsynonymous SNV   \n",
       "15037                  .  nonsynonymous SNV   \n",
       "17096                  .  nonsynonymous SNV   \n",
       "17231                  .  nonsynonymous SNV   \n",
       "17234                  .  nonsynonymous SNV   \n",
       "17235                  .  nonsynonymous SNV   \n",
       "17236                  .  nonsynonymous SNV   \n",
       "17386                  .  nonsynonymous SNV   \n",
       "17387                  .  nonsynonymous SNV   \n",
       "17388                  .  nonsynonymous SNV   \n",
       "17390                  .  nonsynonymous SNV   \n",
       "17392                  .  nonsynonymous SNV   \n",
       "17393                  .  nonsynonymous SNV   \n",
       "17557                  .  nonsynonymous SNV   \n",
       "17558                  .  nonsynonymous SNV   \n",
       "17561                  .  nonsynonymous SNV   \n",
       "17562                  .  nonsynonymous SNV   \n",
       "17563                  .  nonsynonymous SNV   \n",
       "17564                  .  nonsynonymous SNV   \n",
       "17565                  .  nonsynonymous SNV   \n",
       "17566                  .  nonsynonymous SNV   \n",
       "17567                  .  nonsynonymous SNV   \n",
       "17735                  .  nonsynonymous SNV   \n",
       "17736                  .  nonsynonymous SNV   \n",
       "17737                  .  nonsynonymous SNV   \n",
       "17740                  .  nonsynonymous SNV   \n",
       "17742                  .  nonsynonymous SNV   \n",
       "17743                  .  nonsynonymous SNV   \n",
       "17744                  .  nonsynonymous SNV   \n",
       "17746                  .  nonsynonymous SNV   \n",
       "\n",
       "                              AAChange.refGene  ... Otherinfo15360  \\\n",
       "13861     SH3GL2:NM_003026:exon2:c.A95C:p.D32A  ...            0/0   \n",
       "15037    SH3GL2:NM_003026:exon3:c.C178G:p.P60A  ...            0/0   \n",
       "17096   SH3GL2:NM_003026:exon5:c.T450A:p.D150E  ...            0/0   \n",
       "17231   SH3GL2:NM_003026:exon6:c.G499T:p.D167Y  ...            0/0   \n",
       "17234   SH3GL2:NM_003026:exon6:c.C547G:p.L183V  ...            0/0   \n",
       "17235   SH3GL2:NM_003026:exon6:c.T573G:p.D191E  ...            0/0   \n",
       "17236   SH3GL2:NM_003026:exon6:c.A619G:p.M207V  ...            0/0   \n",
       "17386   SH3GL2:NM_003026:exon7:c.G666C:p.Q222H  ...            0/0   \n",
       "17387   SH3GL2:NM_003026:exon7:c.G670C:p.E224Q  ...            0/0   \n",
       "17388   SH3GL2:NM_003026:exon7:c.G684C:p.Q228H  ...            0/0   \n",
       "17390   SH3GL2:NM_003026:exon7:c.G702C:p.Q234H  ...            0/0   \n",
       "17392   SH3GL2:NM_003026:exon7:c.C710T:p.T237M  ...            0/0   \n",
       "17393   SH3GL2:NM_003026:exon7:c.G712A:p.V238I  ...            0/0   \n",
       "17557   SH3GL2:NM_003026:exon8:c.C748G:p.Q250E  ...            0/0   \n",
       "17558   SH3GL2:NM_003026:exon8:c.C751G:p.P251A  ...            0/0   \n",
       "17561   SH3GL2:NM_003026:exon8:c.C769T:p.P257S  ...            0/0   \n",
       "17562   SH3GL2:NM_003026:exon8:c.G779A:p.R260Q  ...            0/0   \n",
       "17563   SH3GL2:NM_003026:exon8:c.A799G:p.T267A  ...            0/0   \n",
       "17564   SH3GL2:NM_003026:exon8:c.A821G:p.N274S  ...            0/0   \n",
       "17565   SH3GL2:NM_003026:exon8:c.G827T:p.G276V  ...            0/0   \n",
       "17566   SH3GL2:NM_003026:exon8:c.C833T:p.S278F  ...            0/0   \n",
       "17567   SH3GL2:NM_003026:exon8:c.C839T:p.T280I  ...            0/0   \n",
       "17735   SH3GL2:NM_003026:exon9:c.A868G:p.M290V  ...            0/0   \n",
       "17736   SH3GL2:NM_003026:exon9:c.G887A:p.R296Q  ...            0/0   \n",
       "17737   SH3GL2:NM_003026:exon9:c.G898A:p.D300N  ...            0/0   \n",
       "17740   SH3GL2:NM_003026:exon9:c.G940A:p.G314S  ...            0/0   \n",
       "17742   SH3GL2:NM_003026:exon9:c.A965G:p.Q322R  ...            0/0   \n",
       "17743   SH3GL2:NM_003026:exon9:c.A998G:p.H333R  ...            0/0   \n",
       "17744  SH3GL2:NM_003026:exon9:c.G1001A:p.G334D  ...            0/0   \n",
       "17746  SH3GL2:NM_003026:exon9:c.A1021G:p.I341V  ...            0/0   \n",
       "\n",
       "       Otherinfo15361 Otherinfo15362 Otherinfo15363  Otherinfo15364  \\\n",
       "13861             0/0            0/0            0/0             0/0   \n",
       "15037             0/0            0/0            0/0             0/0   \n",
       "17096             0/0            0/0            0/0             0/0   \n",
       "17231             0/0            0/0            0/0             0/0   \n",
       "17234             0/0            0/0            0/0             0/0   \n",
       "17235             0/0            0/0            0/0             0/0   \n",
       "17236             0/0            0/0            0/0             0/0   \n",
       "17386             0/0            0/0            0/0             0/0   \n",
       "17387             0/0            0/0            0/0             0/0   \n",
       "17388             0/0            0/0            0/0             0/0   \n",
       "17390             0/0            0/0            0/0             0/0   \n",
       "17392             0/0            0/0            0/0             0/0   \n",
       "17393             0/0            0/0            0/0             0/0   \n",
       "17557             0/0            0/0            0/0             0/0   \n",
       "17558             0/0            0/0            0/0             0/0   \n",
       "17561             0/0            0/0            0/0             0/0   \n",
       "17562             0/0            0/0            0/0             0/0   \n",
       "17563             0/0            0/0            0/0             0/0   \n",
       "17564             0/0            0/0            0/0             0/0   \n",
       "17565             0/0            0/0            0/0             0/0   \n",
       "17566             0/0            0/0            0/0             0/0   \n",
       "17567             0/0            0/0            0/0             0/0   \n",
       "17735             0/0            0/0            0/0             0/0   \n",
       "17736             0/0            0/0            0/0             0/0   \n",
       "17737             0/0            0/0            0/0             0/0   \n",
       "17740             0/0            0/0            0/0             0/0   \n",
       "17742             0/0            0/0            0/0             0/0   \n",
       "17743             0/0            0/0            0/0             0/0   \n",
       "17744             0/0            0/0            0/0             0/0   \n",
       "17746             0/0            0/0            0/0             0/0   \n",
       "\n",
       "       Otherinfo15365 Otherinfo15366 Otherinfo15367 Otherinfo15368  \\\n",
       "13861             0/0            0/0            0/0            0/0   \n",
       "15037             0/0            0/0            0/0            0/0   \n",
       "17096             0/0            0/0            0/0            0/0   \n",
       "17231             0/0            0/0            0/0            0/0   \n",
       "17234             0/0            0/0            0/0            0/0   \n",
       "17235             0/0            0/0            0/0            0/0   \n",
       "17236             0/0            0/0            0/0            0/0   \n",
       "17386             0/0            0/0            0/0            0/0   \n",
       "17387             0/0            0/0            0/0            0/0   \n",
       "17388             0/0            0/0            0/0            0/0   \n",
       "17390             0/0            0/0            0/0            0/0   \n",
       "17392             0/0            0/0            0/0            0/0   \n",
       "17393             0/0            0/0            0/0            0/0   \n",
       "17557             0/0            0/0            0/0            0/0   \n",
       "17558             0/0            0/0            0/0            0/0   \n",
       "17561             0/0            0/0            0/0            0/0   \n",
       "17562             0/0            0/0            0/0            0/0   \n",
       "17563             0/0            0/0            0/0            0/0   \n",
       "17564             0/0            0/0            0/0            0/0   \n",
       "17565             0/0            0/0            0/0            0/0   \n",
       "17566             0/0            0/0            0/0            0/0   \n",
       "17567             0/0            0/0            0/0            0/0   \n",
       "17735             0/0            0/0            0/0            0/0   \n",
       "17736             0/0            0/0            0/0            0/0   \n",
       "17737             0/0            0/0            0/0            0/0   \n",
       "17740             0/0            0/0            0/0            0/0   \n",
       "17742             0/0            0/0            0/0            0/0   \n",
       "17743             0/0            0/0            0/0            0/0   \n",
       "17744             0/0            0/0            0/0            0/0   \n",
       "17746             0/0            0/0            0/0            0/0   \n",
       "\n",
       "                    Chr_bp  \n",
       "13861  9:17747115:17747115  \n",
       "15037  9:17761500:17761500  \n",
       "17096  9:17787498:17787498  \n",
       "17231  9:17789425:17789425  \n",
       "17234  9:17789473:17789473  \n",
       "17235  9:17789499:17789499  \n",
       "17236  9:17789545:17789545  \n",
       "17386  9:17791272:17791272  \n",
       "17387  9:17791276:17791276  \n",
       "17388  9:17791290:17791290  \n",
       "17390  9:17791308:17791308  \n",
       "17392  9:17791316:17791316  \n",
       "17393  9:17791318:17791318  \n",
       "17557  9:17793386:17793386  \n",
       "17558  9:17793389:17793389  \n",
       "17561  9:17793407:17793407  \n",
       "17562  9:17793417:17793417  \n",
       "17563  9:17793437:17793437  \n",
       "17564  9:17793459:17793459  \n",
       "17565  9:17793465:17793465  \n",
       "17566  9:17793471:17793471  \n",
       "17567  9:17793477:17793477  \n",
       "17735  9:17795552:17795552  \n",
       "17736  9:17795571:17795571  \n",
       "17737  9:17795582:17795582  \n",
       "17740  9:17795624:17795624  \n",
       "17742  9:17795649:17795649  \n",
       "17743  9:17795682:17795682  \n",
       "17744  9:17795685:17795685  \n",
       "17746  9:17795705:17795705  \n",
       "\n",
       "[30 rows x 15380 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic and non-syn vars\n",
    "coding_nonsynonymous_EUR = SH3GL2_EUR[(SH3GL2_EUR['Func.refGene'] == 'exonic') & (SH3GL2_EUR['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_nonsynonymous_EUR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "coding_nonsynonymous_EUR.to_csv(f'{WORK_DIR}/coding_nonsynonymous.txt', sep = '\\t', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculate freq - cases vs controls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "reduced_coding_nonsynonymous_EUR = coding_nonsynonymous_EUR[[\"Chr\", \"Start\", \"End\", \"Gene.refGene\"]]\n",
    "reduced_coding_nonsynonymous_EUR.to_csv(f'{WORK_DIR}/reduced_coding_nonsynonymous_EUR.txt', sep = '\\t', index = False, header= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_SH3GL2_EUR.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile pheno_SH3GL2_EUR\n",
      "  --ci 0.95\n",
      "  --extract range reduced_coding_nonsynonymous_EUR.txt\n",
      "  --out coding_nonsynonymous_pheno_SH3GL2_EUR\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "17869 variants loaded from .bim file.\n",
      "15356 people (9460 males, 5896 females) loaded from .fam.\n",
      "14196 phenotype values loaded from .fam.\n",
      "--extract range: 17839 variants excluded.\n",
      "--extract range: 30 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 15356 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999954.\n",
      "30 variants and 15356 people pass filters and QC.\n",
      "Among remaining phenotypes, 9230 are cases and 4966 are controls.  (1160\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to coding_nonsynonymous_pheno_SH3GL2_EUR.assoc ...\n",
      "0%\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_SH3GL2_EUR --extract range reduced_coding_nonsynonymous_EUR.txt --assoc --ci 0.95 --out coding_nonsynonymous_pheno_SH3GL2_EUR --allow-no-sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17747115:A:C</td>\n",
       "      <td>17747115</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17761500:C:G</td>\n",
       "      <td>17761500</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000108</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>C</td>\n",
       "      <td>0.003584</td>\n",
       "      <td>0.9523</td>\n",
       "      <td>1.076</td>\n",
       "      <td>1.2250</td>\n",
       "      <td>0.09756</td>\n",
       "      <td>11.870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17787498:T:A</td>\n",
       "      <td>17787498</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000162</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>T</td>\n",
       "      <td>0.175300</td>\n",
       "      <td>0.6754</td>\n",
       "      <td>1.614</td>\n",
       "      <td>1.1550</td>\n",
       "      <td>0.16790</td>\n",
       "      <td>15.520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789425:G:T</td>\n",
       "      <td>17789425</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789473:C:G</td>\n",
       "      <td>17789473</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789499:T:G</td>\n",
       "      <td>17789499</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789545:A:G</td>\n",
       "      <td>17789545</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.538000</td>\n",
       "      <td>0.4632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17791272:G:C</td>\n",
       "      <td>17791272</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17791276:G:C</td>\n",
       "      <td>17791276</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17791290:G:C</td>\n",
       "      <td>17791290</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.538000</td>\n",
       "      <td>0.4632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17791308:G:C</td>\n",
       "      <td>17791308</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17791316:C:T</td>\n",
       "      <td>17791316</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000162</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>C</td>\n",
       "      <td>0.174900</td>\n",
       "      <td>0.6758</td>\n",
       "      <td>1.613</td>\n",
       "      <td>1.1550</td>\n",
       "      <td>0.16780</td>\n",
       "      <td>15.510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17791318:G:A</td>\n",
       "      <td>17791318</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793386:C:G</td>\n",
       "      <td>17793386</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793389:C:G</td>\n",
       "      <td>17793389</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793407:C:T</td>\n",
       "      <td>17793407</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793417:G:A</td>\n",
       "      <td>17793417</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>G</td>\n",
       "      <td>1.859000</td>\n",
       "      <td>0.1728</td>\n",
       "      <td>0.000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793437:A:G</td>\n",
       "      <td>17793437</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>A</td>\n",
       "      <td>1.859000</td>\n",
       "      <td>0.1728</td>\n",
       "      <td>0.000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793459:A:G</td>\n",
       "      <td>17793459</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793465:G:T</td>\n",
       "      <td>17793465</td>\n",
       "      <td>T</td>\n",
       "      <td>0.003955</td>\n",
       "      <td>0.003828</td>\n",
       "      <td>G</td>\n",
       "      <td>0.027060</td>\n",
       "      <td>0.8693</td>\n",
       "      <td>1.034</td>\n",
       "      <td>0.2004</td>\n",
       "      <td>0.69780</td>\n",
       "      <td>1.531</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793471:C:T</td>\n",
       "      <td>17793471</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793477:C:T</td>\n",
       "      <td>17793477</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795552:A:G</td>\n",
       "      <td>17795552</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795571:G:A</td>\n",
       "      <td>17795571</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795582:G:A</td>\n",
       "      <td>17795582</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.538000</td>\n",
       "      <td>0.4632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795624:G:A</td>\n",
       "      <td>17795624</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.538000</td>\n",
       "      <td>0.4632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795649:A:G</td>\n",
       "      <td>17795649</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795682:A:G</td>\n",
       "      <td>17795682</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795685:G:A</td>\n",
       "      <td>17795685</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795705:A:G</td>\n",
       "      <td>17795705</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.538100</td>\n",
       "      <td>0.4632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHR                SNP        BP A1       F_A       F_U A2     CHISQ  \\\n",
       "0     9  chr9:17747115:A:C  17747115  C  0.000000  0.000000  A       NaN   \n",
       "1     9  chr9:17761500:C:G  17761500  G  0.000108  0.000101  C  0.003584   \n",
       "2     9  chr9:17787498:T:A  17787498  A  0.000162  0.000101  T  0.175300   \n",
       "3     9  chr9:17789425:G:T  17789425  T  0.000000  0.000000  G       NaN   \n",
       "4     9  chr9:17789473:C:G  17789473  G  0.000000  0.000000  C       NaN   \n",
       "5     9  chr9:17789499:T:G  17789499  G  0.000000  0.000000  T       NaN   \n",
       "6     9  chr9:17789545:A:G  17789545  G  0.000054  0.000000  A  0.538000   \n",
       "7     9  chr9:17791272:G:C  17791272  C  0.000000  0.000000  G       NaN   \n",
       "8     9  chr9:17791276:G:C  17791276  C  0.000000  0.000000  G       NaN   \n",
       "9     9  chr9:17791290:G:C  17791290  C  0.000054  0.000000  G  0.538000   \n",
       "10    9  chr9:17791308:G:C  17791308  C  0.000000  0.000000  G       NaN   \n",
       "11    9  chr9:17791316:C:T  17791316  T  0.000162  0.000101  C  0.174900   \n",
       "12    9  chr9:17791318:G:A  17791318  A  0.000000  0.000000  G       NaN   \n",
       "13    9  chr9:17793386:C:G  17793386  G  0.000000  0.000000  C       NaN   \n",
       "14    9  chr9:17793389:C:G  17793389  G  0.000000  0.000000  C       NaN   \n",
       "15    9  chr9:17793407:C:T  17793407  T  0.000000  0.000000  C       NaN   \n",
       "16    9  chr9:17793417:G:A  17793417  A  0.000000  0.000101  G  1.859000   \n",
       "17    9  chr9:17793437:A:G  17793437  G  0.000000  0.000101  A  1.859000   \n",
       "18    9  chr9:17793459:A:G  17793459  G  0.000000  0.000000  A       NaN   \n",
       "19    9  chr9:17793465:G:T  17793465  T  0.003955  0.003828  G  0.027060   \n",
       "20    9  chr9:17793471:C:T  17793471  T  0.000000  0.000000  C       NaN   \n",
       "21    9  chr9:17793477:C:T  17793477  T  0.000000  0.000000  C       NaN   \n",
       "22    9  chr9:17795552:A:G  17795552  G  0.000000  0.000000  A       NaN   \n",
       "23    9  chr9:17795571:G:A  17795571  A  0.000000  0.000000  G       NaN   \n",
       "24    9  chr9:17795582:G:A  17795582  A  0.000054  0.000000  G  0.538000   \n",
       "25    9  chr9:17795624:G:A  17795624  A  0.000054  0.000000  G  0.538000   \n",
       "26    9  chr9:17795649:A:G  17795649  G  0.000000  0.000000  A       NaN   \n",
       "27    9  chr9:17795682:A:G  17795682  G  0.000000  0.000000  A       NaN   \n",
       "28    9  chr9:17795685:G:A  17795685  A  0.000000  0.000000  G       NaN   \n",
       "29    9  chr9:17795705:A:G  17795705  G  0.000054  0.000000  A  0.538100   \n",
       "\n",
       "         P     OR      SE      L95     U95  \n",
       "0      NaN    NaN     NaN      NaN     NaN  \n",
       "1   0.9523  1.076  1.2250  0.09756  11.870  \n",
       "2   0.6754  1.614  1.1550  0.16790  15.520  \n",
       "3      NaN    NaN     NaN      NaN     NaN  \n",
       "4      NaN    NaN     NaN      NaN     NaN  \n",
       "5      NaN    NaN     NaN      NaN     NaN  \n",
       "6   0.4632    NaN     NaN      NaN     NaN  \n",
       "7      NaN    NaN     NaN      NaN     NaN  \n",
       "8      NaN    NaN     NaN      NaN     NaN  \n",
       "9   0.4632    NaN     NaN      NaN     NaN  \n",
       "10     NaN    NaN     NaN      NaN     NaN  \n",
       "11  0.6758  1.613  1.1550  0.16780  15.510  \n",
       "12     NaN    NaN     NaN      NaN     NaN  \n",
       "13     NaN    NaN     NaN      NaN     NaN  \n",
       "14     NaN    NaN     NaN      NaN     NaN  \n",
       "15     NaN    NaN     NaN      NaN     NaN  \n",
       "16  0.1728  0.000     inf  0.00000     NaN  \n",
       "17  0.1728  0.000     inf  0.00000     NaN  \n",
       "18     NaN    NaN     NaN      NaN     NaN  \n",
       "19  0.8693  1.034  0.2004  0.69780   1.531  \n",
       "20     NaN    NaN     NaN      NaN     NaN  \n",
       "21     NaN    NaN     NaN      NaN     NaN  \n",
       "22     NaN    NaN     NaN      NaN     NaN  \n",
       "23     NaN    NaN     NaN      NaN     NaN  \n",
       "24  0.4632    NaN     NaN      NaN     NaN  \n",
       "25  0.4632    NaN     NaN      NaN     NaN  \n",
       "26     NaN    NaN     NaN      NaN     NaN  \n",
       "27     NaN    NaN     NaN      NaN     NaN  \n",
       "28     NaN    NaN     NaN      NaN     NaN  \n",
       "29  0.4632    NaN     NaN      NaN     NaN  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_EUR_freq = pd.read_csv(f'{WORK_DIR}coding_nonsynonymous_pheno_SH3GL2_EUR.assoc', delim_whitespace=True)\n",
    "SH3GL2_EUR_freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_SH3GL2_EUR.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_SH3GL2_EUR\n",
      "  --extract range reduced_coding_nonsynonymous_EUR.txt\n",
      "  --out coding_nonsynonymous_pheno_SH3GL2_EUR\n",
      "  --recode A\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "17869 variants loaded from .bim file.\n",
      "15356 people (9460 males, 5896 females) loaded from .fam.\n",
      "14196 phenotype values loaded from .fam.\n",
      "--extract range: 17839 variants excluded.\n",
      "--extract range: 30 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 15356 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999954.\n",
      "30 variants and 15356 people pass filters and QC.\n",
      "Among remaining phenotypes, 9230 are cases and 4966 are controls.  (1160\n",
      "phenotypes are missing.)\n",
      "--recode A to coding_nonsynonymous_pheno_SH3GL2_EUR.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_SH3GL2_EUR --extract range reduced_coding_nonsynonymous_EUR.txt --recode A --out coding_nonsynonymous_pheno_SH3GL2_EUR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:17747115:A:C_C</th>\n",
       "      <th>chr9:17761500:C:G_G</th>\n",
       "      <th>chr9:17787498:T:A_A</th>\n",
       "      <th>chr9:17789425:G:T_T</th>\n",
       "      <th>...</th>\n",
       "      <th>chr9:17793471:C:T_T</th>\n",
       "      <th>chr9:17793477:C:T_T</th>\n",
       "      <th>chr9:17795552:A:G_G</th>\n",
       "      <th>chr9:17795571:G:A_A</th>\n",
       "      <th>chr9:17795582:G:A_A</th>\n",
       "      <th>chr9:17795624:G:A_A</th>\n",
       "      <th>chr9:17795649:A:G_G</th>\n",
       "      <th>chr9:17795682:A:G_G</th>\n",
       "      <th>chr9:17795685:G:A_A</th>\n",
       "      <th>chr9:17795705:A:G_G</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000001_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000002_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000003_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000004_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000005_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   FID             IID  PAT  MAT  SEX  PHENOTYPE  chr9:17747115:A:C_C  \\\n",
       "0    0  APGS_000001_s1    0    0    1          2                    0   \n",
       "1    0  APGS_000002_s1    0    0    1          2                    0   \n",
       "2    0  APGS_000003_s1    0    0    1          2                    0   \n",
       "3    0  APGS_000004_s1    0    0    2          2                    0   \n",
       "4    0  APGS_000005_s1    0    0    1          2                    0   \n",
       "\n",
       "   chr9:17761500:C:G_G  chr9:17787498:T:A_A  chr9:17789425:G:T_T  ...  \\\n",
       "0                    0                  0.0                    0  ...   \n",
       "1                    0                  0.0                    0  ...   \n",
       "2                    0                  0.0                    0  ...   \n",
       "3                    0                  0.0                    0  ...   \n",
       "4                    0                  0.0                    0  ...   \n",
       "\n",
       "   chr9:17793471:C:T_T  chr9:17793477:C:T_T  chr9:17795552:A:G_G  \\\n",
       "0                  0.0                    0                  0.0   \n",
       "1                  0.0                    0                  0.0   \n",
       "2                  0.0                    0                  0.0   \n",
       "3                  0.0                    0                  0.0   \n",
       "4                  0.0                    0                  0.0   \n",
       "\n",
       "   chr9:17795571:G:A_A  chr9:17795582:G:A_A  chr9:17795624:G:A_A  \\\n",
       "0                    0                    0                    0   \n",
       "1                    0                    0                    0   \n",
       "2                    0                    0                    0   \n",
       "3                    0                    0                    0   \n",
       "4                    0                    0                    0   \n",
       "\n",
       "   chr9:17795649:A:G_G  chr9:17795682:A:G_G  chr9:17795685:G:A_A  \\\n",
       "0                    0                    0                    0   \n",
       "1                    0                    0                    0   \n",
       "2                    0                    0                    0   \n",
       "3                    0                    0                    0   \n",
       "4                    0                    0                    0   \n",
       "\n",
       "   chr9:17795705:A:G_G  \n",
       "0                  0.0  \n",
       "1                  0.0  \n",
       "2                  0.0  \n",
       "3                  0.0  \n",
       "4                  0.0  \n",
       "\n",
       "[5 rows x 36 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_EUR_recode = pd.read_csv(f'{WORK_DIR}coding_nonsynonymous_pheno_SH3GL2_EUR.raw', delim_whitespace=True)\n",
    "SH3GL2_EUR_recode.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save out results..!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -mu terra-18d8e41c cp -r /home/jupyter/SH3GL2_GP2/ gs://fc-f7a400c1-827e-48f8-b7b6-90c488a000a4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AAC_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AJ_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_FIN_release5.log [Content-Type=application/octet-stream]...\n",
      "==> NOTE: You are uploading one or more large file(s), which would run          \n",
      "significantly faster if you enable parallel composite uploads. This\n",
      "feature can be enabled by editing the\n",
      "\"parallel_composite_upload_threshold\" value in your .boto\n",
      "configuration file. However, note that if you do this large files will\n",
      "be uploaded as `composite objects\n",
      "<https://cloud.google.com/storage/docs/composite-objects>`_,which\n",
      "means that any user who downloads such objects will need to have a\n",
      "compiled crcmod installed (see \"gsutil help crcmod\"). This is because\n",
      "without a compiled crcmod, computing checksums on composite objects is\n",
      "so slow that gsutil disables downloads of composite objects.\n",
      "\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_EUR.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release4.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_EUR_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_MDE_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_EUR_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_EUR.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_AJ.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AMR_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_CAS_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/temp.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_MDE_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_CAS_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_AAC_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/SEX.txt [Content-Type=text/plain]...    \n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.nosex [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EUR_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release4.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AMR_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_AAC.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_FIN.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EUR_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AJ_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_CAS_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_EUR_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/master_key_release4_final.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_AFR_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_FIN.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_EUR.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_EUR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_AFR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_CAS_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_AJ.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_AAC.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EUR_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AMR_release4.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AMR_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_AFR.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_AMR.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/PHENO.txt [Content-Type=text/plain]...  \n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EUR_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AMR_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AMR_release4.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release4.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release4.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_FIN_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AAC_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_MDE.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_EUR.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_MDE_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AAC_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AAC_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_AFR.nosex [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_FIN_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AJ_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/master_key_release5_final.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_FIN_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AJ_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.nosex [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_MDE_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AMR_release4.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_CAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_MDE.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_CAS_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_AMR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AMR_release4.pgen [Content-Type=application/octet-stream]...\n",
      "INFO 0703 18:49:39.115679 retry_util.py] Retrying request, attempt #1...9       \n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_CAS.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_EUR_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_MDE_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_AMR_rare_0.05.nosex [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_AMR.nosex [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_AFR_rare_0.05.nosex [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_AMR_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "- [156/156 files][ 10.7 GiB/ 10.7 GiB] 100% Done  58.8 MiB/s ETA 00:00:00       \n",
      "Operation completed over 156 objects/10.7 GiB.                                   \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR} {WORKSPACE_BUCKET}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {
    "height": "248.972px",
    "width": "263.062px"
   },
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "268.372px"
   },
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
