{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## __Single gene analysis__\n",
    "\n",
    "```GP2 ‚ù§Ô∏è Open Science üòç```\n",
    "\n",
    "### Description\n",
    "\n",
    "Using \"individual level data\"....:\n",
    "\n",
    "__0. Getting Started__\n",
    "- Loading Python libraries\n",
    "- Defining functions\n",
    "- Installing packages\n",
    "\n",
    "__1. Copy data from workspace to cloud environment__\n",
    "\n",
    "__2. Extract SH3GL2__\n",
    "\n",
    "__3. Annotate SH3GL2 variants__\n",
    "\n",
    "__4. Extract coding/non-syn variants__\n",
    "\n",
    "__5. Calculate frequency in cases versus controls__\n",
    "\n",
    "__6. Calculate frequency (homozygotes)in cases versus controls__\n",
    "\n",
    "__7. Save out results__\n",
    "\n"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "## Getting started"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Loading Python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "\n",
    "# Numpy for basics\n",
    "\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery\n",
    "\n",
    "#Import Sys\n",
    "import sys as sys"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Defining functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "https://notebooks.firecloud.org/proxy/terra-18d8e41c/saturn-c0307c3a-6cd0-4eca-9068-61afda6b299f/jupyter/notebooks/Endophilin-A/edit/SH3GL2_AMP-PD_ALF.ipynb#    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Billing and Workspace\n",
      "Workspace Name: Endophilin-A\n",
      "Billing Project: terra-18d8e41c\n",
      "Workspace Bucket, where you can upload and download data: gs://fc-f7a400c1-827e-48f8-b7b6-90c488a000a4\n",
      "\n",
      "GP2 v5.0\n",
      "Path to GP2 v5.0 Clinical Data: gs://gp2tier2/release5_11052023/clinical_data\n",
      "Path to GP2 v5.0 Raw Genotype Data: gs://gp2tier2/release5_11052023/raw_genotypes\n",
      "Path to GP2 v5.0 Imputed Genotype Data: gs://gp2tier2/release5_11052023/imputed_genotypes\n",
      "AMP-PD v3.0\n",
      "Path to AMP-PD v3.0 Clinical Data: gs://amp-pd-data/releases/2022_v3release_1115/clinical\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "\n",
    "## GP2 v5.0\n",
    "## Explicitly define release v5.0 path \n",
    "GP2_RELEASE_PATH = 'gs://gp2tier2/release5_11052023'\n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v5.0')\n",
    "print(f'Path to GP2 v5.0 Clinical Data: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v5.0 Raw Genotype Data: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v5.0 Imputed Genotype Data: {GP2_IMPUTED_GENO_PATH}')\n",
    "\n",
    "\n",
    "## AMP-PD v3.0\n",
    "## Explicitly define release v3.0 path \n",
    "AMP_RELEASE_PATH = 'gs://amp-pd-data/releases/2022_v3release_1115'\n",
    "AMP_CLINICAL_RELEASE_PATH = f'{AMP_RELEASE_PATH}/clinical'\n",
    "\n",
    "AMP_WGS_RELEASE_PATH = 'gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS'\n",
    "AMP_WGS_RELEASE_PLINK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'plink')\n",
    "\n",
    "print('AMP-PD v3.0')\n",
    "print(f'Path to AMP-PD v3.0 Clinical Data: {AMP_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PATH}')\n",
    "print('')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install Plink 1.9 and Plink 2.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n",
      "Plink2 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi\n",
    "\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/alpha3/plink2_linux_avx2_20220603.zip\n",
    "unzip -o plink2_linux_avx2_20220603.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remote restrictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 1.9 \n",
    "chmod u+x /home/jupyter/tools/plink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 2.0\n",
    "chmod u+x /home/jupyter/tools/plink2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install ANNOVAR: Download sources of annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGene.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneMrna.fa.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneVersion.txt.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotate_variation.pl\n",
      "coding_change.pl\n",
      "convert2annovar.pl\n",
      "example\n",
      "humandb\n",
      "retrieve_seq_from_fasta.pl\n",
      "table_annovar.pl\n",
      "variants_reduction.pl\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/annovar/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Copy data from AMP-PD bucket to workspace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Making a working directory\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: mkdir -p /home/jupyter/SH3GL2/\n"
     ]
    }
   ],
   "source": [
    "# Make a directory\n",
    "print(\"Making a working directory\")\n",
    "WORK_DIR = f'/home/jupyter/SH3GL2/'\n",
    "shell_do(f'mkdir -p {WORK_DIR}') # f' means f-string - contains expressions to execute the code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "code_folding": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-18d8e41c ls gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/all_chrs_merged.log\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/all_chrs_merged.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/all_chrs_merged.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/all_chrs_merged.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr1.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr1.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr1.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr10.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr10.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr10.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr11.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr11.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr11.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr12.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr12.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr12.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr13.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr13.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr13.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr14.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr14.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr14.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr15.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr15.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr15.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr16.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr16.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr16.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr17.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr17.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr17.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr18.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr18.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr18.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr19.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr19.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr19.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr2.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr2.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr2.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr20.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr20.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr20.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr21.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr21.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr21.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr22.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr22.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr22.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr3.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr3.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr3.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr4.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr4.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr4.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr5.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr5.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr5.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr6.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr6.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr6.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr7.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr7.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr7.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr8.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr8.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr8.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr9.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr9.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr9.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chrX.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chrX.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chrX.pvar\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chrY.pgen\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chrY.psam\r\n",
      "gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chrY.pvar\r\n"
     ]
    }
   ],
   "source": [
    "# Check directory where AMP-PD data is\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {AMP_WGS_RELEASE_PLINK_PATH}/pfiles')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-18d8e41c -m cp -r gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr9* /home/jupyter/SH3GL2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr9.pgen...\n",
      "Copying gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr9.psam...\n",
      "Copying gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles/chr9.pvar...\n",
      "\\ [3/3 files][  8.8 GiB/  8.8 GiB] 100% Done  52.2 MiB/s ETA 00:00:00           \n",
      "Operation completed over 3 objects/8.8 GiB.                                      \n"
     ]
    }
   ],
   "source": [
    "#Copy data\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {AMP_WGS_RELEASE_PLINK_PATH}/pfiles/chr9* {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create a covariate file with AMP-PD data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-18d8e41c -m cp -r gs://amp-pd-data/releases/2022_v3release_1115/amp_pd_case_control.csv /home/jupyter/SH3GL2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://amp-pd-data/releases/2022_v3release_1115/amp_pd_case_control.csv...\n",
      "/ [1/1 files][735.1 KiB/735.1 KiB] 100% Done                                    \n",
      "Operation completed over 1 objects/735.1 KiB.                                    \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-18d8e41c -m cp -r gs://amp-pd-data/releases/2022_v3release_1115/clinical/Demographics.csv /home/jupyter/SH3GL2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://amp-pd-data/releases/2022_v3release_1115/clinical/Demographics.csv...\n",
      "/ [1/1 files][622.2 KiB/622.2 KiB] 100% Done                                    \n",
      "Operation completed over 1 objects/622.2 KiB.                                    \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {AMP_RELEASE_PATH}/amp_pd_case_control.csv {WORK_DIR}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {AMP_CLINICAL_RELEASE_PATH}/Demographics.csv {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd_case_control_df = pd.read_csv(f'{WORK_DIR}/amp_pd_case_control.csv')\n",
    "#pd_case_control_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd_case_control_latest_df = pd_case_control_df[['participant_id', 'diagnosis_latest', 'case_control_other_latest']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd_case_control_latest_df.columns = ['ID', 'LATEST_DX', 'CASE_CONTROL']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "case_con_reduced = pd_case_control_latest_df.copy()\n",
    "case_con_reduced.drop_duplicates(subset=['ID'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "conditions = [\n",
    "    (case_con_reduced['CASE_CONTROL'] == \"Case\"),\n",
    "    (case_con_reduced['CASE_CONTROL'] == \"Control\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "choices = [2,1]\n",
    "case_con_reduced['PHENO'] = np.select(conditions, choices, default=-9).astype(np.int64)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "case_con_reduced.reset_index(inplace=True)\n",
    "case_con_reduced.drop(columns=[\"index\"], inplace=True)\n",
    "#case_con_reduced"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "demographics_df = pd.read_csv(f'{WORK_DIR}/Demographics.csv')\n",
    "#demographics_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "demographics_df.rename(columns = {'participant_id':'ID'}, inplace = True)\n",
    "demographics_df.rename(columns = {'age_at_baseline':'BASELINE_AGE'}, inplace = True)\n",
    "demographics_df.rename(columns = {'race':'RACE'}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "demographics_baseline_df = demographics_df \\\n",
    ".sort_values('visit_month', ascending=True) \\\n",
    ".drop_duplicates('ID').sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "demographics_df_casecon = demographics_df.merge(case_con_reduced, on='ID', how='outer')\n",
    "#demographics_df_casecon.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "conditions = [\n",
    "     (demographics_df_casecon['sex'] == \"Male\"),\n",
    "     (demographics_df_casecon['sex'] == \"Female\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "choices = [1,2]\n",
    "demographics_df_casecon['SEX'] = np.select(conditions, choices, default=None).astype(np.int64)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "demographics_df_casecon_toKeep_sex = demographics_df_casecon[['ID', 'ID', 'SEX'\n",
    "                                                          ]].copy()\n",
    "demographics_df_casecon_toKeep_sex.columns = ['FID','IID', 'SEX']\n",
    "demographics_df_casecon_toKeep_sex.to_csv(f'{WORK_DIR}/SEX.txt',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "demographics_df_casecon_toKeep_pheno = demographics_df_casecon[['ID', 'ID', 'PHENO'\n",
    "                                                          ]].copy()\n",
    "demographics_df_casecon_toKeep_pheno.columns = ['FID','IID', 'PHENO']\n",
    "demographics_df_casecon_toKeep_pheno.to_csv(f'{WORK_DIR}/PHENO.txt',index=False)\n",
    "demographics_df_casecon_toKeep_pheno.to_csv(f'{WORKSPACE_BUCKET}/PHENO_AMPPD.txt',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract SH3GL2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a3.3LM AVX2 Intel (3 Jun 2022)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to temp_AMPPD.log.\n",
      "Options in effect:\n",
      "  --chr 9\n",
      "  --from-bp 17579066\n",
      "  --make-bed\n",
      "  --max-alleles 2\n",
      "  --out temp_AMPPD\n",
      "  --pfile chr9\n",
      "  --to-bp 17797124\n",
      "\n",
      "Start time: Thu Nov 23 12:26:28 2023\n",
      "7450 MiB RAM detected; reserving 3725 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "10418 samples (0 females, 0 males, 10418 ambiguous; 10418 founders) loaded from\n",
      "chr9.psam.\n",
      "6246283 out of 6777486 variants loaded from chr9.pvar.\n",
      "Note: No phenotype data present.\n",
      "14592 variants remaining after main filters.\n",
      "Writing temp_AMPPD.fam ... done.\n",
      "Writing temp_AMPPD.bim ... done.\n",
      "Writing temp_AMPPD.bed ... 0%\b\b15%\b\b\bdone.\n",
      "End time: Thu Nov 23 12:27:47 2023\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# SH3GL2: gene positions on hg38 (from https://useast.ensembl.org/index.html)\n",
    "/home/jupyter/tools/plink2 \\\n",
    "--pfile chr9 \\\n",
    "--chr 9 \\\n",
    "--from-bp 17579066 \\\n",
    "--to-bp 17797124 \\\n",
    "--make-bed \\\n",
    "--max-alleles 2 \\\n",
    "--out temp_AMPPD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\t.\t0\t17579072\tA\tG\n",
      "9\t.\t0\t17579076\tA\tT\n",
      "9\t.\t0\t17579077\tG\tT\n",
      "9\t.\t0\t17579086\tA\tG\n",
      "9\t.\t0\t17579091\tG\tC\n",
      "9\t.\t0\t17579098\tT\tC\n",
      "9\t.\t0\t17579102\tG\tA\n",
      "9\t.\t0\t17579107\tT\tC\n",
      "9\t.\t0\t17579112\tT\tC\n",
      "9\t.\t0\t17579119\tC\tG\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head temp_AMPPD.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "bim = pd.read_csv(\"/home/jupyter/SH3GL2/temp_AMPPD.bim\",sep=\"\\t\",names=['chr','rsID','pos','bp','A1','A2'])\n",
    "bim['chr'] = bim['chr'].astype(str)\n",
    "bim['bp'] = bim['bp'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "bim['rsID'] = bim.chr.str.cat(bim.bp, sep=':')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "bim.to_csv(f'{WORK_DIR}/temp_AMPPD.bim',sep=\"\\t\",index=False,header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HB-PD_INVAB109VHC\tHB-PD_INVAB109VHC\t0\t0\t0\t-9\n",
      "HB-PD_INVAB289LG3\tHB-PD_INVAB289LG3\t0\t0\t0\t-9\n",
      "HB-PD_INVAC488AAF\tHB-PD_INVAC488AAF\t0\t0\t0\t-9\n",
      "HB-PD_INVAE296YP8\tHB-PD_INVAE296YP8\t0\t0\t0\t-9\n",
      "HB-PD_INVAJ549VWD\tHB-PD_INVAJ549VWD\t0\t0\t0\t-9\n",
      "HB-PD_INVAK106DV5\tHB-PD_INVAK106DV5\t0\t0\t0\t-9\n",
      "HB-PD_INVAK293NGR\tHB-PD_INVAK293NGR\t0\t0\t0\t-9\n",
      "HB-PD_INVAM835KNR\tHB-PD_INVAM835KNR\t0\t0\t0\t-9\n",
      "HB-PD_INVAM864YC9\tHB-PD_INVAM864YC9\t0\t0\t0\t-9\n",
      "HB-PD_INVAN082DKK\tHB-PD_INVAN082DKK\t0\t0\t0\t-9\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head temp_AMPPD.fam\n",
    "#Store the first column of IDs in a new file\n",
    "awk '{ print $1 }' temp_AMPPD.fam > IDs.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "sex = pd.read_csv(f'{WORK_DIR}/SEX.txt')\n",
    "pheno = pd.read_csv(f'{WORK_DIR}/PHENO.txt')\n",
    "ids = pd.read_csv(f'{WORK_DIR}/IDs.txt',names=['FID'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "sex_new = pd.merge(sex,ids,on=\"FID\")\n",
    "sex_new.to_csv(f'{WORK_DIR}/SEX_new.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno_new = pd.merge(pheno,ids,on=\"FID\")\n",
    "pheno_new.to_csv(f'{WORK_DIR}/PHENO_new.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_SH3GL2.log.\n",
      "Options in effect:\n",
      "  --bfile temp_AMPPD\n",
      "  --make-bed\n",
      "  --out pheno_SH3GL2\n",
      "  --pheno PHENO_new.txt\n",
      "  --update-sex SEX_new.txt\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "14592 variants loaded from .bim file.\n",
      "10418 people (0 males, 0 females, 10418 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to pheno_SH3GL2.nosex .\n",
      "7512 phenotype values present after --pheno.\n",
      "--update-sex: 10418 people updated, 1 ID not present.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 10418 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999769.\n",
      "14592 variants and 10418 people pass filters and QC.\n",
      "Among remaining phenotypes, 3359 are cases and 4153 are controls.  (2906\n",
      "phenotypes are missing.)\n",
      "--make-bed to pheno_SH3GL2.bed + pheno_SH3GL2.bim + pheno_SH3GL2.fam ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# Update sex variable\n",
    "/home/jupyter/tools/plink \\\n",
    "--bfile temp_AMPPD \\\n",
    "--update-sex SEX_new.txt  \\\n",
    "--pheno PHENO_new.txt \\\n",
    "--make-bed \\\n",
    "--out pheno_SH3GL2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualize plink files bim, fam and bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\t9:17579072\t0\t17579072\tA\tG\n",
      "9\t9:17579076\t0\t17579076\tA\tT\n",
      "9\t9:17579077\t0\t17579077\tG\tT\n",
      "9\t9:17579086\t0\t17579086\tA\tG\n",
      "9\t9:17579091\t0\t17579091\tG\tC\n",
      "9\t9:17579098\t0\t17579098\tT\tC\n",
      "9\t9:17579102\t0\t17579102\tG\tA\n",
      "9\t9:17579107\t0\t17579107\tT\tC\n",
      "9\t9:17579112\t0\t17579112\tT\tC\n",
      "9\t9:17579119\t0\t17579119\tC\tG\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize bim file\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_SH3GL2.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HB-PD_INVAB109VHC HB-PD_INVAB109VHC 0 0 2 1\n",
      "HB-PD_INVAB289LG3 HB-PD_INVAB289LG3 0 0 2 1\n",
      "HB-PD_INVAC488AAF HB-PD_INVAC488AAF 0 0 1 1\n",
      "HB-PD_INVAE296YP8 HB-PD_INVAE296YP8 0 0 2 1\n",
      "HB-PD_INVAJ549VWD HB-PD_INVAJ549VWD 0 0 2 1\n",
      "HB-PD_INVAK106DV5 HB-PD_INVAK106DV5 0 0 1 1\n",
      "HB-PD_INVAK293NGR HB-PD_INVAK293NGR 0 0 1 1\n",
      "HB-PD_INVAM835KNR HB-PD_INVAM835KNR 0 0 2 1\n",
      "HB-PD_INVAM864YC9 HB-PD_INVAM864YC9 0 0 2 1\n",
      "HB-PD_INVAN082DKK HB-PD_INVAN082DKK 0 0 2 1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize fam file\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_SH3GL2.fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_SH3GL2.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile pheno_SH3GL2\n",
      "  --out pheno_SH3GL2\n",
      "  --recode vcf-fid\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "14592 variants loaded from .bim file.\n",
      "10418 people (5782 males, 4636 females) loaded from .fam.\n",
      "7512 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 10418 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999769.\n",
      "14592 variants and 10418 people pass filters and QC.\n",
      "Among remaining phenotypes, 3359 are cases and 4153 are controls.  (2906\n",
      "phenotypes are missing.)\n",
      "--recode vcf-fid to pheno_SH3GL2.vcf ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# Turn binary files into VCF\n",
    "/home/jupyter/tools/plink \\\n",
    "--bfile pheno_SH3GL2 \\\n",
    "--recode vcf-fid \\\n",
    "--allow-no-sex \\\n",
    "--out pheno_SH3GL2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate SH3GL2 variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bgzip] can't create pheno_SH3GL2.vcf.gz: File exists\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "export PATH=$PATH:/home/jupyter/tools/rvtests/third/tabix-0.2.6/\n",
    "\n",
    "### Bgzip and Tabix\n",
    "bgzip pheno_SH3GL2.vcf\n",
    "\n",
    "tabix -f -p vcf pheno_SH3GL2.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 pheno_SH3GL2.vcf.gz > pheno_SH3GL2.annovar.avinput>\n",
      "NOTICE: Finished reading 14599 lines from VCF file\n",
      "NOTICE: A total of 14592 locus in VCF file passed QC threshold, representing 13709 SNPs (8549 transitions and 5160 transversions) and 883 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 142820362 SNP genotypes (89063482 transitions and 53756880 transversions) and 9199094 indels/substitutions for 10418 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl pheno_SH3GL2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile pheno_SH3GL2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile pheno_SH3GL2.annovar.refGene -exonsort -nofirstcodondel pheno_SH3GL2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to pheno_SH3GL2.annovar.refGene.variant_function, pheno_SH3GL2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 14592 unique variants in 14592 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile pheno_SH3GL2.annovar pheno_SH3GL2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to pheno_SH3GL2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to pheno_SH3GL2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 14592 unique variants in 14592 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 1\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to pheno_SH3GL2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from pheno_SH3GL2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to pheno_SH3GL2.annovar.hg38_multianno.vcf\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "### Annotate variants using ANNOVAR: https://annovar.openbioinformatics.org/en/latest/ \n",
    "perl /home/jupyter/tools/annovar/table_annovar.pl pheno_SH3GL2.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out pheno_SH3GL2.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract coding and non-syn variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Visualize multianno file\n",
    "SH3GL2 = pd.read_csv(f'{WORK_DIR}/pheno_SH3GL2.annovar.hg38_multianno.txt', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "SH3GL2['Chr_bp'] = SH3GL2['Chr'].astype(str) + ':' + SH3GL2['Start'].astype(str) + ':' + SH3GL2['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo10422</th>\n",
       "      <th>Otherinfo10423</th>\n",
       "      <th>Otherinfo10424</th>\n",
       "      <th>Otherinfo10425</th>\n",
       "      <th>Otherinfo10426</th>\n",
       "      <th>Otherinfo10427</th>\n",
       "      <th>Otherinfo10428</th>\n",
       "      <th>Otherinfo10429</th>\n",
       "      <th>Otherinfo10430</th>\n",
       "      <th>Chr_bp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>17579072</td>\n",
       "      <td>17579072</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.-171G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17579072:17579072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>17579076</td>\n",
       "      <td>17579076</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.-167T&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17579076:17579076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>17579077</td>\n",
       "      <td>17579077</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.-166T&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17579077:17579077</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>17579086</td>\n",
       "      <td>17579086</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.-157G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17579086:17579086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>17579091</td>\n",
       "      <td>17579091</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.-152C&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17579091:17579091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14587</th>\n",
       "      <td>9</td>\n",
       "      <td>17797040</td>\n",
       "      <td>17797040</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.*1297G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17797040:17797040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14588</th>\n",
       "      <td>9</td>\n",
       "      <td>17797052</td>\n",
       "      <td>17797052</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.*1309T&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17797052:17797052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14589</th>\n",
       "      <td>9</td>\n",
       "      <td>17797079</td>\n",
       "      <td>17797079</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.*1336C&gt;T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17797079:17797079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14590</th>\n",
       "      <td>9</td>\n",
       "      <td>17797093</td>\n",
       "      <td>17797093</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.*1350G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17797093:17797093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14591</th>\n",
       "      <td>9</td>\n",
       "      <td>17797120</td>\n",
       "      <td>17797120</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.*1377C&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17797120:17797120</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14592 rows √ó 10442 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chr     Start       End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0        9  17579072  17579072   G   A         UTR5       SH3GL2   \n",
       "1        9  17579076  17579076   T   A         UTR5       SH3GL2   \n",
       "2        9  17579077  17579077   T   G         UTR5       SH3GL2   \n",
       "3        9  17579086  17579086   G   A         UTR5       SH3GL2   \n",
       "4        9  17579091  17579091   C   G         UTR5       SH3GL2   \n",
       "...    ...       ...       ...  ..  ..          ...          ...   \n",
       "14587    9  17797040  17797040   G   A         UTR3       SH3GL2   \n",
       "14588    9  17797052  17797052   T   A         UTR3       SH3GL2   \n",
       "14589    9  17797079  17797079   C   T         UTR3       SH3GL2   \n",
       "14590    9  17797093  17797093   G   A         UTR3       SH3GL2   \n",
       "14591    9  17797120  17797120   C   G         UTR3       SH3GL2   \n",
       "\n",
       "         GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  ...  \\\n",
       "0       NM_003026:c.-171G>A                  .                .  ...   \n",
       "1       NM_003026:c.-167T>A                  .                .  ...   \n",
       "2       NM_003026:c.-166T>G                  .                .  ...   \n",
       "3       NM_003026:c.-157G>A                  .                .  ...   \n",
       "4       NM_003026:c.-152C>G                  .                .  ...   \n",
       "...                     ...                ...              ...  ...   \n",
       "14587  NM_003026:c.*1297G>A                  .                .  ...   \n",
       "14588  NM_003026:c.*1309T>A                  .                .  ...   \n",
       "14589  NM_003026:c.*1336C>T                  .                .  ...   \n",
       "14590  NM_003026:c.*1350G>A                  .                .  ...   \n",
       "14591  NM_003026:c.*1377C>G                  .                .  ...   \n",
       "\n",
       "      Otherinfo10422  Otherinfo10423 Otherinfo10424 Otherinfo10425  \\\n",
       "0                0/0             0/0            0/0            0/0   \n",
       "1                0/0             0/0            0/0            0/0   \n",
       "2                0/0             0/0            0/0            0/0   \n",
       "3                0/0             0/0            0/0            0/0   \n",
       "4                0/0             0/0            0/0            0/0   \n",
       "...              ...             ...            ...            ...   \n",
       "14587            0/0             0/0            0/0            0/0   \n",
       "14588            0/0             0/0            0/0            0/0   \n",
       "14589            0/0             0/0            0/0            0/0   \n",
       "14590            0/0             0/0            0/0            0/0   \n",
       "14591            0/0             0/0            0/0            0/0   \n",
       "\n",
       "       Otherinfo10426  Otherinfo10427 Otherinfo10428 Otherinfo10429  \\\n",
       "0                 0/0             0/0            0/0            0/0   \n",
       "1                 0/0             0/0            0/0            0/0   \n",
       "2                 0/0             0/0            0/0            0/0   \n",
       "3                 0/0             0/0            0/0            0/0   \n",
       "4                 0/0             0/0            0/0            0/0   \n",
       "...               ...             ...            ...            ...   \n",
       "14587             0/0             0/0            0/0            0/0   \n",
       "14588             0/0             0/0            0/0            0/0   \n",
       "14589             0/0             0/0            0/0            0/0   \n",
       "14590             0/0             0/0            0/0            0/0   \n",
       "14591             0/0             0/0            0/0            0/0   \n",
       "\n",
       "      Otherinfo10430               Chr_bp  \n",
       "0                0/0  9:17579072:17579072  \n",
       "1                0/0  9:17579076:17579076  \n",
       "2                0/0  9:17579077:17579077  \n",
       "3                0/0  9:17579086:17579086  \n",
       "4                0/0  9:17579091:17579091  \n",
       "...              ...                  ...  \n",
       "14587            0/0  9:17797040:17797040  \n",
       "14588            0/0  9:17797052:17797052  \n",
       "14589            0/0  9:17797079:17797079  \n",
       "14590            0/0  9:17797093:17797093  \n",
       "14591            0/0  9:17797120:17797120  \n",
       "\n",
       "[14592 rows x 10442 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "SH3GL2 = SH3GL2.drop_duplicates(subset=['Chr_bp'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo10422</th>\n",
       "      <th>Otherinfo10423</th>\n",
       "      <th>Otherinfo10424</th>\n",
       "      <th>Otherinfo10425</th>\n",
       "      <th>Otherinfo10426</th>\n",
       "      <th>Otherinfo10427</th>\n",
       "      <th>Otherinfo10428</th>\n",
       "      <th>Otherinfo10429</th>\n",
       "      <th>Otherinfo10430</th>\n",
       "      <th>Chr_bp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>17579072</td>\n",
       "      <td>17579072</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.-171G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17579072:17579072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>17579076</td>\n",
       "      <td>17579076</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.-167T&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17579076:17579076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>17579077</td>\n",
       "      <td>17579077</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.-166T&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17579077:17579077</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>17579086</td>\n",
       "      <td>17579086</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.-157G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17579086:17579086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>17579091</td>\n",
       "      <td>17579091</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.-152C&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17579091:17579091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14587</th>\n",
       "      <td>9</td>\n",
       "      <td>17797040</td>\n",
       "      <td>17797040</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.*1297G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17797040:17797040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14588</th>\n",
       "      <td>9</td>\n",
       "      <td>17797052</td>\n",
       "      <td>17797052</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.*1309T&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17797052:17797052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14589</th>\n",
       "      <td>9</td>\n",
       "      <td>17797079</td>\n",
       "      <td>17797079</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.*1336C&gt;T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17797079:17797079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14590</th>\n",
       "      <td>9</td>\n",
       "      <td>17797093</td>\n",
       "      <td>17797093</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.*1350G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17797093:17797093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14591</th>\n",
       "      <td>9</td>\n",
       "      <td>17797120</td>\n",
       "      <td>17797120</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>NM_003026:c.*1377C&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17797120:17797120</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14590 rows √ó 10442 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chr     Start       End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0        9  17579072  17579072   G   A         UTR5       SH3GL2   \n",
       "1        9  17579076  17579076   T   A         UTR5       SH3GL2   \n",
       "2        9  17579077  17579077   T   G         UTR5       SH3GL2   \n",
       "3        9  17579086  17579086   G   A         UTR5       SH3GL2   \n",
       "4        9  17579091  17579091   C   G         UTR5       SH3GL2   \n",
       "...    ...       ...       ...  ..  ..          ...          ...   \n",
       "14587    9  17797040  17797040   G   A         UTR3       SH3GL2   \n",
       "14588    9  17797052  17797052   T   A         UTR3       SH3GL2   \n",
       "14589    9  17797079  17797079   C   T         UTR3       SH3GL2   \n",
       "14590    9  17797093  17797093   G   A         UTR3       SH3GL2   \n",
       "14591    9  17797120  17797120   C   G         UTR3       SH3GL2   \n",
       "\n",
       "         GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  ...  \\\n",
       "0       NM_003026:c.-171G>A                  .                .  ...   \n",
       "1       NM_003026:c.-167T>A                  .                .  ...   \n",
       "2       NM_003026:c.-166T>G                  .                .  ...   \n",
       "3       NM_003026:c.-157G>A                  .                .  ...   \n",
       "4       NM_003026:c.-152C>G                  .                .  ...   \n",
       "...                     ...                ...              ...  ...   \n",
       "14587  NM_003026:c.*1297G>A                  .                .  ...   \n",
       "14588  NM_003026:c.*1309T>A                  .                .  ...   \n",
       "14589  NM_003026:c.*1336C>T                  .                .  ...   \n",
       "14590  NM_003026:c.*1350G>A                  .                .  ...   \n",
       "14591  NM_003026:c.*1377C>G                  .                .  ...   \n",
       "\n",
       "      Otherinfo10422  Otherinfo10423 Otherinfo10424 Otherinfo10425  \\\n",
       "0                0/0             0/0            0/0            0/0   \n",
       "1                0/0             0/0            0/0            0/0   \n",
       "2                0/0             0/0            0/0            0/0   \n",
       "3                0/0             0/0            0/0            0/0   \n",
       "4                0/0             0/0            0/0            0/0   \n",
       "...              ...             ...            ...            ...   \n",
       "14587            0/0             0/0            0/0            0/0   \n",
       "14588            0/0             0/0            0/0            0/0   \n",
       "14589            0/0             0/0            0/0            0/0   \n",
       "14590            0/0             0/0            0/0            0/0   \n",
       "14591            0/0             0/0            0/0            0/0   \n",
       "\n",
       "       Otherinfo10426  Otherinfo10427 Otherinfo10428 Otherinfo10429  \\\n",
       "0                 0/0             0/0            0/0            0/0   \n",
       "1                 0/0             0/0            0/0            0/0   \n",
       "2                 0/0             0/0            0/0            0/0   \n",
       "3                 0/0             0/0            0/0            0/0   \n",
       "4                 0/0             0/0            0/0            0/0   \n",
       "...               ...             ...            ...            ...   \n",
       "14587             0/0             0/0            0/0            0/0   \n",
       "14588             0/0             0/0            0/0            0/0   \n",
       "14589             0/0             0/0            0/0            0/0   \n",
       "14590             0/0             0/0            0/0            0/0   \n",
       "14591             0/0             0/0            0/0            0/0   \n",
       "\n",
       "      Otherinfo10430               Chr_bp  \n",
       "0                0/0  9:17579072:17579072  \n",
       "1                0/0  9:17579076:17579076  \n",
       "2                0/0  9:17579077:17579077  \n",
       "3                0/0  9:17579086:17579086  \n",
       "4                0/0  9:17579091:17579091  \n",
       "...              ...                  ...  \n",
       "14587            0/0  9:17797040:17797040  \n",
       "14588            0/0  9:17797052:17797052  \n",
       "14589            0/0  9:17797079:17797079  \n",
       "14590            0/0  9:17797093:17797093  \n",
       "14591            0/0  9:17797120:17797120  \n",
       "\n",
       "[14590 rows x 10442 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr               14590\n",
       "Start             14590\n",
       "End               14590\n",
       "Ref               14590\n",
       "Alt               14590\n",
       "                  ...  \n",
       "Otherinfo10427    14590\n",
       "Otherinfo10428    14590\n",
       "Otherinfo10429    14590\n",
       "Otherinfo10430    14590\n",
       "Chr_bp            14590\n",
       "Length: 10442, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter all variants\n",
    "all_sh3gl2 = SH3GL2[SH3GL2['Gene.refGene'] == 'SH3GL2']\n",
    "all_sh3gl2.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "intronic    14425\n",
       "UTR3           81\n",
       "exonic         47\n",
       "UTR5           37\n",
       "Name: Func.refGene, dtype: int64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_sh3gl2[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr               47\n",
       "Start             47\n",
       "End               47\n",
       "Ref               47\n",
       "Alt               47\n",
       "                  ..\n",
       "Otherinfo10427    47\n",
       "Otherinfo10428    47\n",
       "Otherinfo10429    47\n",
       "Otherinfo10430    47\n",
       "Chr_bp            47\n",
       "Length: 10442, dtype: int64"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic variants\n",
    "coding = SH3GL2[SH3GL2['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "synonymous SNV       23\n",
       "nonsynonymous SNV    22\n",
       "stopgain              2\n",
       "Name: ExonicFunc.refGene, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter exonic and stopgain\n",
    "coding_stopgain = SH3GL2[(SH3GL2['Func.refGene'] == 'exonic') & (SH3GL2['ExonicFunc.refGene'] == 'stopgain')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter exonic and synonymous\n",
    "coding_synonymous = SH3GL2[(SH3GL2['Func.refGene'] == 'exonic') & (SH3GL2['ExonicFunc.refGene'] == 'synonymous SNV')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter exonic and non-syn \n",
    "coding_nonsynonymous = SH3GL2[(SH3GL2['Func.refGene'] == 'exonic') & (SH3GL2['ExonicFunc.refGene'] == 'nonsynonymous SNV')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo10422</th>\n",
       "      <th>Otherinfo10423</th>\n",
       "      <th>Otherinfo10424</th>\n",
       "      <th>Otherinfo10425</th>\n",
       "      <th>Otherinfo10426</th>\n",
       "      <th>Otherinfo10427</th>\n",
       "      <th>Otherinfo10428</th>\n",
       "      <th>Otherinfo10429</th>\n",
       "      <th>Otherinfo10430</th>\n",
       "      <th>Chr_bp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11378</th>\n",
       "      <td>9</td>\n",
       "      <td>17747120</td>\n",
       "      <td>17747120</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon2:c.A100C:p.K34Q</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17747120:17747120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11379</th>\n",
       "      <td>9</td>\n",
       "      <td>17747126</td>\n",
       "      <td>17747126</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon2:c.A106T:p.M36L</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17747126:17747126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13886</th>\n",
       "      <td>9</td>\n",
       "      <td>17786478</td>\n",
       "      <td>17786478</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon4:c.G285C:p.E95D</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17786478:17786478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13887</th>\n",
       "      <td>9</td>\n",
       "      <td>17786494</td>\n",
       "      <td>17786494</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon4:c.G301A:p.G101R</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17786494:17786494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13964</th>\n",
       "      <td>9</td>\n",
       "      <td>17787437</td>\n",
       "      <td>17787437</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon5:c.C389T:p.S130F</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17787437:17787437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13966</th>\n",
       "      <td>9</td>\n",
       "      <td>17787457</td>\n",
       "      <td>17787457</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon5:c.C409G:p.Q137E</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17787457:17787457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13967</th>\n",
       "      <td>9</td>\n",
       "      <td>17787498</td>\n",
       "      <td>17787498</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon5:c.T450A:p.D150E</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17787498:17787498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14101</th>\n",
       "      <td>9</td>\n",
       "      <td>17789477</td>\n",
       "      <td>17789477</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.G551A:p.R184H</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789477:17789477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14231</th>\n",
       "      <td>9</td>\n",
       "      <td>17791290</td>\n",
       "      <td>17791290</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.G684C:p.Q228H</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791290:17791290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14233</th>\n",
       "      <td>9</td>\n",
       "      <td>17791300</td>\n",
       "      <td>17791300</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.A694G:p.I232V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791300:17791300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14234</th>\n",
       "      <td>9</td>\n",
       "      <td>17791316</td>\n",
       "      <td>17791316</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.C710T:p.T237M</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791316:17791316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14366</th>\n",
       "      <td>9</td>\n",
       "      <td>17793419</td>\n",
       "      <td>17793419</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.A781G:p.M261V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793419:17793419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14367</th>\n",
       "      <td>9</td>\n",
       "      <td>17793459</td>\n",
       "      <td>17793459</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.A821G:p.N274S</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793459:17793459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14368</th>\n",
       "      <td>9</td>\n",
       "      <td>17793465</td>\n",
       "      <td>17793465</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.G827T:p.G276V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793465:17793465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14369</th>\n",
       "      <td>9</td>\n",
       "      <td>17793468</td>\n",
       "      <td>17793468</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.T830C:p.L277P</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793468:17793468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14370</th>\n",
       "      <td>9</td>\n",
       "      <td>17793471</td>\n",
       "      <td>17793471</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.C833T:p.S278F</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793471:17793471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14500</th>\n",
       "      <td>9</td>\n",
       "      <td>17795552</td>\n",
       "      <td>17795552</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.A868G:p.M290V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795552:17795552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14501</th>\n",
       "      <td>9</td>\n",
       "      <td>17795571</td>\n",
       "      <td>17795571</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.G887A:p.R296Q</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795571:17795571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14503</th>\n",
       "      <td>9</td>\n",
       "      <td>17795582</td>\n",
       "      <td>17795582</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.G898A:p.D300N</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795582:17795582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14504</th>\n",
       "      <td>9</td>\n",
       "      <td>17795598</td>\n",
       "      <td>17795598</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.A914G:p.N305S</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795598:17795598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14507</th>\n",
       "      <td>9</td>\n",
       "      <td>17795649</td>\n",
       "      <td>17795649</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.A965G:p.Q322R</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795649:17795649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14509</th>\n",
       "      <td>9</td>\n",
       "      <td>17795705</td>\n",
       "      <td>17795705</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.A1021G:p.I341V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795705:17795705</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22 rows √ó 10442 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chr     Start       End Ref Alt Func.refGene Gene.refGene  \\\n",
       "11378    9  17747120  17747120   A   C       exonic       SH3GL2   \n",
       "11379    9  17747126  17747126   A   T       exonic       SH3GL2   \n",
       "13886    9  17786478  17786478   G   C       exonic       SH3GL2   \n",
       "13887    9  17786494  17786494   G   A       exonic       SH3GL2   \n",
       "13964    9  17787437  17787437   C   T       exonic       SH3GL2   \n",
       "13966    9  17787457  17787457   C   G       exonic       SH3GL2   \n",
       "13967    9  17787498  17787498   T   A       exonic       SH3GL2   \n",
       "14101    9  17789477  17789477   G   A       exonic       SH3GL2   \n",
       "14231    9  17791290  17791290   G   C       exonic       SH3GL2   \n",
       "14233    9  17791300  17791300   A   G       exonic       SH3GL2   \n",
       "14234    9  17791316  17791316   C   T       exonic       SH3GL2   \n",
       "14366    9  17793419  17793419   A   G       exonic       SH3GL2   \n",
       "14367    9  17793459  17793459   A   G       exonic       SH3GL2   \n",
       "14368    9  17793465  17793465   G   T       exonic       SH3GL2   \n",
       "14369    9  17793468  17793468   T   C       exonic       SH3GL2   \n",
       "14370    9  17793471  17793471   C   T       exonic       SH3GL2   \n",
       "14500    9  17795552  17795552   A   G       exonic       SH3GL2   \n",
       "14501    9  17795571  17795571   G   A       exonic       SH3GL2   \n",
       "14503    9  17795582  17795582   G   A       exonic       SH3GL2   \n",
       "14504    9  17795598  17795598   A   G       exonic       SH3GL2   \n",
       "14507    9  17795649  17795649   A   G       exonic       SH3GL2   \n",
       "14509    9  17795705  17795705   A   G       exonic       SH3GL2   \n",
       "\n",
       "      GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "11378                  .  nonsynonymous SNV   \n",
       "11379                  .  nonsynonymous SNV   \n",
       "13886                  .  nonsynonymous SNV   \n",
       "13887                  .  nonsynonymous SNV   \n",
       "13964                  .  nonsynonymous SNV   \n",
       "13966                  .  nonsynonymous SNV   \n",
       "13967                  .  nonsynonymous SNV   \n",
       "14101                  .  nonsynonymous SNV   \n",
       "14231                  .  nonsynonymous SNV   \n",
       "14233                  .  nonsynonymous SNV   \n",
       "14234                  .  nonsynonymous SNV   \n",
       "14366                  .  nonsynonymous SNV   \n",
       "14367                  .  nonsynonymous SNV   \n",
       "14368                  .  nonsynonymous SNV   \n",
       "14369                  .  nonsynonymous SNV   \n",
       "14370                  .  nonsynonymous SNV   \n",
       "14500                  .  nonsynonymous SNV   \n",
       "14501                  .  nonsynonymous SNV   \n",
       "14503                  .  nonsynonymous SNV   \n",
       "14504                  .  nonsynonymous SNV   \n",
       "14507                  .  nonsynonymous SNV   \n",
       "14509                  .  nonsynonymous SNV   \n",
       "\n",
       "                              AAChange.refGene  ... Otherinfo10422  \\\n",
       "11378    SH3GL2:NM_003026:exon2:c.A100C:p.K34Q  ...            0/0   \n",
       "11379    SH3GL2:NM_003026:exon2:c.A106T:p.M36L  ...            0/0   \n",
       "13886    SH3GL2:NM_003026:exon4:c.G285C:p.E95D  ...            0/0   \n",
       "13887   SH3GL2:NM_003026:exon4:c.G301A:p.G101R  ...            0/0   \n",
       "13964   SH3GL2:NM_003026:exon5:c.C389T:p.S130F  ...            0/0   \n",
       "13966   SH3GL2:NM_003026:exon5:c.C409G:p.Q137E  ...            0/0   \n",
       "13967   SH3GL2:NM_003026:exon5:c.T450A:p.D150E  ...            0/0   \n",
       "14101   SH3GL2:NM_003026:exon6:c.G551A:p.R184H  ...            0/0   \n",
       "14231   SH3GL2:NM_003026:exon7:c.G684C:p.Q228H  ...            0/0   \n",
       "14233   SH3GL2:NM_003026:exon7:c.A694G:p.I232V  ...            0/0   \n",
       "14234   SH3GL2:NM_003026:exon7:c.C710T:p.T237M  ...            0/0   \n",
       "14366   SH3GL2:NM_003026:exon8:c.A781G:p.M261V  ...            0/0   \n",
       "14367   SH3GL2:NM_003026:exon8:c.A821G:p.N274S  ...            0/0   \n",
       "14368   SH3GL2:NM_003026:exon8:c.G827T:p.G276V  ...            0/0   \n",
       "14369   SH3GL2:NM_003026:exon8:c.T830C:p.L277P  ...            0/0   \n",
       "14370   SH3GL2:NM_003026:exon8:c.C833T:p.S278F  ...            0/0   \n",
       "14500   SH3GL2:NM_003026:exon9:c.A868G:p.M290V  ...            0/0   \n",
       "14501   SH3GL2:NM_003026:exon9:c.G887A:p.R296Q  ...            0/0   \n",
       "14503   SH3GL2:NM_003026:exon9:c.G898A:p.D300N  ...            0/0   \n",
       "14504   SH3GL2:NM_003026:exon9:c.A914G:p.N305S  ...            0/0   \n",
       "14507   SH3GL2:NM_003026:exon9:c.A965G:p.Q322R  ...            0/0   \n",
       "14509  SH3GL2:NM_003026:exon9:c.A1021G:p.I341V  ...            0/0   \n",
       "\n",
       "       Otherinfo10423 Otherinfo10424 Otherinfo10425  Otherinfo10426  \\\n",
       "11378             0/0            0/0            0/0             0/0   \n",
       "11379             0/0            0/0            0/0             0/0   \n",
       "13886             0/0            0/0            0/0             0/0   \n",
       "13887             0/0            0/0            0/0             0/0   \n",
       "13964             0/0            0/0            0/0             0/0   \n",
       "13966             0/0            0/0            0/0             0/0   \n",
       "13967             0/0            0/0            0/0             0/0   \n",
       "14101             0/0            0/0            0/0             0/0   \n",
       "14231             0/0            0/0            0/0             0/0   \n",
       "14233             0/0            0/0            0/0             0/0   \n",
       "14234             0/0            0/0            0/0             0/0   \n",
       "14366             0/0            0/0            0/0             0/0   \n",
       "14367             0/0            0/0            0/0             0/0   \n",
       "14368             0/0            0/0            0/0             0/0   \n",
       "14369             0/0            0/0            0/0             0/0   \n",
       "14370             0/0            0/0            0/0             0/0   \n",
       "14500             0/0            0/0            0/0             0/0   \n",
       "14501             0/0            0/0            0/0             0/0   \n",
       "14503             0/0            0/0            0/0             0/0   \n",
       "14504             0/0            0/0            0/0             0/0   \n",
       "14507             0/0            0/0            0/0             0/0   \n",
       "14509             0/0            0/0            0/0             0/0   \n",
       "\n",
       "       Otherinfo10427 Otherinfo10428 Otherinfo10429 Otherinfo10430  \\\n",
       "11378             0/0            0/0            0/0            0/0   \n",
       "11379             0/0            0/0            0/0            0/0   \n",
       "13886             0/0            0/0            0/0            0/0   \n",
       "13887             0/0            0/0            0/0            0/0   \n",
       "13964             0/0            0/0            0/0            0/0   \n",
       "13966             0/0            0/0            0/0            0/0   \n",
       "13967             0/0            0/0            0/0            0/0   \n",
       "14101             0/0            0/0            0/0            0/0   \n",
       "14231             0/0            0/0            0/0            0/0   \n",
       "14233             0/0            0/0            0/0            0/0   \n",
       "14234             0/0            0/0            0/0            0/0   \n",
       "14366             0/0            0/0            0/0            0/0   \n",
       "14367             0/0            0/0            0/0            0/0   \n",
       "14368             0/0            0/0            0/0            0/0   \n",
       "14369             0/0            0/0            0/0            0/0   \n",
       "14370             0/0            0/0            0/0            0/0   \n",
       "14500             0/0            0/0            0/0            0/0   \n",
       "14501             0/0            0/0            0/0            0/0   \n",
       "14503             0/0            0/0            0/0            0/0   \n",
       "14504             0/0            0/0            0/0            0/0   \n",
       "14507             0/0            0/0            0/0            0/0   \n",
       "14509             0/0            0/0            0/0            0/0   \n",
       "\n",
       "                    Chr_bp  \n",
       "11378  9:17747120:17747120  \n",
       "11379  9:17747126:17747126  \n",
       "13886  9:17786478:17786478  \n",
       "13887  9:17786494:17786494  \n",
       "13964  9:17787437:17787437  \n",
       "13966  9:17787457:17787457  \n",
       "13967  9:17787498:17787498  \n",
       "14101  9:17789477:17789477  \n",
       "14231  9:17791290:17791290  \n",
       "14233  9:17791300:17791300  \n",
       "14234  9:17791316:17791316  \n",
       "14366  9:17793419:17793419  \n",
       "14367  9:17793459:17793459  \n",
       "14368  9:17793465:17793465  \n",
       "14369  9:17793468:17793468  \n",
       "14370  9:17793471:17793471  \n",
       "14500  9:17795552:17795552  \n",
       "14501  9:17795571:17795571  \n",
       "14503  9:17795582:17795582  \n",
       "14504  9:17795598:17795598  \n",
       "14507  9:17795649:17795649  \n",
       "14509  9:17795705:17795705  \n",
       "\n",
       "[22 rows x 10442 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_nonsynonymous.to_csv(f'{WORK_DIR}/coding_nonsynonymous.txt', sep = '\\t', index = False)\n",
    "coding_nonsynonymous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter intronic variants\n",
    "intronic = SH3GL2[SH3GL2['Func.refGene'] == 'intronic']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter UTR3 variants\n",
    "UTR3 = SH3GL2[SH3GL2['Func.refGene'] == 'UTR3']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter UTR5 variants\n",
    "UTR5 = SH3GL2[SH3GL2['Func.refGene'] == 'UTR5']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculate freq of coding and non-syn vars in cases versus controls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "reduced_coding_nonsynonymous = coding_nonsynonymous[[\"Chr\", \"Start\", \"End\", \"Gene.refGene\"]]\n",
    "reduced_coding_nonsynonymous.to_csv(f'{WORK_DIR}/reduced_coding_nonsynonymous.txt', sep = '\\t', index = False, header= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\t17747120\t17747120\tSH3GL2\n",
      "9\t17747126\t17747126\tSH3GL2\n",
      "9\t17786478\t17786478\tSH3GL2\n",
      "9\t17786494\t17786494\tSH3GL2\n",
      "9\t17787437\t17787437\tSH3GL2\n",
      "9\t17787457\t17787457\tSH3GL2\n",
      "9\t17787498\t17787498\tSH3GL2\n",
      "9\t17789477\t17789477\tSH3GL2\n",
      "9\t17791290\t17791290\tSH3GL2\n",
      "9\t17791300\t17791300\tSH3GL2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head reduced_coding_nonsynonymous.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to assoc_pheno_SH3GL2.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile pheno_SH3GL2\n",
      "  --out assoc_pheno_SH3GL2\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "14592 variants loaded from .bim file.\n",
      "10418 people (5782 males, 4636 females) loaded from .fam.\n",
      "7512 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 10418 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999769.\n",
      "14592 variants and 10418 people pass filters and QC.\n",
      "Among remaining phenotypes, 3359 are cases and 4153 are controls.  (2906\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to assoc_pheno_SH3GL2.assoc ...\n",
      "0%\b\b7%\b\b14%\b\b\b21%\b\b\b28%\b\b\b35%\b\b\b42%\b\b\b49%\b\b\b56%\b\b\b63%\b\b\b70%\b\b\b77%\b\b\b84%\b\b\b91%\b\b\b98%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# calculate freq for ALL variants\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_SH3GL2 --assoc --out assoc_pheno_SH3GL2 --allow-no-sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17579072</td>\n",
       "      <td>17579072</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>G</td>\n",
       "      <td>0.8089</td>\n",
       "      <td>0.3685</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17579076</td>\n",
       "      <td>17579076</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>1.2360</td>\n",
       "      <td>0.2662</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17579077</td>\n",
       "      <td>17579077</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>T</td>\n",
       "      <td>0.8089</td>\n",
       "      <td>0.3685</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17579086</td>\n",
       "      <td>17579086</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000298</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>G</td>\n",
       "      <td>0.5849</td>\n",
       "      <td>0.4444</td>\n",
       "      <td>2.473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17579091</td>\n",
       "      <td>17579091</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>C</td>\n",
       "      <td>0.8089</td>\n",
       "      <td>0.3685</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14587</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17797040</td>\n",
       "      <td>17797040</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>G</td>\n",
       "      <td>0.8089</td>\n",
       "      <td>0.3685</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14588</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17797052</td>\n",
       "      <td>17797052</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14589</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17797079</td>\n",
       "      <td>17797079</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>1.2360</td>\n",
       "      <td>0.2662</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14590</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17797093</td>\n",
       "      <td>17797093</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.000482</td>\n",
       "      <td>G</td>\n",
       "      <td>1.2360</td>\n",
       "      <td>0.2663</td>\n",
       "      <td>0.309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14591</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17797120</td>\n",
       "      <td>17797120</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>C</td>\n",
       "      <td>0.8089</td>\n",
       "      <td>0.3685</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14592 rows √ó 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHR         SNP        BP A1       F_A       F_U A2   CHISQ       P  \\\n",
       "0        9  9:17579072  17579072  A  0.000000  0.000120  G  0.8089  0.3685   \n",
       "1        9  9:17579076  17579076  A  0.000149  0.000000  T  1.2360  0.2662   \n",
       "2        9  9:17579077  17579077  G  0.000000  0.000120  T  0.8089  0.3685   \n",
       "3        9  9:17579086  17579086  A  0.000298  0.000120  G  0.5849  0.4444   \n",
       "4        9  9:17579091  17579091  G  0.000000  0.000120  C  0.8089  0.3685   \n",
       "...    ...         ...       ... ..       ...       ... ..     ...     ...   \n",
       "14587    9  9:17797040  17797040  A  0.000000  0.000120  G  0.8089  0.3685   \n",
       "14588    9  9:17797052  17797052  A  0.000000  0.000000  T     NaN     NaN   \n",
       "14589    9  9:17797079  17797079  T  0.000149  0.000000  C  1.2360  0.2662   \n",
       "14590    9  9:17797093  17797093  A  0.000149  0.000482  G  1.2360  0.2663   \n",
       "14591    9  9:17797120  17797120  G  0.000000  0.000120  C  0.8089  0.3685   \n",
       "\n",
       "          OR  \n",
       "0      0.000  \n",
       "1        NaN  \n",
       "2      0.000  \n",
       "3      2.473  \n",
       "4      0.000  \n",
       "...      ...  \n",
       "14587  0.000  \n",
       "14588    NaN  \n",
       "14589    NaN  \n",
       "14590  0.309  \n",
       "14591  0.000  \n",
       "\n",
       "[14592 rows x 10 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# visualize freq file for ALL variants\n",
    "all_SH3GL2_freq = pd.read_csv(f'{WORK_DIR}/assoc_pheno_SH3GL2.assoc', delim_whitespace=True)\n",
    "all_SH3GL2_freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17579692</td>\n",
       "      <td>17579692</td>\n",
       "      <td>T</td>\n",
       "      <td>0.336400</td>\n",
       "      <td>0.370800</td>\n",
       "      <td>G</td>\n",
       "      <td>19.190</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.8602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17579765</td>\n",
       "      <td>17579765</td>\n",
       "      <td>G</td>\n",
       "      <td>0.293500</td>\n",
       "      <td>0.324500</td>\n",
       "      <td>C</td>\n",
       "      <td>16.590</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.8651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17580008</td>\n",
       "      <td>17580008</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000893</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>7.421</td>\n",
       "      <td>0.006446</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17580618</td>\n",
       "      <td>17580618</td>\n",
       "      <td>T</td>\n",
       "      <td>0.328400</td>\n",
       "      <td>0.360600</td>\n",
       "      <td>A</td>\n",
       "      <td>17.030</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.8670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17581041</td>\n",
       "      <td>17581041</td>\n",
       "      <td>A</td>\n",
       "      <td>0.326400</td>\n",
       "      <td>0.311200</td>\n",
       "      <td>G</td>\n",
       "      <td>3.965</td>\n",
       "      <td>0.046470</td>\n",
       "      <td>1.0730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14018</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17788305</td>\n",
       "      <td>17788305</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000893</td>\n",
       "      <td>0.002288</td>\n",
       "      <td>C</td>\n",
       "      <td>4.347</td>\n",
       "      <td>0.037080</td>\n",
       "      <td>0.3899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14040</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17788598</td>\n",
       "      <td>17788598</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000893</td>\n",
       "      <td>0.002288</td>\n",
       "      <td>C</td>\n",
       "      <td>4.347</td>\n",
       "      <td>0.037080</td>\n",
       "      <td>0.3899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14139</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17789907</td>\n",
       "      <td>17789907</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000893</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>G</td>\n",
       "      <td>4.762</td>\n",
       "      <td>0.029090</td>\n",
       "      <td>7.4240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14336</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17792820</td>\n",
       "      <td>17792820</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001191</td>\n",
       "      <td>0.000241</td>\n",
       "      <td>T</td>\n",
       "      <td>5.040</td>\n",
       "      <td>0.024770</td>\n",
       "      <td>4.9500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14568</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17796672</td>\n",
       "      <td>17796672</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.001324</td>\n",
       "      <td>C</td>\n",
       "      <td>6.430</td>\n",
       "      <td>0.011220</td>\n",
       "      <td>0.1123</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>294 rows √ó 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHR         SNP        BP A1       F_A       F_U A2   CHISQ         P  \\\n",
       "77       9  9:17579692  17579692  T  0.336400  0.370800  G  19.190  0.000012   \n",
       "87       9  9:17579765  17579765  G  0.293500  0.324500  C  16.590  0.000046   \n",
       "109      9  9:17580008  17580008  C  0.000893  0.000000  T   7.421  0.006446   \n",
       "148      9  9:17580618  17580618  T  0.328400  0.360600  A  17.030  0.000037   \n",
       "166      9  9:17581041  17581041  A  0.326400  0.311200  G   3.965  0.046470   \n",
       "...    ...         ...       ... ..       ...       ... ..     ...       ...   \n",
       "14018    9  9:17788305  17788305  T  0.000893  0.002288  C   4.347  0.037080   \n",
       "14040    9  9:17788598  17788598  G  0.000893  0.002288  C   4.347  0.037080   \n",
       "14139    9  9:17789907  17789907  A  0.000893  0.000120  G   4.762  0.029090   \n",
       "14336    9  9:17792820  17792820  C  0.001191  0.000241  T   5.040  0.024770   \n",
       "14568    9  9:17796672  17796672  A  0.000149  0.001324  C   6.430  0.011220   \n",
       "\n",
       "           OR  \n",
       "77     0.8602  \n",
       "87     0.8651  \n",
       "109       NaN  \n",
       "148    0.8670  \n",
       "166    1.0730  \n",
       "...       ...  \n",
       "14018  0.3899  \n",
       "14040  0.3899  \n",
       "14139  7.4240  \n",
       "14336  4.9500  \n",
       "14568  0.1123  \n",
       "\n",
       "[294 rows x 10 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# visualize freq file for ALL variants comparing HC vs PD with p < 0.05 \n",
    "all_SH3GL2_freq_p_0_05 = all_SH3GL2_freq[all_SH3GL2_freq['P']  <= 0.05]\n",
    "all_SH3GL2_freq_p_0_05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_SH3GL2.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile pheno_SH3GL2\n",
      "  --extract range reduced_coding_nonsynonymous.txt\n",
      "  --out coding_nonsynonymous_pheno_SH3GL2\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "14592 variants loaded from .bim file.\n",
      "10418 people (5782 males, 4636 females) loaded from .fam.\n",
      "7512 phenotype values loaded from .fam.\n",
      "--extract range: 14570 variants excluded.\n",
      "--extract range: 22 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 10418 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "22 variants and 10418 people pass filters and QC.\n",
      "Among remaining phenotypes, 3359 are cases and 4153 are controls.  (2906\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to coding_nonsynonymous_pheno_SH3GL2.assoc ...\n",
      "0%\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_SH3GL2 --extract range reduced_coding_nonsynonymous.txt --assoc --out coding_nonsynonymous_pheno_SH3GL2 --allow-no-sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17747120</td>\n",
       "      <td>17747120</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17747126</td>\n",
       "      <td>17747126</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>1.2360</td>\n",
       "      <td>0.2662</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17786478</td>\n",
       "      <td>17786478</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>G</td>\n",
       "      <td>0.8089</td>\n",
       "      <td>0.3685</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17786494</td>\n",
       "      <td>17786494</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>G</td>\n",
       "      <td>0.8089</td>\n",
       "      <td>0.3685</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17787437</td>\n",
       "      <td>17787437</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17787457</td>\n",
       "      <td>17787457</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>C</td>\n",
       "      <td>0.8089</td>\n",
       "      <td>0.3685</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17787498</td>\n",
       "      <td>17787498</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>T</td>\n",
       "      <td>0.8089</td>\n",
       "      <td>0.3685</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17789477</td>\n",
       "      <td>17789477</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17791290</td>\n",
       "      <td>17791290</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17791300</td>\n",
       "      <td>17791300</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>1.2360</td>\n",
       "      <td>0.2662</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17791316</td>\n",
       "      <td>17791316</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>C</td>\n",
       "      <td>0.8089</td>\n",
       "      <td>0.3685</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17793419</td>\n",
       "      <td>17793419</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>1.2360</td>\n",
       "      <td>0.2662</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17793459</td>\n",
       "      <td>17793459</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000241</td>\n",
       "      <td>A</td>\n",
       "      <td>1.6180</td>\n",
       "      <td>0.2034</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17793465</td>\n",
       "      <td>17793465</td>\n",
       "      <td>T</td>\n",
       "      <td>0.003275</td>\n",
       "      <td>0.002528</td>\n",
       "      <td>G</td>\n",
       "      <td>0.7252</td>\n",
       "      <td>0.3944</td>\n",
       "      <td>1.2960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17793468</td>\n",
       "      <td>17793468</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>T</td>\n",
       "      <td>0.8089</td>\n",
       "      <td>0.3685</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17793471</td>\n",
       "      <td>17793471</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.000241</td>\n",
       "      <td>C</td>\n",
       "      <td>0.1572</td>\n",
       "      <td>0.6917</td>\n",
       "      <td>0.6181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17795552</td>\n",
       "      <td>17795552</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>1.2360</td>\n",
       "      <td>0.2662</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17795571</td>\n",
       "      <td>17795571</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>1.2360</td>\n",
       "      <td>0.2662</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17795582</td>\n",
       "      <td>17795582</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17795598</td>\n",
       "      <td>17795598</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>1.2360</td>\n",
       "      <td>0.2662</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17795649</td>\n",
       "      <td>17795649</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>1.2360</td>\n",
       "      <td>0.2662</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>9</td>\n",
       "      <td>9:17795705</td>\n",
       "      <td>17795705</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>1.2360</td>\n",
       "      <td>0.2662</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHR         SNP        BP A1       F_A       F_U A2   CHISQ       P  \\\n",
       "0     9  9:17747120  17747120  C  0.000000  0.000000  A     NaN     NaN   \n",
       "1     9  9:17747126  17747126  T  0.000149  0.000000  A  1.2360  0.2662   \n",
       "2     9  9:17786478  17786478  C  0.000000  0.000120  G  0.8089  0.3685   \n",
       "3     9  9:17786494  17786494  A  0.000000  0.000120  G  0.8089  0.3685   \n",
       "4     9  9:17787437  17787437  T  0.000000  0.000000  C     NaN     NaN   \n",
       "5     9  9:17787457  17787457  G  0.000000  0.000120  C  0.8089  0.3685   \n",
       "6     9  9:17787498  17787498  A  0.000000  0.000120  T  0.8089  0.3685   \n",
       "7     9  9:17789477  17789477  A  0.000000  0.000000  G     NaN     NaN   \n",
       "8     9  9:17791290  17791290  C  0.000000  0.000000  G     NaN     NaN   \n",
       "9     9  9:17791300  17791300  G  0.000149  0.000000  A  1.2360  0.2662   \n",
       "10    9  9:17791316  17791316  T  0.000000  0.000120  C  0.8089  0.3685   \n",
       "11    9  9:17793419  17793419  G  0.000149  0.000000  A  1.2360  0.2662   \n",
       "12    9  9:17793459  17793459  G  0.000000  0.000241  A  1.6180  0.2034   \n",
       "13    9  9:17793465  17793465  T  0.003275  0.002528  G  0.7252  0.3944   \n",
       "14    9  9:17793468  17793468  C  0.000000  0.000120  T  0.8089  0.3685   \n",
       "15    9  9:17793471  17793471  T  0.000149  0.000241  C  0.1572  0.6917   \n",
       "16    9  9:17795552  17795552  G  0.000149  0.000000  A  1.2360  0.2662   \n",
       "17    9  9:17795571  17795571  A  0.000149  0.000000  G  1.2360  0.2662   \n",
       "18    9  9:17795582  17795582  A  0.000000  0.000000  G     NaN     NaN   \n",
       "19    9  9:17795598  17795598  G  0.000149  0.000000  A  1.2360  0.2662   \n",
       "20    9  9:17795649  17795649  G  0.000149  0.000000  A  1.2360  0.2662   \n",
       "21    9  9:17795705  17795705  G  0.000149  0.000000  A  1.2360  0.2662   \n",
       "\n",
       "        OR  \n",
       "0      NaN  \n",
       "1      NaN  \n",
       "2   0.0000  \n",
       "3   0.0000  \n",
       "4      NaN  \n",
       "5   0.0000  \n",
       "6   0.0000  \n",
       "7      NaN  \n",
       "8      NaN  \n",
       "9      NaN  \n",
       "10  0.0000  \n",
       "11     NaN  \n",
       "12  0.0000  \n",
       "13  1.2960  \n",
       "14  0.0000  \n",
       "15  0.6181  \n",
       "16     NaN  \n",
       "17     NaN  \n",
       "18     NaN  \n",
       "19     NaN  \n",
       "20     NaN  \n",
       "21     NaN  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_SH3GL2.assoc', delim_whitespace=True)\n",
    "SH3GL2_freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [CHR, SNP, BP, A1, F_A, F_U, A2, CHISQ, P, OR]\n",
       "Index: []"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# visualize freq file for nonsyn variants comparing HC vs PD with p < 0.05 \n",
    "nonsyn_SH3GL2_freq_p_0_05 = SH3GL2_freq[SH3GL2_freq['P']  <= 0.05]\n",
    "nonsyn_SH3GL2_freq_p_0_05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "reduced_intronic = intronic[[\"Chr\", \"Start\", \"End\", \"Gene.refGene\"]]\n",
    "reduced_intronic.to_csv(f'{WORK_DIR}/reduced_intronic.txt', sep = '\\t', index = False, header= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to intronic_pheno_SH3GL2.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile pheno_SH3GL2\n",
      "  --extract range reduced_intronic.txt\n",
      "  --out intronic_pheno_SH3GL2\n",
      "  --recode A\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "14592 variants loaded from .bim file.\n",
      "10418 people (5782 males, 4636 females) loaded from .fam.\n",
      "7512 phenotype values loaded from .fam.\n",
      "--extract range: 751 variants excluded.\n",
      "--extract range: 13841 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 10418 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999808.\n",
      "13841 variants and 10418 people pass filters and QC.\n",
      "Among remaining phenotypes, 3359 are cases and 4153 are controls.  (2906\n",
      "phenotypes are missing.)\n",
      "--recode A to intronic_pheno_SH3GL2.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_SH3GL2 --extract range reduced_intronic.txt --recode A --out intronic_pheno_SH3GL2 --allow-no-sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>9:17579314_A</th>\n",
       "      <th>9:17579333_C</th>\n",
       "      <th>9:17579344_G</th>\n",
       "      <th>9:17579348_T</th>\n",
       "      <th>...</th>\n",
       "      <th>9:17795344_T</th>\n",
       "      <th>9:17795347_G</th>\n",
       "      <th>9:17795381_C</th>\n",
       "      <th>9:17795467_T</th>\n",
       "      <th>9:17795472_A</th>\n",
       "      <th>9:17795484_A</th>\n",
       "      <th>9:17795497_G</th>\n",
       "      <th>9:17795505_G</th>\n",
       "      <th>9:17795527_T</th>\n",
       "      <th>9:17795533_G</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HB-PD_INVAB109VHC</td>\n",
       "      <td>HB-PD_INVAB109VHC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HB-PD_INVAB289LG3</td>\n",
       "      <td>HB-PD_INVAB289LG3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HB-PD_INVAC488AAF</td>\n",
       "      <td>HB-PD_INVAC488AAF</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HB-PD_INVAE296YP8</td>\n",
       "      <td>HB-PD_INVAE296YP8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HB-PD_INVAJ549VWD</td>\n",
       "      <td>HB-PD_INVAJ549VWD</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows √ó 13847 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 FID                IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0  HB-PD_INVAB109VHC  HB-PD_INVAB109VHC    0    0    2          1   \n",
       "1  HB-PD_INVAB289LG3  HB-PD_INVAB289LG3    0    0    2          1   \n",
       "2  HB-PD_INVAC488AAF  HB-PD_INVAC488AAF    0    0    1          1   \n",
       "3  HB-PD_INVAE296YP8  HB-PD_INVAE296YP8    0    0    2          1   \n",
       "4  HB-PD_INVAJ549VWD  HB-PD_INVAJ549VWD    0    0    2          1   \n",
       "\n",
       "   9:17579314_A  9:17579333_C  9:17579344_G  9:17579348_T  ...  9:17795344_T  \\\n",
       "0             0             0             0             0  ...             0   \n",
       "1             0             0             0             0  ...             0   \n",
       "2             0             0             0             0  ...             0   \n",
       "3             0             0             0             0  ...             0   \n",
       "4             0             0             0             0  ...             0   \n",
       "\n",
       "   9:17795347_G  9:17795381_C  9:17795467_T  9:17795472_A  9:17795484_A  \\\n",
       "0             0             0             0             0             0   \n",
       "1             0             0             0             0             0   \n",
       "2             0             0             0             0             0   \n",
       "3             0             0             0             0             0   \n",
       "4             0             0             0             0             0   \n",
       "\n",
       "   9:17795497_G  9:17795505_G  9:17795527_T  9:17795533_G  \n",
       "0             0             0             0             0  \n",
       "1             0             0             0             0  \n",
       "2             0             0             0             0  \n",
       "3             0             0             0             0  \n",
       "4             0             0             0             0  \n",
       "\n",
       "[5 rows x 13847 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_intronic = pd.read_csv(f'{WORK_DIR}/intronic_pheno_SH3GL2.raw', delim_whitespace=True)\n",
    "SH3GL2_intronic.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>9:17579314_A</th>\n",
       "      <th>9:17579333_C</th>\n",
       "      <th>9:17579344_G</th>\n",
       "      <th>9:17579348_T</th>\n",
       "      <th>...</th>\n",
       "      <th>9:17795344_T</th>\n",
       "      <th>9:17795347_G</th>\n",
       "      <th>9:17795381_C</th>\n",
       "      <th>9:17795467_T</th>\n",
       "      <th>9:17795472_A</th>\n",
       "      <th>9:17795484_A</th>\n",
       "      <th>9:17795497_G</th>\n",
       "      <th>9:17795505_G</th>\n",
       "      <th>9:17795527_T</th>\n",
       "      <th>9:17795533_G</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4899</th>\n",
       "      <td>LC-100008</td>\n",
       "      <td>LC-100008</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4926</th>\n",
       "      <td>LC-1080006</td>\n",
       "      <td>LC-1080006</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4935</th>\n",
       "      <td>LC-1120006</td>\n",
       "      <td>LC-1120006</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4973</th>\n",
       "      <td>LC-1370006</td>\n",
       "      <td>LC-1370006</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5055</th>\n",
       "      <td>LC-230005</td>\n",
       "      <td>LC-230005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows √ó 13847 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             FID         IID  PAT  MAT  SEX  PHENOTYPE  9:17579314_A  \\\n",
       "4899   LC-100008   LC-100008    0    0    2          2             0   \n",
       "4926  LC-1080006  LC-1080006    0    0    2          2             0   \n",
       "4935  LC-1120006  LC-1120006    0    0    1          2             0   \n",
       "4973  LC-1370006  LC-1370006    0    0    2          2             0   \n",
       "5055   LC-230005   LC-230005    0    0    2          2             0   \n",
       "\n",
       "      9:17579333_C  9:17579344_G  9:17579348_T  ...  9:17795344_T  \\\n",
       "4899             0             0             0  ...             0   \n",
       "4926             0             0             0  ...             0   \n",
       "4935             0             0             0  ...             0   \n",
       "4973             0             0             0  ...             0   \n",
       "5055             0             0             0  ...             0   \n",
       "\n",
       "      9:17795347_G  9:17795381_C  9:17795467_T  9:17795472_A  9:17795484_A  \\\n",
       "4899             0             0             0             0             0   \n",
       "4926             0             0             0             0             0   \n",
       "4935             0             0             0             0             0   \n",
       "4973             0             0             0             0             0   \n",
       "5055             0             0             0             0             0   \n",
       "\n",
       "      9:17795497_G  9:17795505_G  9:17795527_T  9:17795533_G  \n",
       "4899             0             0             0             0  \n",
       "4926             0             0             0             0  \n",
       "4935             0             0             0             0  \n",
       "4973             0             0             0             0  \n",
       "5055             0             0             0             0  \n",
       "\n",
       "[5 rows x 13847 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore homozygotes for 9:17579692_T in cases\n",
    "SH3GL2_hom_cases_nalls_rs13294100 = SH3GL2_intronic[(SH3GL2_intronic['9:17579692_T'] == 2) & (SH3GL2_intronic['PHENOTYPE'] == 2)]\n",
    "SH3GL2_hom_cases_nalls_rs13294100.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(393, 13847)"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_hom_cases_nalls_rs13294100.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Calculate frequency against the total N of cases: 0.116999'"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'Calculate frequency against the total N of cases: {:f}'.format(393/3359)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>9:17579314_A</th>\n",
       "      <th>9:17579333_C</th>\n",
       "      <th>9:17579344_G</th>\n",
       "      <th>9:17579348_T</th>\n",
       "      <th>...</th>\n",
       "      <th>9:17795344_T</th>\n",
       "      <th>9:17795347_G</th>\n",
       "      <th>9:17795381_C</th>\n",
       "      <th>9:17795467_T</th>\n",
       "      <th>9:17795472_A</th>\n",
       "      <th>9:17795484_A</th>\n",
       "      <th>9:17795497_G</th>\n",
       "      <th>9:17795505_G</th>\n",
       "      <th>9:17795527_T</th>\n",
       "      <th>9:17795533_G</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HB-PD_INVAR650VWG</td>\n",
       "      <td>HB-PD_INVAR650VWG</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>HB-PD_INVBD604UJD</td>\n",
       "      <td>HB-PD_INVBD604UJD</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HB-PD_INVBF956HZC</td>\n",
       "      <td>HB-PD_INVBF956HZC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>HB-PD_INVBL268JTC</td>\n",
       "      <td>HB-PD_INVBL268JTC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>HB-PD_INVBX181DJ7</td>\n",
       "      <td>HB-PD_INVBX181DJ7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows √ó 13847 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  FID                IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "11  HB-PD_INVAR650VWG  HB-PD_INVAR650VWG    0    0    2          1   \n",
       "17  HB-PD_INVBD604UJD  HB-PD_INVBD604UJD    0    0    2          1   \n",
       "20  HB-PD_INVBF956HZC  HB-PD_INVBF956HZC    0    0    2          1   \n",
       "23  HB-PD_INVBL268JTC  HB-PD_INVBL268JTC    0    0    2          1   \n",
       "33  HB-PD_INVBX181DJ7  HB-PD_INVBX181DJ7    0    0    2          1   \n",
       "\n",
       "    9:17579314_A  9:17579333_C  9:17579344_G  9:17579348_T  ...  9:17795344_T  \\\n",
       "11             0             0             0             0  ...             0   \n",
       "17             0             0             0             0  ...             0   \n",
       "20             0             0             0             0  ...             0   \n",
       "23             0             0             0             0  ...             0   \n",
       "33             0             0             0             0  ...             0   \n",
       "\n",
       "    9:17795347_G  9:17795381_C  9:17795467_T  9:17795472_A  9:17795484_A  \\\n",
       "11             0             0             0             0             0   \n",
       "17             0             0             0             0             0   \n",
       "20             0             0             0             0             0   \n",
       "23             0             0             0             0             0   \n",
       "33             0             0             0             0             0   \n",
       "\n",
       "    9:17795497_G  9:17795505_G  9:17795527_T  9:17795533_G  \n",
       "11             0             0             0             0  \n",
       "17             0             0             0             0  \n",
       "20             0             0             0             0  \n",
       "23             0             0             0             0  \n",
       "33             0             0             0             0  \n",
       "\n",
       "[5 rows x 13847 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for 9:17579692_T in controls\n",
    "SH3GL2_hom_controls_nalls_rs13294100 = SH3GL2_intronic[(SH3GL2_intronic['9:17579692_T'] == 2) & (SH3GL2_intronic['PHENOTYPE'] == 1)]\n",
    "SH3GL2_hom_controls_nalls_rs13294100.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(581, 13847)"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_hom_controls_nalls_rs13294100.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Calculate frequency against the total N of controls: 0.139899'"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'Calculate frequency against the total N of controls: {:f}'.format(581/4153)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(355, 13847)"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# unknwon phenotype -9\n",
    "SH3GL2_hom_U_nalls_rs13294100 = SH3GL2_intronic[(SH3GL2_intronic['9:17579692_T'] == 2) & (SH3GL2_intronic['PHENOTYPE'] == -9)]\n",
    "SH3GL2_hom_U_nalls_rs13294100.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>9:17579314_A</th>\n",
       "      <th>9:17579333_C</th>\n",
       "      <th>9:17579344_G</th>\n",
       "      <th>9:17579348_T</th>\n",
       "      <th>...</th>\n",
       "      <th>9:17795344_T</th>\n",
       "      <th>9:17795347_G</th>\n",
       "      <th>9:17795381_C</th>\n",
       "      <th>9:17795467_T</th>\n",
       "      <th>9:17795472_A</th>\n",
       "      <th>9:17795484_A</th>\n",
       "      <th>9:17795497_G</th>\n",
       "      <th>9:17795505_G</th>\n",
       "      <th>9:17795527_T</th>\n",
       "      <th>9:17795533_G</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4949</th>\n",
       "      <td>LC-120008</td>\n",
       "      <td>LC-120008</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4962</th>\n",
       "      <td>LC-130004</td>\n",
       "      <td>LC-130004</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4967</th>\n",
       "      <td>LC-1310006</td>\n",
       "      <td>LC-1310006</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5010</th>\n",
       "      <td>LC-1620001</td>\n",
       "      <td>LC-1620001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5033</th>\n",
       "      <td>LC-1970001</td>\n",
       "      <td>LC-1970001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows √ó 13847 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             FID         IID  PAT  MAT  SEX  PHENOTYPE  9:17579314_A  \\\n",
       "4949   LC-120008   LC-120008    0    0    2          2             0   \n",
       "4962   LC-130004   LC-130004    0    0    2          2             0   \n",
       "4967  LC-1310006  LC-1310006    0    0    2          2             0   \n",
       "5010  LC-1620001  LC-1620001    0    0    1          2             0   \n",
       "5033  LC-1970001  LC-1970001    0    0    1          2             0   \n",
       "\n",
       "      9:17579333_C  9:17579344_G  9:17579348_T  ...  9:17795344_T  \\\n",
       "4949             0             0             0  ...             0   \n",
       "4962             0             0             0  ...             0   \n",
       "4967             0             0             0  ...             0   \n",
       "5010             0             0             0  ...             0   \n",
       "5033             0             0             0  ...             0   \n",
       "\n",
       "      9:17795347_G  9:17795381_C  9:17795467_T  9:17795472_A  9:17795484_A  \\\n",
       "4949             0             0             0             0             0   \n",
       "4962             0             0             0             0             0   \n",
       "4967             0             0             0             0             0   \n",
       "5010             0             0             0             0             0   \n",
       "5033             0             0             0             0             0   \n",
       "\n",
       "      9:17795497_G  9:17795505_G  9:17795527_T  9:17795533_G  \n",
       "4949             0             0             0             0  \n",
       "4962             0             0             0             0  \n",
       "4967             0             0             0             0  \n",
       "5010             0             0             0             0  \n",
       "5033             0             0             0             0  \n",
       "\n",
       "[5 rows x 13847 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore homozygotes for 9:17727067_G in cases\n",
    "SH3GL2_hom_cases_nalls_rs10756907 = SH3GL2_intronic[(SH3GL2_intronic['9:17727067_G'] == 2) & (SH3GL2_intronic['PHENOTYPE'] == 2)]\n",
    "SH3GL2_hom_cases_nalls_rs10756907.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(190, 13847)"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_hom_cases_nalls_rs10756907.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Calculate frequency against the total N of cases: 0.056564'"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'Calculate frequency against the total N of cases: {:f}'.format(190/3359)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>9:17579314_A</th>\n",
       "      <th>9:17579333_C</th>\n",
       "      <th>9:17579344_G</th>\n",
       "      <th>9:17579348_T</th>\n",
       "      <th>...</th>\n",
       "      <th>9:17795344_T</th>\n",
       "      <th>9:17795347_G</th>\n",
       "      <th>9:17795381_C</th>\n",
       "      <th>9:17795467_T</th>\n",
       "      <th>9:17795472_A</th>\n",
       "      <th>9:17795484_A</th>\n",
       "      <th>9:17795497_G</th>\n",
       "      <th>9:17795505_G</th>\n",
       "      <th>9:17795527_T</th>\n",
       "      <th>9:17795533_G</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>HB-PD_INVBT337PCG</td>\n",
       "      <td>HB-PD_INVBT337PCG</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>HB-PD_INVCR113DZ8</td>\n",
       "      <td>HB-PD_INVCR113DZ8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>HB-PD_INVEW090MBW</td>\n",
       "      <td>HB-PD_INVEW090MBW</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>HB-PD_INVNK524UDW</td>\n",
       "      <td>HB-PD_INVNK524UDW</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>HB-PD_INVNP408GYU</td>\n",
       "      <td>HB-PD_INVNP408GYU</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows √ó 13847 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   FID                IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "29   HB-PD_INVBT337PCG  HB-PD_INVBT337PCG    0    0    2          1   \n",
       "45   HB-PD_INVCR113DZ8  HB-PD_INVCR113DZ8    0    0    1          1   \n",
       "78   HB-PD_INVEW090MBW  HB-PD_INVEW090MBW    0    0    2          1   \n",
       "171  HB-PD_INVNK524UDW  HB-PD_INVNK524UDW    0    0    2          1   \n",
       "174  HB-PD_INVNP408GYU  HB-PD_INVNP408GYU    0    0    2          1   \n",
       "\n",
       "     9:17579314_A  9:17579333_C  9:17579344_G  9:17579348_T  ...  \\\n",
       "29              0             0             0             0  ...   \n",
       "45              0             0             0             0  ...   \n",
       "78              0             0             0             0  ...   \n",
       "171             0             0             0             0  ...   \n",
       "174             0             0             0             0  ...   \n",
       "\n",
       "     9:17795344_T  9:17795347_G  9:17795381_C  9:17795467_T  9:17795472_A  \\\n",
       "29              0             0             0             0             0   \n",
       "45              0             0             0             0             0   \n",
       "78              0             0             0             0             0   \n",
       "171             0             0             0             0             0   \n",
       "174             0             0             0             0             0   \n",
       "\n",
       "     9:17795484_A  9:17795497_G  9:17795505_G  9:17795527_T  9:17795533_G  \n",
       "29              0             0             0             0             0  \n",
       "45              0             0             0             0             0  \n",
       "78              0             0             0             0             0  \n",
       "171             0             0             0             0             0  \n",
       "174             0             0             0             0             0  \n",
       "\n",
       "[5 rows x 13847 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for 9:17727067_G in controls\n",
    "SH3GL2_hom_controls_nalls_rs10756907 = SH3GL2_intronic[(SH3GL2_intronic['9:17727067_G'] == 2) & (SH3GL2_intronic['PHENOTYPE'] == 1)]\n",
    "SH3GL2_hom_controls_nalls_rs10756907.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(190, 13847)"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_hom_controls_nalls_rs10756907.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Calculate frequency against the total N of controls: 0.045750'"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'Calculate frequency against the total N of controls: {:f}'.format(190/4153)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(170, 13847)"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# unknwon phenotype -9\n",
    "SH3GL2_hom_U_nalls_rs10756907 = SH3GL2_intronic[(SH3GL2_intronic['9:17727067_G'] == 2) & (SH3GL2_intronic['PHENOTYPE'] == -9)]\n",
    "SH3GL2_hom_U_nalls_rs10756907.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_SH3GL2.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile pheno_SH3GL2\n",
      "  --extract range reduced_coding_nonsynonymous.txt\n",
      "  --out coding_nonsynonymous_pheno_SH3GL2\n",
      "  --recode A\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "14592 variants loaded from .bim file.\n",
      "10418 people (5782 males, 4636 females) loaded from .fam.\n",
      "7512 phenotype values loaded from .fam.\n",
      "--extract range: 14570 variants excluded.\n",
      "--extract range: 22 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 10418 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "22 variants and 10418 people pass filters and QC.\n",
      "Among remaining phenotypes, 3359 are cases and 4153 are controls.  (2906\n",
      "phenotypes are missing.)\n",
      "--recode A to coding_nonsynonymous_pheno_SH3GL2.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_SH3GL2 --extract range reduced_coding_nonsynonymous.txt --recode A --out coding_nonsynonymous_pheno_SH3GL2 --allow-no-sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>9:17747120_C</th>\n",
       "      <th>9:17747126_T</th>\n",
       "      <th>9:17786478_C</th>\n",
       "      <th>9:17786494_A</th>\n",
       "      <th>...</th>\n",
       "      <th>9:17793459_G</th>\n",
       "      <th>9:17793465_T</th>\n",
       "      <th>9:17793468_C</th>\n",
       "      <th>9:17793471_T</th>\n",
       "      <th>9:17795552_G</th>\n",
       "      <th>9:17795571_A</th>\n",
       "      <th>9:17795582_A</th>\n",
       "      <th>9:17795598_G</th>\n",
       "      <th>9:17795649_G</th>\n",
       "      <th>9:17795705_G</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HB-PD_INVAB109VHC</td>\n",
       "      <td>HB-PD_INVAB109VHC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HB-PD_INVAB289LG3</td>\n",
       "      <td>HB-PD_INVAB289LG3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HB-PD_INVAC488AAF</td>\n",
       "      <td>HB-PD_INVAC488AAF</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HB-PD_INVAE296YP8</td>\n",
       "      <td>HB-PD_INVAE296YP8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HB-PD_INVAJ549VWD</td>\n",
       "      <td>HB-PD_INVAJ549VWD</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows √ó 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 FID                IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0  HB-PD_INVAB109VHC  HB-PD_INVAB109VHC    0    0    2          1   \n",
       "1  HB-PD_INVAB289LG3  HB-PD_INVAB289LG3    0    0    2          1   \n",
       "2  HB-PD_INVAC488AAF  HB-PD_INVAC488AAF    0    0    1          1   \n",
       "3  HB-PD_INVAE296YP8  HB-PD_INVAE296YP8    0    0    2          1   \n",
       "4  HB-PD_INVAJ549VWD  HB-PD_INVAJ549VWD    0    0    2          1   \n",
       "\n",
       "   9:17747120_C  9:17747126_T  9:17786478_C  9:17786494_A  ...  9:17793459_G  \\\n",
       "0             0             0             0             0  ...             0   \n",
       "1             0             0             0             0  ...             0   \n",
       "2             0             0             0             0  ...             0   \n",
       "3             0             0             0             0  ...             0   \n",
       "4             0             0             0             0  ...             0   \n",
       "\n",
       "   9:17793465_T  9:17793468_C  9:17793471_T  9:17795552_G  9:17795571_A  \\\n",
       "0             0             0             0             0             0   \n",
       "1             0             0             0             0             0   \n",
       "2             0             0             0             0             0   \n",
       "3             0             0             0             0             0   \n",
       "4             0             0             0             0             0   \n",
       "\n",
       "   9:17795582_A  9:17795598_G  9:17795649_G  9:17795705_G  \n",
       "0             0             0             0             0  \n",
       "1             0             0             0             0  \n",
       "2             0             0             0             0  \n",
       "3             0             0             0             0  \n",
       "4             0             0             0             0  \n",
       "\n",
       "[5 rows x 28 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_recode = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_SH3GL2.raw', delim_whitespace=True)\n",
    "SH3GL2_recode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>9:17747120_C</th>\n",
       "      <th>9:17747126_T</th>\n",
       "      <th>9:17786478_C</th>\n",
       "      <th>9:17786494_A</th>\n",
       "      <th>...</th>\n",
       "      <th>9:17793459_G</th>\n",
       "      <th>9:17793465_T</th>\n",
       "      <th>9:17793468_C</th>\n",
       "      <th>9:17793471_T</th>\n",
       "      <th>9:17795552_G</th>\n",
       "      <th>9:17795571_A</th>\n",
       "      <th>9:17795582_A</th>\n",
       "      <th>9:17795598_G</th>\n",
       "      <th>9:17795649_G</th>\n",
       "      <th>9:17795705_G</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows √ó 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [FID, IID, PAT, MAT, SEX, PHENOTYPE, 9:17747120_C, 9:17747126_T, 9:17786478_C, 9:17786494_A, 9:17787437_T, 9:17787457_G, 9:17787498_A, 9:17789477_A, 9:17791290_C, 9:17791300_G, 9:17791316_T, 9:17793419_G, 9:17793459_G, 9:17793465_T, 9:17793468_C, 9:17793471_T, 9:17795552_G, 9:17795571_A, 9:17795582_A, 9:17795598_G, 9:17795649_G, 9:17795705_G]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 28 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore homozygotes for 9:17793465_T in cases\n",
    "SH3GL2_hom_cases_G276V = SH3GL2_recode[(SH3GL2_recode['9:17793465_T'] == 2) & (SH3GL2_recode['PHENOTYPE'] == 2)]\n",
    "SH3GL2_hom_cases_G276V.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0, 28)"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_hom_cases_G276V.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Calculate frequency against the total N of cases: 0.000000'"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'Calculate frequency against the total N of cases: {:f}'.format(0/3359)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>9:17747120_C</th>\n",
       "      <th>9:17747126_T</th>\n",
       "      <th>9:17786478_C</th>\n",
       "      <th>9:17786494_A</th>\n",
       "      <th>...</th>\n",
       "      <th>9:17793459_G</th>\n",
       "      <th>9:17793465_T</th>\n",
       "      <th>9:17793468_C</th>\n",
       "      <th>9:17793471_T</th>\n",
       "      <th>9:17795552_G</th>\n",
       "      <th>9:17795571_A</th>\n",
       "      <th>9:17795582_A</th>\n",
       "      <th>9:17795598_G</th>\n",
       "      <th>9:17795649_G</th>\n",
       "      <th>9:17795705_G</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows √ó 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [FID, IID, PAT, MAT, SEX, PHENOTYPE, 9:17747120_C, 9:17747126_T, 9:17786478_C, 9:17786494_A, 9:17787437_T, 9:17787457_G, 9:17787498_A, 9:17789477_A, 9:17791290_C, 9:17791300_G, 9:17791316_T, 9:17793419_G, 9:17793459_G, 9:17793465_T, 9:17793468_C, 9:17793471_T, 9:17795552_G, 9:17795571_A, 9:17795582_A, 9:17795598_G, 9:17795649_G, 9:17795705_G]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 28 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for 9:17793465_T in controls\n",
    "SH3GL2_hom_controls_G276V = SH3GL2_recode[(SH3GL2_recode['9:17793465_T'] == 2) & (SH3GL2_recode['PHENOTYPE'] == 1)]\n",
    "SH3GL2_hom_controls_G276V.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0, 28)"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_hom_controls_G276V.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Calculate frequency against the total N of controls: 0.000000'"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'Calculate frequency against the total N of controls: {:f}'.format(0/4153)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1, 28)"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# unknwon phenotype -9\n",
    "SH3GL2_hom_U_G276V = SH3GL2_recode[(SH3GL2_recode['9:17793465_T'] == 2) & (SH3GL2_recode['PHENOTYPE'] == -9)]\n",
    "SH3GL2_hom_U_G276V.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pearson correlation and conditional analyses rs13294100 (Nalls et al 2019) and rs150543523 (AMP PD coding variant)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_SH3GL2_rs13294100_nalls.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_SH3GL2\n",
      "  --chr 9\n",
      "  --double-id\n",
      "  --from-bp 17579692\n",
      "  --out pheno_SH3GL2_rs13294100_nalls\n",
      "  --recode A\n",
      "  --to-bp 17579692\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "1 out of 14592 variants loaded from .bim file.\n",
      "10418 people (5782 males, 4636 females) loaded from .fam.\n",
      "7512 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 10418 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "1 variant and 10418 people pass filters and QC.\n",
      "Among remaining phenotypes, 3359 are cases and 4153 are controls.  (2906\n",
      "phenotypes are missing.)\n",
      "--recode A to pheno_SH3GL2_rs13294100_nalls.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_SH3GL2 --chr 9 --from-bp 17579692 --to-bp 17579692 --recode A --out pheno_SH3GL2_rs13294100_nalls --double-id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_SH3GL2_rs150543523_G276V.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_SH3GL2\n",
      "  --chr 9\n",
      "  --double-id\n",
      "  --from-bp 17793465\n",
      "  --out pheno_SH3GL2_rs150543523_G276V\n",
      "  --recode A\n",
      "  --to-bp 17793465\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "1 out of 14592 variants loaded from .bim file.\n",
      "10418 people (5782 males, 4636 females) loaded from .fam.\n",
      "7512 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 10418 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "1 variant and 10418 people pass filters and QC.\n",
      "Among remaining phenotypes, 3359 are cases and 4153 are controls.  (2906\n",
      "phenotypes are missing.)\n",
      "--recode A to pheno_SH3GL2_rs150543523_G276V.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_SH3GL2 --chr 9 --from-bp 17793465 --to-bp 17793465 --recode A --out pheno_SH3GL2_rs150543523_G276V --double-id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "rs13294100 = pd.read_csv(\"/home/jupyter/SH3GL2/pheno_SH3GL2_rs13294100_nalls.raw\",sep=\" \")\n",
    "rs150543523 = pd.read_csv(\"/home/jupyter/SH3GL2/pheno_SH3GL2_rs150543523_G276V.raw\",sep=\" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>9:17579692_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HB-PD_INVAB109VHC</td>\n",
       "      <td>HB-PD_INVAB109VHC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HB-PD_INVAB289LG3</td>\n",
       "      <td>HB-PD_INVAB289LG3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HB-PD_INVAC488AAF</td>\n",
       "      <td>HB-PD_INVAC488AAF</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HB-PD_INVAE296YP8</td>\n",
       "      <td>HB-PD_INVAE296YP8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HB-PD_INVAJ549VWD</td>\n",
       "      <td>HB-PD_INVAJ549VWD</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10413</th>\n",
       "      <td>SY-PDZV652BE6</td>\n",
       "      <td>SY-PDZV652BE6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10414</th>\n",
       "      <td>SY-PDZX554VNB</td>\n",
       "      <td>SY-PDZX554VNB</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10415</th>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10416</th>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10417</th>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10418 rows √ó 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     FID                IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0      HB-PD_INVAB109VHC  HB-PD_INVAB109VHC    0    0    2          1   \n",
       "1      HB-PD_INVAB289LG3  HB-PD_INVAB289LG3    0    0    2          1   \n",
       "2      HB-PD_INVAC488AAF  HB-PD_INVAC488AAF    0    0    1          1   \n",
       "3      HB-PD_INVAE296YP8  HB-PD_INVAE296YP8    0    0    2          1   \n",
       "4      HB-PD_INVAJ549VWD  HB-PD_INVAJ549VWD    0    0    2          1   \n",
       "...                  ...                ...  ...  ...  ...        ...   \n",
       "10413      SY-PDZV652BE6      SY-PDZV652BE6    0    0    1          2   \n",
       "10414      SY-PDZX554VNB      SY-PDZX554VNB    0    0    1          2   \n",
       "10415      SY-PDZX943HWN      SY-PDZX943HWN    0    0    1          2   \n",
       "10416      SY-PDZY968RFA      SY-PDZY968RFA    0    0    1          2   \n",
       "10417      SY-PDZZ260HUM      SY-PDZZ260HUM    0    0    2          2   \n",
       "\n",
       "       9:17579692_T  \n",
       "0                 0  \n",
       "1                 1  \n",
       "2                 0  \n",
       "3                 0  \n",
       "4                 0  \n",
       "...             ...  \n",
       "10413             0  \n",
       "10414             1  \n",
       "10415             0  \n",
       "10416             1  \n",
       "10417             0  \n",
       "\n",
       "[10418 rows x 7 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs13294100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>9:17793465_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HB-PD_INVAB109VHC</td>\n",
       "      <td>HB-PD_INVAB109VHC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HB-PD_INVAB289LG3</td>\n",
       "      <td>HB-PD_INVAB289LG3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HB-PD_INVAC488AAF</td>\n",
       "      <td>HB-PD_INVAC488AAF</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HB-PD_INVAE296YP8</td>\n",
       "      <td>HB-PD_INVAE296YP8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HB-PD_INVAJ549VWD</td>\n",
       "      <td>HB-PD_INVAJ549VWD</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10413</th>\n",
       "      <td>SY-PDZV652BE6</td>\n",
       "      <td>SY-PDZV652BE6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10414</th>\n",
       "      <td>SY-PDZX554VNB</td>\n",
       "      <td>SY-PDZX554VNB</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10415</th>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10416</th>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10417</th>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10418 rows √ó 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     FID                IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0      HB-PD_INVAB109VHC  HB-PD_INVAB109VHC    0    0    2          1   \n",
       "1      HB-PD_INVAB289LG3  HB-PD_INVAB289LG3    0    0    2          1   \n",
       "2      HB-PD_INVAC488AAF  HB-PD_INVAC488AAF    0    0    1          1   \n",
       "3      HB-PD_INVAE296YP8  HB-PD_INVAE296YP8    0    0    2          1   \n",
       "4      HB-PD_INVAJ549VWD  HB-PD_INVAJ549VWD    0    0    2          1   \n",
       "...                  ...                ...  ...  ...  ...        ...   \n",
       "10413      SY-PDZV652BE6      SY-PDZV652BE6    0    0    1          2   \n",
       "10414      SY-PDZX554VNB      SY-PDZX554VNB    0    0    1          2   \n",
       "10415      SY-PDZX943HWN      SY-PDZX943HWN    0    0    1          2   \n",
       "10416      SY-PDZY968RFA      SY-PDZY968RFA    0    0    1          2   \n",
       "10417      SY-PDZZ260HUM      SY-PDZZ260HUM    0    0    2          2   \n",
       "\n",
       "       9:17793465_T  \n",
       "0                 0  \n",
       "1                 0  \n",
       "2                 0  \n",
       "3                 0  \n",
       "4                 0  \n",
       "...             ...  \n",
       "10413             0  \n",
       "10414             0  \n",
       "10415             0  \n",
       "10416             0  \n",
       "10417             0  \n",
       "\n",
       "[10418 rows x 7 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs150543523"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "total = pd.merge(rs13294100, rs150543523, on=\"FID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "total = total[(total['PHENOTYPE_x']==1) | (total['PHENOTYPE_x']==2)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "total = total[[\"FID\",'PHENOTYPE_x',\"9:17579692_T\",\"9:17793465_T\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>Nalls_9_17579692_T</th>\n",
       "      <th>G276V_9_17793465_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HB-PD_INVAB109VHC</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HB-PD_INVAB289LG3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HB-PD_INVAC488AAF</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HB-PD_INVAE296YP8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HB-PD_INVAJ549VWD</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10413</th>\n",
       "      <td>SY-PDZV652BE6</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10414</th>\n",
       "      <td>SY-PDZX554VNB</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10415</th>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10416</th>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10417</th>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7512 rows √ó 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     FID  PHENOTYPE  Nalls_9_17579692_T  G276V_9_17793465_T\n",
       "0      HB-PD_INVAB109VHC          1                   0                   0\n",
       "1      HB-PD_INVAB289LG3          1                   1                   0\n",
       "2      HB-PD_INVAC488AAF          1                   0                   0\n",
       "3      HB-PD_INVAE296YP8          1                   0                   0\n",
       "4      HB-PD_INVAJ549VWD          1                   0                   0\n",
       "...                  ...        ...                 ...                 ...\n",
       "10413      SY-PDZV652BE6          2                   0                   0\n",
       "10414      SY-PDZX554VNB          2                   1                   0\n",
       "10415      SY-PDZX943HWN          2                   0                   0\n",
       "10416      SY-PDZY968RFA          2                   1                   0\n",
       "10417      SY-PDZZ260HUM          2                   0                   0\n",
       "\n",
       "[7512 rows x 4 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total.columns = [\"FID\",'PHENOTYPE',\"Nalls_9_17579692_T\",\"G276V_9_17793465_T\"]\n",
    "total"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "total.to_csv(f'{WORKSPACE_BUCKET}/AMPPD_WGS_Nalls_G276V_rs13294100.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "In addition: Warning message:\n",
       "In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
       "  libraries ‚Äò/home/jupyter/packages‚Äô, ‚Äò/usr/lib/R/site-library‚Äô contain no packages\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "BILLING_PROJECT_ID <- Sys.getenv('GOOGLE_PROJECT')\n",
    "WORKSPACE_NAMESPACE <- Sys.getenv('WORKSPACE_NAMESPACE')\n",
    "WORKSPACE_NAME <- Sys.getenv('WORKSPACE_NAME')\n",
    "WORKSPACE_BUCKET <- Sys.getenv('WORKSPACE_BUCKET')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "gcs_read_file <- function(path) {\n",
    "    pipe(str_glue('gsutil -u {BILLING_PROJECT_ID} cat {path}'))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "shell_do <- function(command) {\n",
    "    print(paste('Executing: ', command))\n",
    "    system(command, intern = TRUE)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  X               FID PHENOTYPE Nalls_9_17579692_T G276V_9_17793465_T\n",
      "1 0 HB-PD_INVAB109VHC         1                  0                  0\n",
      "2 1 HB-PD_INVAB289LG3         1                  1                  0\n",
      "3 2 HB-PD_INVAC488AAF         1                  0                  0\n",
      "4 3 HB-PD_INVAE296YP8         1                  0                  0\n",
      "5 4 HB-PD_INVAJ549VWD         1                  0                  0\n",
      "6 5 HB-PD_INVAK106DV5         1                  1                  0\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "library(stringr)\n",
    "\n",
    "totalPath <- file.path({WORKSPACE_BUCKET}, 'AMPPD_WGS_Nalls_G276V_rs13294100.csv')\n",
    "tmp <- gcs_read_file(totalPath)\n",
    "tmp\n",
    "total <- read.csv(tmp,header=TRUE)\n",
    "head(total)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Call:\n",
      "lm(formula = Nalls_9_17579692_T ~ G276V_9_17793465_T, data = total)\n",
      "\n",
      "Residuals:\n",
      "    Min      1Q  Median      3Q     Max \n",
      "-0.9535 -0.7095  0.2905  0.2905  1.2905 \n",
      "\n",
      "Coefficients:\n",
      "                   Estimate Std. Error t value Pr(>|t|)    \n",
      "(Intercept)        0.709466   0.007887  89.951   <2e-16 ***\n",
      "G276V_9_17793465_T 0.244023   0.104249   2.341   0.0193 *  \n",
      "---\n",
      "Signif. codes:  0 ‚Äò***‚Äô 0.001 ‚Äò**‚Äô 0.01 ‚Äò*‚Äô 0.05 ‚Äò.‚Äô 0.1 ‚Äò ‚Äô 1\n",
      "\n",
      "Residual standard error: 0.6816 on 7510 degrees of freedom\n",
      "Multiple R-squared:  0.0007291,\tAdjusted R-squared:  0.000596 \n",
      "F-statistic: 5.479 on 1 and 7510 DF,  p-value: 0.01927\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "correlation.model <- lm(Nalls_9_17579692_T ~ G276V_9_17793465_T, data = total)\n",
    "summary(correlation.model)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Call:\n",
      "glm(formula = PHENOTYPE ~ Nalls_9_17579692_T + G276V_9_17793465_T, \n",
      "    data = total)\n",
      "\n",
      "Deviance Residuals: \n",
      "    Min       1Q   Median       3Q      Max  \n",
      "-0.5467  -0.4361  -0.3993   0.5271   0.6007  \n",
      "\n",
      "Coefficients:\n",
      "                    Estimate Std. Error t value Pr(>|t|)    \n",
      "(Intercept)         1.472899   0.008283 177.830  < 2e-16 ***\n",
      "Nalls_9_17579692_T -0.036815   0.008407  -4.379 1.21e-05 ***\n",
      "G276V_9_17793465_T  0.073832   0.075982   0.972    0.331    \n",
      "---\n",
      "Signif. codes:  0 ‚Äò***‚Äô 0.001 ‚Äò**‚Äô 0.01 ‚Äò*‚Äô 0.05 ‚Äò.‚Äô 0.1 ‚Äò ‚Äô 1\n",
      "\n",
      "(Dispersion parameter for gaussian family taken to be 0.246652)\n",
      "\n",
      "    Null deviance: 1857.0  on 7511  degrees of freedom\n",
      "Residual deviance: 1852.1  on 7509  degrees of freedom\n",
      "AIC: 10808\n",
      "\n",
      "Number of Fisher Scoring iterations: 2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "dep_corr<- glm(PHENOTYPE ~ Nalls_9_17579692_T + G276V_9_17793465_T, data=total)\n",
    "summary(dep_corr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                      Estimate  Std. Error     t value     Pr(>|t|)\n",
      "(Intercept)         1.47289902 0.008282644 177.8295677 0.000000e+00\n",
      "Nalls_9_17579692_T -0.03681502 0.008407444  -4.3788602 1.209197e-05\n",
      "G276V_9_17793465_T  0.07383158 0.075982402   0.9716931 3.312345e-01\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "summary(dep_corr)$coefficients"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 1.076625\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "exp(0.07383158)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pearson correlation and conditional analyses rs10756907 (Nalls et al 2019) and rs150543523 (AMP PD coding variant)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_SH3GL2_rs10756907_nalls.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_SH3GL2\n",
      "  --chr 9\n",
      "  --double-id\n",
      "  --from-bp 17727067\n",
      "  --out pheno_SH3GL2_rs10756907_nalls\n",
      "  --recode A\n",
      "  --to-bp 17727067\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "1 out of 14592 variants loaded from .bim file.\n",
      "10418 people (5782 males, 4636 females) loaded from .fam.\n",
      "7512 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 10418 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "1 variant and 10418 people pass filters and QC.\n",
      "Among remaining phenotypes, 3359 are cases and 4153 are controls.  (2906\n",
      "phenotypes are missing.)\n",
      "--recode A to pheno_SH3GL2_rs10756907_nalls.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_SH3GL2 --chr 9 --from-bp 17727067 --to-bp 17727067 --recode A --out pheno_SH3GL2_rs10756907_nalls --double-id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "rs10756907 = pd.read_csv(\"/home/jupyter/SH3GL2/pheno_SH3GL2_rs10756907_nalls.raw\",sep=\" \")\n",
    "rs150543523 = pd.read_csv(\"/home/jupyter/SH3GL2/pheno_SH3GL2_rs150543523_G276V.raw\",sep=\" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "total = pd.merge(rs10756907, rs150543523, on=\"FID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID_x</th>\n",
       "      <th>PAT_x</th>\n",
       "      <th>MAT_x</th>\n",
       "      <th>SEX_x</th>\n",
       "      <th>PHENOTYPE_x</th>\n",
       "      <th>9:17727067_G</th>\n",
       "      <th>IID_y</th>\n",
       "      <th>PAT_y</th>\n",
       "      <th>MAT_y</th>\n",
       "      <th>SEX_y</th>\n",
       "      <th>PHENOTYPE_y</th>\n",
       "      <th>9:17793465_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HB-PD_INVAB109VHC</td>\n",
       "      <td>HB-PD_INVAB109VHC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>HB-PD_INVAB109VHC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HB-PD_INVAB289LG3</td>\n",
       "      <td>HB-PD_INVAB289LG3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>HB-PD_INVAB289LG3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HB-PD_INVAC488AAF</td>\n",
       "      <td>HB-PD_INVAC488AAF</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>HB-PD_INVAC488AAF</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HB-PD_INVAE296YP8</td>\n",
       "      <td>HB-PD_INVAE296YP8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>HB-PD_INVAE296YP8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HB-PD_INVAJ549VWD</td>\n",
       "      <td>HB-PD_INVAJ549VWD</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>HB-PD_INVAJ549VWD</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10413</th>\n",
       "      <td>SY-PDZV652BE6</td>\n",
       "      <td>SY-PDZV652BE6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>SY-PDZV652BE6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10414</th>\n",
       "      <td>SY-PDZX554VNB</td>\n",
       "      <td>SY-PDZX554VNB</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>SY-PDZX554VNB</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10415</th>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10416</th>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10417</th>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7512 rows √ó 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     FID              IID_x  PAT_x  MAT_x  SEX_x  PHENOTYPE_x  \\\n",
       "0      HB-PD_INVAB109VHC  HB-PD_INVAB109VHC      0      0      2            1   \n",
       "1      HB-PD_INVAB289LG3  HB-PD_INVAB289LG3      0      0      2            1   \n",
       "2      HB-PD_INVAC488AAF  HB-PD_INVAC488AAF      0      0      1            1   \n",
       "3      HB-PD_INVAE296YP8  HB-PD_INVAE296YP8      0      0      2            1   \n",
       "4      HB-PD_INVAJ549VWD  HB-PD_INVAJ549VWD      0      0      2            1   \n",
       "...                  ...                ...    ...    ...    ...          ...   \n",
       "10413      SY-PDZV652BE6      SY-PDZV652BE6      0      0      1            2   \n",
       "10414      SY-PDZX554VNB      SY-PDZX554VNB      0      0      1            2   \n",
       "10415      SY-PDZX943HWN      SY-PDZX943HWN      0      0      1            2   \n",
       "10416      SY-PDZY968RFA      SY-PDZY968RFA      0      0      1            2   \n",
       "10417      SY-PDZZ260HUM      SY-PDZZ260HUM      0      0      2            2   \n",
       "\n",
       "       9:17727067_G              IID_y  PAT_y  MAT_y  SEX_y  PHENOTYPE_y  \\\n",
       "0                 1  HB-PD_INVAB109VHC      0      0      2            1   \n",
       "1                 0  HB-PD_INVAB289LG3      0      0      2            1   \n",
       "2                 0  HB-PD_INVAC488AAF      0      0      1            1   \n",
       "3                 1  HB-PD_INVAE296YP8      0      0      2            1   \n",
       "4                 1  HB-PD_INVAJ549VWD      0      0      2            1   \n",
       "...             ...                ...    ...    ...    ...          ...   \n",
       "10413             1      SY-PDZV652BE6      0      0      1            2   \n",
       "10414             1      SY-PDZX554VNB      0      0      1            2   \n",
       "10415             1      SY-PDZX943HWN      0      0      1            2   \n",
       "10416             0      SY-PDZY968RFA      0      0      1            2   \n",
       "10417             1      SY-PDZZ260HUM      0      0      2            2   \n",
       "\n",
       "       9:17793465_T  \n",
       "0                 0  \n",
       "1                 0  \n",
       "2                 0  \n",
       "3                 0  \n",
       "4                 0  \n",
       "...             ...  \n",
       "10413             0  \n",
       "10414             0  \n",
       "10415             0  \n",
       "10416             0  \n",
       "10417             0  \n",
       "\n",
       "[7512 rows x 13 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total = total[(total['PHENOTYPE_x']==1) | (total['PHENOTYPE_x']==2)]\n",
    "total"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "total = total[[\"FID\",\"PHENOTYPE_x\",\"9:17727067_G\",\"9:17793465_T\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Nalls_9_17727067_G</th>\n",
       "      <th>G276V_9_17793465_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HB-PD_INVAB109VHC</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HB-PD_INVAB289LG3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HB-PD_INVAC488AAF</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HB-PD_INVAE296YP8</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HB-PD_INVAJ549VWD</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10413</th>\n",
       "      <td>SY-PDZV652BE6</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10414</th>\n",
       "      <td>SY-PDZX554VNB</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10415</th>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10416</th>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10417</th>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7512 rows √ó 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     FID  Phenotype  Nalls_9_17727067_G  G276V_9_17793465_T\n",
       "0      HB-PD_INVAB109VHC          1                   1                   0\n",
       "1      HB-PD_INVAB289LG3          1                   0                   0\n",
       "2      HB-PD_INVAC488AAF          1                   0                   0\n",
       "3      HB-PD_INVAE296YP8          1                   1                   0\n",
       "4      HB-PD_INVAJ549VWD          1                   1                   0\n",
       "...                  ...        ...                 ...                 ...\n",
       "10413      SY-PDZV652BE6          2                   1                   0\n",
       "10414      SY-PDZX554VNB          2                   1                   0\n",
       "10415      SY-PDZX943HWN          2                   1                   0\n",
       "10416      SY-PDZY968RFA          2                   0                   0\n",
       "10417      SY-PDZZ260HUM          2                   1                   0\n",
       "\n",
       "[7512 rows x 4 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total.columns = [\"FID\",\"Phenotype\",\"Nalls_9_17727067_G\",\"G276V_9_17793465_T\"]\n",
    "total"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "total.to_csv(f'{WORKSPACE_BUCKET}/AMPPD_WGS_Nalls_G276V_rs10756907.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  X               FID Phenotype Nalls_9_17727067_G G276V_9_17793465_T\n",
      "1 0 HB-PD_INVAB109VHC         1                  1                  0\n",
      "2 1 HB-PD_INVAB289LG3         1                  0                  0\n",
      "3 2 HB-PD_INVAC488AAF         1                  0                  0\n",
      "4 3 HB-PD_INVAE296YP8         1                  1                  0\n",
      "5 4 HB-PD_INVAJ549VWD         1                  1                  0\n",
      "6 5 HB-PD_INVAK106DV5         1                  1                  0\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "library(stringr)\n",
    "\n",
    "totalPath <- file.path({WORKSPACE_BUCKET}, 'AMPPD_WGS_Nalls_G276V_rs10756907.csv')\n",
    "tmp <- gcs_read_file(totalPath)\n",
    "tmp\n",
    "total <- read.csv(tmp,header=TRUE)\n",
    "head(total)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Call:\n",
      "lm(formula = Nalls_9_17727067_G ~ G276V_9_17793465_T, data = total)\n",
      "\n",
      "Residuals:\n",
      "    Min      1Q  Median      3Q     Max \n",
      "-0.4442 -0.4442 -0.4442  0.5558  1.5558 \n",
      "\n",
      "Coefficients:\n",
      "                    Estimate Std. Error t value Pr(>|t|)    \n",
      "(Intercept)         0.444236   0.006824  65.103   <2e-16 ***\n",
      "G276V_9_17793465_T -0.211678   0.090189  -2.347   0.0189 *  \n",
      "---\n",
      "Signif. codes:  0 ‚Äò***‚Äô 0.001 ‚Äò**‚Äô 0.01 ‚Äò*‚Äô 0.05 ‚Äò.‚Äô 0.1 ‚Äò ‚Äô 1\n",
      "\n",
      "Residual standard error: 0.5897 on 7510 degrees of freedom\n",
      "Multiple R-squared:  0.000733,\tAdjusted R-squared:  0.0005999 \n",
      "F-statistic: 5.509 on 1 and 7510 DF,  p-value: 0.01895\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "correlation.model <- lm(Nalls_9_17727067_G ~ G276V_9_17793465_T, data = total)\n",
    "summary(correlation.model)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Call:\n",
      "glm(formula = Phenotype ~ Nalls_9_17727067_G + G276V_9_17793465_T, \n",
      "    data = total)\n",
      "\n",
      "Deviance Residuals: \n",
      "    Min       1Q   Median       3Q      Max  \n",
      "-0.5271  -0.4379  -0.4379   0.5621   0.5621  \n",
      "\n",
      "Coefficients:\n",
      "                   Estimate Std. Error t value Pr(>|t|)    \n",
      "(Intercept)        1.437856   0.007195 199.850   <2e-16 ***\n",
      "Nalls_9_17727067_G 0.020089   0.009728   2.065   0.0389 *  \n",
      "G276V_9_17793465_T 0.069100   0.076058   0.909   0.3636    \n",
      "---\n",
      "Signif. codes:  0 ‚Äò***‚Äô 0.001 ‚Äò**‚Äô 0.01 ‚Äò*‚Äô 0.05 ‚Äò.‚Äô 0.1 ‚Äò ‚Äô 1\n",
      "\n",
      "(Dispersion parameter for gaussian family taken to be 0.2471415)\n",
      "\n",
      "    Null deviance: 1857.0  on 7511  degrees of freedom\n",
      "Residual deviance: 1855.8  on 7509  degrees of freedom\n",
      "AIC: 10823\n",
      "\n",
      "Number of Fisher Scoring iterations: 2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "dep_corr<- glm(Phenotype ~ Nalls_9_17727067_G + G276V_9_17793465_T, data=total)\n",
    "summary(dep_corr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                     Estimate  Std. Error     t value   Pr(>|t|)\n",
      "(Intercept)        1.43785556 0.007194687 199.8496335 0.00000000\n",
      "Nalls_9_17727067_G 0.02008945 0.009727724   2.0651750 0.03894069\n",
      "G276V_9_17793465_T 0.06910038 0.076057904   0.9085233 0.36363096\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "summary(dep_corr)$coefficients"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 1.071544\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "exp(0.06910038)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save out results..!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR} {WORKSPACE_BUCKET}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
