{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SH3GL2 - Single gene analysis in GP2 data for SAS population"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Description\n",
    "\n",
    "Using \"individual level data\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 0. Getting Started\n",
    "\n",
    "- Loading Python libraries\n",
    "- Defining functions\n",
    "- Installing packages\n",
    "\n",
    "### 1. Copy data from workspace to cloud environment\n",
    "\n",
    "### 2. Extract SH3GL2\n",
    "\n",
    "### 3. Annotate SH3GL2 variants\n",
    "\n",
    "### 4. Extract coding/non-syn variants\n",
    "\n",
    "### 5. Calculate frequency in cases versus controls\n",
    "\n",
    "### 6. Calculate frequency (homozygotes) in cases versus controls\n",
    "\n",
    "### 7. Save out results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting Started"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Loading Python libraries"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Defining functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "\n",
    "# Numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery\n",
    "\n",
    "#Import Sys\n",
    "import sys as sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "code_folding": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Billing and Workspace\n",
      "Workspace Name: Endophilin-A\n",
      "Billing Project: terra-18d8e41c\n",
      "Workspace Bucket, where you can upload and download data: gs://fc-f7a400c1-827e-48f8-b7b6-90c488a000a4\n",
      "\n",
      "GP2 v5.0\n",
      "Path to GP2 v5.0 Clinical Data: gs://gp2tier2/release5_11052023/clinical_data\n",
      "Path to GP2 v5.0 Raw Genotype Data: gs://gp2tier2/release5_11052023/raw_genotypes\n",
      "Path to GP2 v5.0 Imputed Genotype Data: gs://gp2tier2/release5_11052023/imputed_genotypes\n",
      "AMP-PD v3.0\n",
      "Path to AMP-PD v3.0 Clinical Data: gs://amp-pd-data/releases/2022_v3release_1115/clinical\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "\n",
    "## GP2 v5.0\n",
    "## Explicitly define release v5.0 path \n",
    "GP2_RELEASE_PATH = 'gs://gp2tier2/release5_11052023'\n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_META_RELEASE_PATH = f'{GP2_RELEASE_PATH}/meta_data'\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v5.0')\n",
    "print(f'Path to GP2 v5.0 Clinical Data: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v5.0 Raw Genotype Data: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v5.0 Imputed Genotype Data: {GP2_IMPUTED_GENO_PATH}')\n",
    "\n",
    "\n",
    "## AMP-PD v3.0\n",
    "## Explicitly define release v3.0 path \n",
    "AMP_RELEASE_PATH = 'gs://amp-pd-data/releases/2022_v3release_1115'\n",
    "AMP_CLINICAL_RELEASE_PATH = f'{AMP_RELEASE_PATH}/clinical'\n",
    "\n",
    "AMP_WGS_RELEASE_PATH = 'gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS'\n",
    "AMP_WGS_RELEASE_PLINK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'plink')\n",
    "AMP_WGS_RELEASE_PLINK_PFILES = os.path.join(AMP_WGS_RELEASE_PLINK_PATH, 'pfiles')\n",
    "\n",
    "print('AMP-PD v3.0')\n",
    "print(f'Path to AMP-PD v3.0 Clinical Data: {AMP_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PFILES}')\n",
    "print('')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install Plink 1.9 and Plink 2.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n",
      "Plink2 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi\n",
    "\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/alpha3/plink2_linux_avx2_20220603.zip\n",
    "unzip -o plink2_linux_avx2_20220603.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Remote restrictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 1.9 \n",
    "chmod u+x /home/jupyter/tools/plink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 2.0\n",
    "chmod u+x /home/jupyter/tools/plink2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install ANNOVAR: Download sources of annotation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGene.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneMrna.fa.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneVersion.txt.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotate_variation.pl\n",
      "coding_change.pl\n",
      "convert2annovar.pl\n",
      "example\n",
      "humandb\n",
      "retrieve_seq_from_fasta.pl\n",
      "table_annovar.pl\n",
      "variants_reduction.pl\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/annovar/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Copy data from GP2 bucket to workspace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Making a working directory\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: mkdir -p /home/jupyter/SH3GL2_GP2/\n"
     ]
    }
   ],
   "source": [
    "# Make a directory\n",
    "print(\"Making a working directory\")\n",
    "WORK_DIR = f'/home/jupyter/SH3GL2_GP2/' \n",
    "shell_do(f'mkdir -p {WORK_DIR}') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-18d8e41c ls gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr10_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr10_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr10_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr10_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr10_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr10_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr10_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr10_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr11_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr11_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr11_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr11_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr11_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr11_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr11_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr11_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr12_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr12_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr12_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr12_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr12_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr12_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr12_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr12_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr13_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr13_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr13_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr13_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr13_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr13_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr13_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr13_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr14_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr14_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr14_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr14_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr14_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr14_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr14_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr14_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr15_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr15_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr15_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr15_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr15_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr15_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr15_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr15_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr16_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr16_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr16_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr16_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr16_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr16_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr16_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr16_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr17_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr17_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr17_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr17_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr17_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr17_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr17_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr17_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr18_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr18_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr18_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr18_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr18_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr18_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr18_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr18_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr19_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr19_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr19_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr19_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr19_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr19_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr19_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr19_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr1_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr1_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr1_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr1_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr1_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr1_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr1_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr1_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr20_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr20_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr20_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr20_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr20_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr20_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr20_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr20_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr21_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr21_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr21_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr21_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr21_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr21_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr21_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr21_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr22_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr22_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr22_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr22_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr22_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr22_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr22_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr22_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr2_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr2_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr2_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr2_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr2_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr2_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr2_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr2_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr3_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr3_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr3_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr3_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr3_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr3_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr3_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr3_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr4_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr4_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr4_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr4_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr4_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr4_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr4_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr4_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr5_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr5_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr5_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr5_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr5_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr5_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr5_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr5_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr6_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr6_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr6_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr6_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr6_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr6_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr6_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr6_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr7_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr7_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr7_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr7_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr7_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr7_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr7_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr7_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr8_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr8_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr8_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr8_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr8_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr8_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr8_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr8_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr9_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr9_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr9_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr9_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr9_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr9_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr9_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr9_SAS_release5_no_mac_filter.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chrX_SAS_release5.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chrX_SAS_release5.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chrX_SAS_release5.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chrX_SAS_release5.pvar\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chrX_SAS_release5_no_mac_filter.log\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chrX_SAS_release5_no_mac_filter.pgen\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chrX_SAS_release5_no_mac_filter.psam\r\n",
      "gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chrX_SAS_release5_no_mac_filter.pvar\r\n"
     ]
    }
   ],
   "source": [
    "# Check directory where GP2 data is\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_IMPUTED_GENO_PATH}/SAS/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-18d8e41c -m cp -r gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr9_SAS_release5_no_mac_filter* /home/jupyter/SH3GL2_GP2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr9_SAS_release5_no_mac_filter.psam...\n",
      "Copying gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr9_SAS_release5_no_mac_filter.pgen...\n",
      "Copying gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr9_SAS_release5_no_mac_filter.pvar...\n",
      "Copying gs://gp2tier2/release5_11052023/imputed_genotypes/SAS/chr9_SAS_release5_no_mac_filter.log...\n",
      "\\ [4/4 files][786.2 MiB/786.2 MiB] 100% Done  51.7 MiB/s ETA 00:00:00           \n",
      "Operation completed over 4 objects/786.2 MiB.                                    \n"
     ]
    }
   ],
   "source": [
    "#Copying data for SAS population \n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {GP2_IMPUTED_GENO_PATH}/SAS/chr9_SAS_release5_no_mac_filter* {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create a covariate file with GP2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-18d8e41c ls gs://gp2tier2/release5_11052023/clinical_data\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://gp2tier2/release5_11052023/clinical_data/master_key_release5_final.csv\r\n",
      "gs://gp2tier2/release5_11052023/clinical_data/release5_data_dictionary.csv\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-18d8e41c -m cp -r gs://gp2tier2/release5_11052023/clinical_data/master_key_release5_final.csv /home/jupyter/SH3GL2_GP2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://gp2tier2/release5_11052023/clinical_data/master_key_release5_final.csv...\n",
      "/ [1/1 files][  2.5 MiB/  2.5 MiB] 100% Done                                    \n",
      "Operation completed over 1 objects/2.5 MiB.                                      \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_CLINICAL_RELEASE_PATH}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {GP2_CLINICAL_RELEASE_PATH}/master_key_release5_final.csv {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['GP2ID', 'GP2sampleID', 'manifest_id', 'phenotype', 'pheno_for_qc',\n",
       "       'other_pheno', 'sex_for_qc', 'age', 'age_of_onset', 'age_at_diagnosis',\n",
       "       'age_at_death', 'race_for_qc', 'family_history', 'region_for_qc',\n",
       "       'study', 'pruned', 'pruned_reason', 'label', 'related'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov = pd.read_csv(f'{WORK_DIR}/master_key_release5_final.csv')\n",
    "cov.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "cov_reduced = cov[['GP2sampleID','GP2sampleID','sex_for_qc', 'age', 'phenotype']]\n",
    "cov_reduced.columns = ['FID','IID', 'SEX','AGE','PHENO']\n",
    "#cov_reduced.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "conditions = [\n",
    "    (cov_reduced['PHENO'] == \"PD\"),\n",
    "    (cov_reduced['PHENO'] == \"Control\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "choices = [2,1]\n",
    "cov_reduced['PHENOTYPE'] = np.select(conditions, choices, default=-9).astype(np.int64)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.7/site-packages/pandas/core/frame.py:4913: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n"
     ]
    }
   ],
   "source": [
    "cov_reduced.reset_index(inplace=True)\n",
    "cov_reduced.drop(columns=[\"index\"], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "cov_reduced.drop(columns=['PHENO'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "sex = cov_reduced[['FID','IID','SEX']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "sex.to_csv(f'{WORK_DIR}/SEX.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno = cov_reduced[['FID','IID','PHENOTYPE']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno.to_csv(f'{WORK_DIR}/PHENO.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract SH3GL2 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a3.3LM AVX2 Intel (3 Jun 2022)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_SH3GL2_SAS.log.\n",
      "Options in effect:\n",
      "  --chr 9\n",
      "  --from-bp 17579066\n",
      "  --make-bed\n",
      "  --out pheno_SH3GL2_SAS\n",
      "  --pfile chr9_SAS_release5_no_mac_filter\n",
      "  --to-bp 17797124\n",
      "\n",
      "Start time: Thu Nov 16 09:18:18 2023\n",
      "7450 MiB RAM detected; reserving 3725 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "220 samples (98 females, 122 males; 220 founders) loaded from\n",
      "chr9_SAS_release5_no_mac_filter.psam.\n",
      "7736231 variants loaded from chr9_SAS_release5_no_mac_filter.pvar.\n",
      "1 binary phenotype loaded (80 cases, 135 controls).\n",
      "20913 variants remaining after main filters.\n",
      "Writing pheno_SH3GL2_SAS.fam ... done.\n",
      "Writing pheno_SH3GL2_SAS.bim ... done.\n",
      "Writing pheno_SH3GL2_SAS.bed ... 0%\b\bdone.\n",
      "End time: Thu Nov 16 09:18:19 2023\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "#For SAS\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# SH3GL2: gene positions on hg38 (from https://useast.ensembl.org/index.html)\n",
    "/home/jupyter/tools/plink2 \\\n",
    "--pfile chr9_SAS_release5_no_mac_filter \\\n",
    "--chr 9 \\\n",
    "--from-bp 17579066  \\\n",
    "--to-bp 17797124 \\\n",
    "--make-bed \\\n",
    "--out pheno_SH3GL2_SAS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualize plink files bim, fam and bed  and turn binary files to vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\tchr9:17579083:C:T\t0\t17579083\tT\tC\n",
      "9\tchr9:17579085:A:G\t0\t17579085\tG\tA\n",
      "9\tchr9:17579090:C:A\t0\t17579090\tA\tC\n",
      "9\tchr9:17579091:C:A\t0\t17579091\tA\tC\n",
      "9\tchr9:17579092:G:A\t0\t17579092\tA\tG\n",
      "9\tchr9:17579096:C:T\t0\t17579096\tT\tC\n",
      "9\tchr9:17579097:C:T\t0\t17579097\tT\tC\n",
      "9\tchr9:17579098:C:T\t0\t17579098\tT\tC\n",
      "9\tchr9:17579105:C:G\t0\t17579105\tG\tC\n",
      "9\tchr9:17579105:C:T\t0\t17579105\tT\tC\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize bim file\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "\n",
    "head pheno_SH3GL2_SAS.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\tAPGS_000062_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000119_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000299_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000331_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000378_s1\t0\t0\t2\t2\n",
      "0\tAPGS_000416_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000454_s1\t0\t0\t2\t2\n",
      "0\tAPGS_000502_s1\t0\t0\t2\t2\n",
      "0\tAPGS_000526_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000604_s1\t0\t0\t2\t2\n",
      "220 pheno_SH3GL2_SAS.fam\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize fam file\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_SH3GL2_SAS.fam\n",
    "\n",
    "#Count lines temp.fam file\n",
    "wc -l pheno_SH3GL2_SAS.fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_SH3GL2_SAS_nomacfilter.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_SH3GL2_SAS\n",
      "  --out pheno_SH3GL2_SAS_nomacfilter\n",
      "  --recode vcf-fid\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "20913 variants loaded from .bim file.\n",
      "220 people (122 males, 98 females) loaded from .fam.\n",
      "215 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 220 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.998258.\n",
      "20913 variants and 220 people pass filters and QC.\n",
      "Among remaining phenotypes, 80 are cases and 135 are controls.  (5 phenotypes\n",
      "are missing.)\n",
      "--recode vcf-fid to pheno_SH3GL2_SAS_nomacfilter.vcf ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# Turn binary files into VCF for SAS\n",
    "/home/jupyter/tools/plink \\\n",
    "--bfile pheno_SH3GL2_SAS \\\n",
    "--recode vcf-fid \\\n",
    "--out pheno_SH3GL2_SAS_nomacfilter"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate SH3GL2 variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#for SAS\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "export PATH=$PATH:/home/jupyter/tools/rvtests/third/tabix-0.2.6/\n",
    "\n",
    "### Bgzip and Tabix\n",
    "bgzip pheno_SH3GL2_SAS_nomacfilter.vcf\n",
    "\n",
    "tabix -f -p vcf pheno_SH3GL2_SAS_nomacfilter.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 pheno_SH3GL2_SAS_nomacfilter.vcf.gz > pheno_SH3GL2_SAS.annovar.avinput>\n",
      "NOTICE: Finished reading 20920 lines from VCF file\n",
      "NOTICE: A total of 20913 locus in VCF file passed QC threshold, representing 19709 SNPs (11614 transitions and 8095 transversions) and 1204 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 4335980 SNP genotypes (2555080 transitions and 1780900 transversions) and 264880 indels/substitutions for 220 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl pheno_SH3GL2_SAS.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile pheno_SH3GL2_SAS.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile pheno_SH3GL2_SAS.annovar.refGene -exonsort -nofirstcodondel pheno_SH3GL2_SAS.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to pheno_SH3GL2_SAS.annovar.refGene.variant_function, pheno_SH3GL2_SAS.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 20913 unique variants in 20913 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile pheno_SH3GL2_SAS.annovar pheno_SH3GL2_SAS.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to pheno_SH3GL2_SAS.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to pheno_SH3GL2_SAS.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 20913 unique variants in 20913 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 1\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to pheno_SH3GL2_SAS.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from pheno_SH3GL2_SAS.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to pheno_SH3GL2_SAS.annovar.hg38_multianno.vcf\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "#For SAS\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "### Annotate variants using ANNOVAR: https://annovar.openbioinformatics.org/en/latest/ \n",
    "perl /home/jupyter/tools/annovar/table_annovar.pl pheno_SH3GL2_SAS_nomacfilter.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out pheno_SH3GL2_SAS.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract coding and non-syn variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Visualize multianno file for SAS\n",
    "SH3GL2_SAS = pd.read_csv(f'{WORK_DIR}/pheno_SH3GL2_SAS.annovar.hg38_multianno.txt', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "SH3GL2_SAS['Chr_bp'] = SH3GL2_SAS['Chr'].astype(str) + ':' + SH3GL2_SAS['Start'].astype(str) + ':' + SH3GL2_SAS['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "SH3GL2_SAS = SH3GL2_SAS.drop_duplicates(subset=['Chr_bp'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "SH3GL2_SAS = SH3GL2_SAS[SH3GL2_SAS['Gene.refGene']=='SH3GL2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr             20252\n",
       "Start           20252\n",
       "End             20252\n",
       "Ref             20252\n",
       "Alt             20252\n",
       "                ...  \n",
       "Otherinfo229    20252\n",
       "Otherinfo230    20252\n",
       "Otherinfo231    20252\n",
       "Otherinfo232    20252\n",
       "Chr_bp          20252\n",
       "Length: 244, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_SAS.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr_bp</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9:17579083:17579083</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9:17579085:17579085</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9:17579090:17579090</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9:17579091:17579091</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9:17579092:17579092</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20907</th>\n",
       "      <td>9:17797040:17797040</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20908</th>\n",
       "      <td>9:17797047:17797047</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20909</th>\n",
       "      <td>9:17797083:17797083</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20911</th>\n",
       "      <td>9:17797092:17797092</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20912</th>\n",
       "      <td>9:17797093:17797093</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20252 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Chr_bp Gene.refGene Func.refGene ExonicFunc.refGene\n",
       "0      9:17579083:17579083       SH3GL2         UTR5                  .\n",
       "1      9:17579085:17579085       SH3GL2         UTR5                  .\n",
       "2      9:17579090:17579090       SH3GL2         UTR5                  .\n",
       "3      9:17579091:17579091       SH3GL2         UTR5                  .\n",
       "4      9:17579092:17579092       SH3GL2         UTR5                  .\n",
       "...                    ...          ...          ...                ...\n",
       "20907  9:17797040:17797040       SH3GL2         UTR3                  .\n",
       "20908  9:17797047:17797047       SH3GL2         UTR3                  .\n",
       "20909  9:17797083:17797083       SH3GL2         UTR3                  .\n",
       "20911  9:17797092:17797092       SH3GL2         UTR3                  .\n",
       "20912  9:17797093:17797093       SH3GL2         UTR3                  .\n",
       "\n",
       "[20252 rows x 4 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reduced_SH3GL2_SAS = SH3GL2_SAS[[\"Chr_bp\", \"Gene.refGene\", 'Func.refGene','ExonicFunc.refGene']]\n",
    "reduced_SH3GL2_SAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "reduced_SH3GL2_SAS.to_csv(f'{WORKSPACE_BUCKET}/reduced_SH3GL2_SAS.txt', sep = '\\t', index = False, header= True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "intronic    20045\n",
       "UTR3          120\n",
       "exonic         49\n",
       "UTR5           38\n",
       "Name: Func.refGene, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_SAS['Func.refGene'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       ".                    20203\n",
       "nonsynonymous SNV       32\n",
       "synonymous SNV          17\n",
       "Name: ExonicFunc.refGene, dtype: int64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_SAS['ExonicFunc.refGene'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo224</th>\n",
       "      <th>Otherinfo225</th>\n",
       "      <th>Otherinfo226</th>\n",
       "      <th>Otherinfo227</th>\n",
       "      <th>Otherinfo228</th>\n",
       "      <th>Otherinfo229</th>\n",
       "      <th>Otherinfo230</th>\n",
       "      <th>Otherinfo231</th>\n",
       "      <th>Otherinfo232</th>\n",
       "      <th>Chr_bp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>9</td>\n",
       "      <td>17579258</td>\n",
       "      <td>17579258</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon1:c.C16T:p.L6F</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17579258:17579258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16099</th>\n",
       "      <td>9</td>\n",
       "      <td>17747115</td>\n",
       "      <td>17747115</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon2:c.A95C:p.D32A</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17747115:17747115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17521</th>\n",
       "      <td>9</td>\n",
       "      <td>17761460</td>\n",
       "      <td>17761460</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon3:c.T138C:p.A46A</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17761460:17761460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17522</th>\n",
       "      <td>9</td>\n",
       "      <td>17761487</td>\n",
       "      <td>17761487</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon3:c.T165A:p.I55I</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17761487:17761487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17523</th>\n",
       "      <td>9</td>\n",
       "      <td>17761500</td>\n",
       "      <td>17761500</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon3:c.C178G:p.P60A</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17761500:17761500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19899</th>\n",
       "      <td>9</td>\n",
       "      <td>17786430</td>\n",
       "      <td>17786430</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon4:c.C237T:p.G79G</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17786430:17786430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19900</th>\n",
       "      <td>9</td>\n",
       "      <td>17786465</td>\n",
       "      <td>17786465</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon4:c.C272T:p.A91V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17786465:17786465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19901</th>\n",
       "      <td>9</td>\n",
       "      <td>17786478</td>\n",
       "      <td>17786478</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon4:c.G285C:p.E95D</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17786478:17786478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19902</th>\n",
       "      <td>9</td>\n",
       "      <td>17786519</td>\n",
       "      <td>17786519</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon4:c.A326G:p.N109S</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17786519:17786519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20000</th>\n",
       "      <td>9</td>\n",
       "      <td>17787399</td>\n",
       "      <td>17787399</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon5:c.C351T:p.V117V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17787399:17787399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20001</th>\n",
       "      <td>9</td>\n",
       "      <td>17787423</td>\n",
       "      <td>17787423</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon5:c.G375A:p.S125S</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17787423:17787423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20002</th>\n",
       "      <td>9</td>\n",
       "      <td>17787501</td>\n",
       "      <td>17787501</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon5:c.T453C:p.L151L</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17787501:17787501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20003</th>\n",
       "      <td>9</td>\n",
       "      <td>17787509</td>\n",
       "      <td>17787509</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon5:c.T461C:p.I154T</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17787509:17787509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20178</th>\n",
       "      <td>9</td>\n",
       "      <td>17789398</td>\n",
       "      <td>17789398</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.C472T:p.L158L</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789398:17789398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20179</th>\n",
       "      <td>9</td>\n",
       "      <td>17789421</td>\n",
       "      <td>17789421</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.C495T:p.R165R</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789421:17789421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20180</th>\n",
       "      <td>9</td>\n",
       "      <td>17789425</td>\n",
       "      <td>17789425</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.G499T:p.D167Y</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789425:17789425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20181</th>\n",
       "      <td>9</td>\n",
       "      <td>17789473</td>\n",
       "      <td>17789473</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.C547G:p.L183V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789473:17789473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20182</th>\n",
       "      <td>9</td>\n",
       "      <td>17789476</td>\n",
       "      <td>17789476</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.C550T:p.R184C</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789476:17789476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20183</th>\n",
       "      <td>9</td>\n",
       "      <td>17789477</td>\n",
       "      <td>17789477</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.G551A:p.R184H</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789477:17789477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20184</th>\n",
       "      <td>9</td>\n",
       "      <td>17789485</td>\n",
       "      <td>17789485</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.C559G:p.L187V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>9:17789485:17789485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20185</th>\n",
       "      <td>9</td>\n",
       "      <td>17789499</td>\n",
       "      <td>17789499</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.T573G:p.D191E</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789499:17789499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20186</th>\n",
       "      <td>9</td>\n",
       "      <td>17789544</td>\n",
       "      <td>17789544</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.G618T:p.E206D</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789544:17789544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20187</th>\n",
       "      <td>9</td>\n",
       "      <td>17789545</td>\n",
       "      <td>17789545</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon6:c.A619G:p.M207V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17789545:17789545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20348</th>\n",
       "      <td>9</td>\n",
       "      <td>17791272</td>\n",
       "      <td>17791272</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.G666C:p.Q222H</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791272:17791272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20349</th>\n",
       "      <td>9</td>\n",
       "      <td>17791276</td>\n",
       "      <td>17791276</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.G670C:p.E224Q</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791276:17791276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20350</th>\n",
       "      <td>9</td>\n",
       "      <td>17791308</td>\n",
       "      <td>17791308</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.G702C:p.Q234H</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791308:17791308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20351</th>\n",
       "      <td>9</td>\n",
       "      <td>17791311</td>\n",
       "      <td>17791311</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.A705G:p.Q235Q</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791311:17791311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20352</th>\n",
       "      <td>9</td>\n",
       "      <td>17791317</td>\n",
       "      <td>17791317</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.G711A:p.T237T</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791317:17791317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20353</th>\n",
       "      <td>9</td>\n",
       "      <td>17791318</td>\n",
       "      <td>17791318</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon7:c.G712A:p.V238I</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17791318:17791318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20548</th>\n",
       "      <td>9</td>\n",
       "      <td>17793385</td>\n",
       "      <td>17793385</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.T747C:p.S249S</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793385:17793385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20549</th>\n",
       "      <td>9</td>\n",
       "      <td>17793386</td>\n",
       "      <td>17793386</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.C748G:p.Q250E</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793386:17793386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20550</th>\n",
       "      <td>9</td>\n",
       "      <td>17793389</td>\n",
       "      <td>17793389</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.C751G:p.P251A</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793389:17793389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20551</th>\n",
       "      <td>9</td>\n",
       "      <td>17793394</td>\n",
       "      <td>17793394</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.A756G:p.R252R</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793394:17793394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20552</th>\n",
       "      <td>9</td>\n",
       "      <td>17793400</td>\n",
       "      <td>17793400</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.A762G:p.E254E</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793400:17793400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20553</th>\n",
       "      <td>9</td>\n",
       "      <td>17793407</td>\n",
       "      <td>17793407</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.C769T:p.P257S</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793407:17793407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20554</th>\n",
       "      <td>9</td>\n",
       "      <td>17793437</td>\n",
       "      <td>17793437</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.A799G:p.T267A</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793437:17793437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20555</th>\n",
       "      <td>9</td>\n",
       "      <td>17793465</td>\n",
       "      <td>17793465</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.G827T:p.G276V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793465:17793465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20556</th>\n",
       "      <td>9</td>\n",
       "      <td>17793474</td>\n",
       "      <td>17793474</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.A836T:p.H279L</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793474:17793474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20557</th>\n",
       "      <td>9</td>\n",
       "      <td>17793477</td>\n",
       "      <td>17793477</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.C839T:p.T280I</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793477:17793477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20558</th>\n",
       "      <td>9</td>\n",
       "      <td>17793481</td>\n",
       "      <td>17793481</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon8:c.C843G:p.G281G</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17793481:17793481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20780</th>\n",
       "      <td>9</td>\n",
       "      <td>17795545</td>\n",
       "      <td>17795545</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.T861A:p.G287G</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795545:17795545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20781</th>\n",
       "      <td>9</td>\n",
       "      <td>17795552</td>\n",
       "      <td>17795552</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.A868G:p.M290V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795552:17795552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20782</th>\n",
       "      <td>9</td>\n",
       "      <td>17795560</td>\n",
       "      <td>17795560</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.G876T:p.Q292H</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795560:17795560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20783</th>\n",
       "      <td>9</td>\n",
       "      <td>17795584</td>\n",
       "      <td>17795584</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.C900T:p.D300D</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795584:17795584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20784</th>\n",
       "      <td>9</td>\n",
       "      <td>17795626</td>\n",
       "      <td>17795626</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.C942T:p.G314G</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795626:17795626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20785</th>\n",
       "      <td>9</td>\n",
       "      <td>17795643</td>\n",
       "      <td>17795643</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.C959G:p.T320S</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795643:17795643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20786</th>\n",
       "      <td>9</td>\n",
       "      <td>17795649</td>\n",
       "      <td>17795649</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.A965G:p.Q322R</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795649:17795649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20787</th>\n",
       "      <td>9</td>\n",
       "      <td>17795682</td>\n",
       "      <td>17795682</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.A998G:p.H333R</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795682:17795682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20788</th>\n",
       "      <td>9</td>\n",
       "      <td>17795685</td>\n",
       "      <td>17795685</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon9:c.G1001A:p.G334D</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17795685:17795685</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>49 rows × 244 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chr     Start       End Ref Alt Func.refGene Gene.refGene  \\\n",
       "43       9  17579258  17579258   C   T       exonic       SH3GL2   \n",
       "16099    9  17747115  17747115   A   C       exonic       SH3GL2   \n",
       "17521    9  17761460  17761460   T   C       exonic       SH3GL2   \n",
       "17522    9  17761487  17761487   T   A       exonic       SH3GL2   \n",
       "17523    9  17761500  17761500   C   G       exonic       SH3GL2   \n",
       "19899    9  17786430  17786430   C   T       exonic       SH3GL2   \n",
       "19900    9  17786465  17786465   C   T       exonic       SH3GL2   \n",
       "19901    9  17786478  17786478   G   C       exonic       SH3GL2   \n",
       "19902    9  17786519  17786519   A   G       exonic       SH3GL2   \n",
       "20000    9  17787399  17787399   C   T       exonic       SH3GL2   \n",
       "20001    9  17787423  17787423   G   A       exonic       SH3GL2   \n",
       "20002    9  17787501  17787501   T   C       exonic       SH3GL2   \n",
       "20003    9  17787509  17787509   T   C       exonic       SH3GL2   \n",
       "20178    9  17789398  17789398   C   T       exonic       SH3GL2   \n",
       "20179    9  17789421  17789421   C   T       exonic       SH3GL2   \n",
       "20180    9  17789425  17789425   G   T       exonic       SH3GL2   \n",
       "20181    9  17789473  17789473   C   G       exonic       SH3GL2   \n",
       "20182    9  17789476  17789476   C   T       exonic       SH3GL2   \n",
       "20183    9  17789477  17789477   G   A       exonic       SH3GL2   \n",
       "20184    9  17789485  17789485   C   G       exonic       SH3GL2   \n",
       "20185    9  17789499  17789499   T   G       exonic       SH3GL2   \n",
       "20186    9  17789544  17789544   G   T       exonic       SH3GL2   \n",
       "20187    9  17789545  17789545   A   G       exonic       SH3GL2   \n",
       "20348    9  17791272  17791272   G   C       exonic       SH3GL2   \n",
       "20349    9  17791276  17791276   G   C       exonic       SH3GL2   \n",
       "20350    9  17791308  17791308   G   C       exonic       SH3GL2   \n",
       "20351    9  17791311  17791311   A   G       exonic       SH3GL2   \n",
       "20352    9  17791317  17791317   G   A       exonic       SH3GL2   \n",
       "20353    9  17791318  17791318   G   A       exonic       SH3GL2   \n",
       "20548    9  17793385  17793385   T   C       exonic       SH3GL2   \n",
       "20549    9  17793386  17793386   C   G       exonic       SH3GL2   \n",
       "20550    9  17793389  17793389   C   G       exonic       SH3GL2   \n",
       "20551    9  17793394  17793394   A   G       exonic       SH3GL2   \n",
       "20552    9  17793400  17793400   A   G       exonic       SH3GL2   \n",
       "20553    9  17793407  17793407   C   T       exonic       SH3GL2   \n",
       "20554    9  17793437  17793437   A   G       exonic       SH3GL2   \n",
       "20555    9  17793465  17793465   G   T       exonic       SH3GL2   \n",
       "20556    9  17793474  17793474   A   T       exonic       SH3GL2   \n",
       "20557    9  17793477  17793477   C   T       exonic       SH3GL2   \n",
       "20558    9  17793481  17793481   C   G       exonic       SH3GL2   \n",
       "20780    9  17795545  17795545   T   A       exonic       SH3GL2   \n",
       "20781    9  17795552  17795552   A   G       exonic       SH3GL2   \n",
       "20782    9  17795560  17795560   G   T       exonic       SH3GL2   \n",
       "20783    9  17795584  17795584   C   T       exonic       SH3GL2   \n",
       "20784    9  17795626  17795626   C   T       exonic       SH3GL2   \n",
       "20785    9  17795643  17795643   C   G       exonic       SH3GL2   \n",
       "20786    9  17795649  17795649   A   G       exonic       SH3GL2   \n",
       "20787    9  17795682  17795682   A   G       exonic       SH3GL2   \n",
       "20788    9  17795685  17795685   G   A       exonic       SH3GL2   \n",
       "\n",
       "      GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "43                     .  nonsynonymous SNV   \n",
       "16099                  .  nonsynonymous SNV   \n",
       "17521                  .     synonymous SNV   \n",
       "17522                  .     synonymous SNV   \n",
       "17523                  .  nonsynonymous SNV   \n",
       "19899                  .     synonymous SNV   \n",
       "19900                  .  nonsynonymous SNV   \n",
       "19901                  .  nonsynonymous SNV   \n",
       "19902                  .  nonsynonymous SNV   \n",
       "20000                  .     synonymous SNV   \n",
       "20001                  .     synonymous SNV   \n",
       "20002                  .     synonymous SNV   \n",
       "20003                  .  nonsynonymous SNV   \n",
       "20178                  .     synonymous SNV   \n",
       "20179                  .     synonymous SNV   \n",
       "20180                  .  nonsynonymous SNV   \n",
       "20181                  .  nonsynonymous SNV   \n",
       "20182                  .  nonsynonymous SNV   \n",
       "20183                  .  nonsynonymous SNV   \n",
       "20184                  .  nonsynonymous SNV   \n",
       "20185                  .  nonsynonymous SNV   \n",
       "20186                  .  nonsynonymous SNV   \n",
       "20187                  .  nonsynonymous SNV   \n",
       "20348                  .  nonsynonymous SNV   \n",
       "20349                  .  nonsynonymous SNV   \n",
       "20350                  .  nonsynonymous SNV   \n",
       "20351                  .     synonymous SNV   \n",
       "20352                  .     synonymous SNV   \n",
       "20353                  .  nonsynonymous SNV   \n",
       "20548                  .     synonymous SNV   \n",
       "20549                  .  nonsynonymous SNV   \n",
       "20550                  .  nonsynonymous SNV   \n",
       "20551                  .     synonymous SNV   \n",
       "20552                  .     synonymous SNV   \n",
       "20553                  .  nonsynonymous SNV   \n",
       "20554                  .  nonsynonymous SNV   \n",
       "20555                  .  nonsynonymous SNV   \n",
       "20556                  .  nonsynonymous SNV   \n",
       "20557                  .  nonsynonymous SNV   \n",
       "20558                  .     synonymous SNV   \n",
       "20780                  .     synonymous SNV   \n",
       "20781                  .  nonsynonymous SNV   \n",
       "20782                  .  nonsynonymous SNV   \n",
       "20783                  .     synonymous SNV   \n",
       "20784                  .     synonymous SNV   \n",
       "20785                  .  nonsynonymous SNV   \n",
       "20786                  .  nonsynonymous SNV   \n",
       "20787                  .  nonsynonymous SNV   \n",
       "20788                  .  nonsynonymous SNV   \n",
       "\n",
       "                              AAChange.refGene  ... Otherinfo224  \\\n",
       "43         SH3GL2:NM_003026:exon1:c.C16T:p.L6F  ...          0/0   \n",
       "16099     SH3GL2:NM_003026:exon2:c.A95C:p.D32A  ...          0/0   \n",
       "17521    SH3GL2:NM_003026:exon3:c.T138C:p.A46A  ...          0/0   \n",
       "17522    SH3GL2:NM_003026:exon3:c.T165A:p.I55I  ...          0/0   \n",
       "17523    SH3GL2:NM_003026:exon3:c.C178G:p.P60A  ...          0/0   \n",
       "19899    SH3GL2:NM_003026:exon4:c.C237T:p.G79G  ...          0/0   \n",
       "19900    SH3GL2:NM_003026:exon4:c.C272T:p.A91V  ...          0/0   \n",
       "19901    SH3GL2:NM_003026:exon4:c.G285C:p.E95D  ...          0/0   \n",
       "19902   SH3GL2:NM_003026:exon4:c.A326G:p.N109S  ...          0/0   \n",
       "20000   SH3GL2:NM_003026:exon5:c.C351T:p.V117V  ...          0/0   \n",
       "20001   SH3GL2:NM_003026:exon5:c.G375A:p.S125S  ...          0/0   \n",
       "20002   SH3GL2:NM_003026:exon5:c.T453C:p.L151L  ...          0/0   \n",
       "20003   SH3GL2:NM_003026:exon5:c.T461C:p.I154T  ...          0/0   \n",
       "20178   SH3GL2:NM_003026:exon6:c.C472T:p.L158L  ...          0/0   \n",
       "20179   SH3GL2:NM_003026:exon6:c.C495T:p.R165R  ...          0/0   \n",
       "20180   SH3GL2:NM_003026:exon6:c.G499T:p.D167Y  ...          0/0   \n",
       "20181   SH3GL2:NM_003026:exon6:c.C547G:p.L183V  ...          0/0   \n",
       "20182   SH3GL2:NM_003026:exon6:c.C550T:p.R184C  ...          0/0   \n",
       "20183   SH3GL2:NM_003026:exon6:c.G551A:p.R184H  ...          0/0   \n",
       "20184   SH3GL2:NM_003026:exon6:c.C559G:p.L187V  ...          0/0   \n",
       "20185   SH3GL2:NM_003026:exon6:c.T573G:p.D191E  ...          0/0   \n",
       "20186   SH3GL2:NM_003026:exon6:c.G618T:p.E206D  ...          0/0   \n",
       "20187   SH3GL2:NM_003026:exon6:c.A619G:p.M207V  ...          0/0   \n",
       "20348   SH3GL2:NM_003026:exon7:c.G666C:p.Q222H  ...          0/0   \n",
       "20349   SH3GL2:NM_003026:exon7:c.G670C:p.E224Q  ...          0/0   \n",
       "20350   SH3GL2:NM_003026:exon7:c.G702C:p.Q234H  ...          0/0   \n",
       "20351   SH3GL2:NM_003026:exon7:c.A705G:p.Q235Q  ...          0/0   \n",
       "20352   SH3GL2:NM_003026:exon7:c.G711A:p.T237T  ...          0/0   \n",
       "20353   SH3GL2:NM_003026:exon7:c.G712A:p.V238I  ...          0/0   \n",
       "20548   SH3GL2:NM_003026:exon8:c.T747C:p.S249S  ...          0/0   \n",
       "20549   SH3GL2:NM_003026:exon8:c.C748G:p.Q250E  ...          0/0   \n",
       "20550   SH3GL2:NM_003026:exon8:c.C751G:p.P251A  ...          0/0   \n",
       "20551   SH3GL2:NM_003026:exon8:c.A756G:p.R252R  ...          0/0   \n",
       "20552   SH3GL2:NM_003026:exon8:c.A762G:p.E254E  ...          0/0   \n",
       "20553   SH3GL2:NM_003026:exon8:c.C769T:p.P257S  ...          0/0   \n",
       "20554   SH3GL2:NM_003026:exon8:c.A799G:p.T267A  ...          0/0   \n",
       "20555   SH3GL2:NM_003026:exon8:c.G827T:p.G276V  ...          0/0   \n",
       "20556   SH3GL2:NM_003026:exon8:c.A836T:p.H279L  ...          0/0   \n",
       "20557   SH3GL2:NM_003026:exon8:c.C839T:p.T280I  ...          0/0   \n",
       "20558   SH3GL2:NM_003026:exon8:c.C843G:p.G281G  ...          0/0   \n",
       "20780   SH3GL2:NM_003026:exon9:c.T861A:p.G287G  ...          0/0   \n",
       "20781   SH3GL2:NM_003026:exon9:c.A868G:p.M290V  ...          0/0   \n",
       "20782   SH3GL2:NM_003026:exon9:c.G876T:p.Q292H  ...          0/0   \n",
       "20783   SH3GL2:NM_003026:exon9:c.C900T:p.D300D  ...          0/0   \n",
       "20784   SH3GL2:NM_003026:exon9:c.C942T:p.G314G  ...          0/0   \n",
       "20785   SH3GL2:NM_003026:exon9:c.C959G:p.T320S  ...          0/0   \n",
       "20786   SH3GL2:NM_003026:exon9:c.A965G:p.Q322R  ...          0/0   \n",
       "20787   SH3GL2:NM_003026:exon9:c.A998G:p.H333R  ...          0/0   \n",
       "20788  SH3GL2:NM_003026:exon9:c.G1001A:p.G334D  ...          0/0   \n",
       "\n",
       "       Otherinfo225 Otherinfo226 Otherinfo227  Otherinfo228  Otherinfo229  \\\n",
       "43              0/0          0/0          0/0           0/0           0/0   \n",
       "16099           0/0          0/0          0/0           0/0           0/0   \n",
       "17521           0/0          0/0          0/0           0/0           0/0   \n",
       "17522           0/0          0/0          0/0           0/0           0/0   \n",
       "17523           0/0          0/0          0/0           0/0           0/0   \n",
       "19899           0/0          0/0          0/0           0/0           0/0   \n",
       "19900           0/0          0/0          0/0           0/0           0/0   \n",
       "19901           0/0          0/0          0/0           0/0           0/0   \n",
       "19902           0/0          0/0          0/0           0/0           0/0   \n",
       "20000           0/0          0/0          0/0           0/0           0/0   \n",
       "20001           0/0          0/0          0/0           0/0           0/0   \n",
       "20002           0/0          0/0          0/0           0/0           0/0   \n",
       "20003           0/0          0/0          0/0           0/0           0/0   \n",
       "20178           0/0          0/0          0/0           0/0           0/0   \n",
       "20179           0/0          0/0          0/0           0/0           0/0   \n",
       "20180           0/0          0/0          0/0           0/0           0/0   \n",
       "20181           0/0          0/0          0/0           0/0           0/0   \n",
       "20182           0/0          0/0          0/0           0/0           0/0   \n",
       "20183           0/0          0/0          0/0           0/0           0/0   \n",
       "20184           0/0          0/0          0/0           0/0           0/0   \n",
       "20185           0/0          0/0          0/0           0/0           0/0   \n",
       "20186           0/0          0/0          0/0           0/0           0/0   \n",
       "20187           0/0          0/0          0/0           0/0           0/0   \n",
       "20348           0/0          0/0          0/0           0/0           0/0   \n",
       "20349           0/0          0/0          0/0           0/0           0/0   \n",
       "20350           0/0          0/0          0/0           0/0           0/0   \n",
       "20351           0/0          0/0          0/0           0/0           0/0   \n",
       "20352           0/0          0/0          0/0           0/0           0/0   \n",
       "20353           0/0          0/0          0/0           0/0           0/0   \n",
       "20548           0/0          0/0          0/0           0/0           0/0   \n",
       "20549           0/0          0/0          0/0           0/0           0/0   \n",
       "20550           0/0          0/0          0/0           0/0           0/0   \n",
       "20551           0/0          0/0          0/0           0/0           0/0   \n",
       "20552           0/0          0/0          0/0           0/0           0/0   \n",
       "20553           0/0          0/0          0/0           0/0           0/0   \n",
       "20554           0/0          0/0          0/0           0/0           0/0   \n",
       "20555           0/0          0/0          0/0           0/0           0/0   \n",
       "20556           0/0          0/0          0/0           0/0           0/0   \n",
       "20557           0/0          0/0          0/0           0/0           0/0   \n",
       "20558           0/0          0/0          0/0           0/0           0/0   \n",
       "20780           0/0          0/0          0/0           0/0           0/0   \n",
       "20781           0/0          0/0          0/0           0/0           0/0   \n",
       "20782           0/0          0/0          0/0           0/0           0/0   \n",
       "20783           0/0          0/0          0/0           0/0           0/0   \n",
       "20784           0/0          0/0          0/0           0/0           0/0   \n",
       "20785           0/0          0/0          0/0           0/0           0/0   \n",
       "20786           0/0          0/0          0/0           0/0           0/0   \n",
       "20787           0/0          0/0          0/0           0/0           0/0   \n",
       "20788           0/0          0/0          0/0           0/0           0/0   \n",
       "\n",
       "      Otherinfo230 Otherinfo231 Otherinfo232               Chr_bp  \n",
       "43             0/0          0/0          0/0  9:17579258:17579258  \n",
       "16099          0/0          0/0          0/0  9:17747115:17747115  \n",
       "17521          0/0          0/0          0/0  9:17761460:17761460  \n",
       "17522          0/0          0/0          0/0  9:17761487:17761487  \n",
       "17523          0/0          0/0          0/0  9:17761500:17761500  \n",
       "19899          0/0          0/0          0/0  9:17786430:17786430  \n",
       "19900          0/0          0/0          0/0  9:17786465:17786465  \n",
       "19901          0/0          0/0          0/0  9:17786478:17786478  \n",
       "19902          0/0          0/0          0/0  9:17786519:17786519  \n",
       "20000          0/0          0/0          0/0  9:17787399:17787399  \n",
       "20001          0/0          0/0          0/0  9:17787423:17787423  \n",
       "20002          0/0          0/0          0/0  9:17787501:17787501  \n",
       "20003          0/0          0/0          0/0  9:17787509:17787509  \n",
       "20178          0/0          0/0          0/0  9:17789398:17789398  \n",
       "20179          0/0          0/0          0/0  9:17789421:17789421  \n",
       "20180          0/0          0/0          0/0  9:17789425:17789425  \n",
       "20181          0/0          0/0          0/0  9:17789473:17789473  \n",
       "20182          0/0          0/0          0/0  9:17789476:17789476  \n",
       "20183          0/0          0/0          0/0  9:17789477:17789477  \n",
       "20184          0/0          0/0          ./.  9:17789485:17789485  \n",
       "20185          0/0          0/0          0/0  9:17789499:17789499  \n",
       "20186          0/0          0/0          0/0  9:17789544:17789544  \n",
       "20187          0/0          0/0          0/0  9:17789545:17789545  \n",
       "20348          0/0          0/0          0/0  9:17791272:17791272  \n",
       "20349          0/0          0/0          0/0  9:17791276:17791276  \n",
       "20350          0/0          0/0          0/0  9:17791308:17791308  \n",
       "20351          0/0          0/0          0/0  9:17791311:17791311  \n",
       "20352          0/0          0/0          0/0  9:17791317:17791317  \n",
       "20353          0/0          0/0          0/0  9:17791318:17791318  \n",
       "20548          0/0          0/0          0/0  9:17793385:17793385  \n",
       "20549          0/0          0/0          0/0  9:17793386:17793386  \n",
       "20550          0/0          0/0          0/0  9:17793389:17793389  \n",
       "20551          0/0          0/0          0/0  9:17793394:17793394  \n",
       "20552          0/0          0/0          0/0  9:17793400:17793400  \n",
       "20553          0/0          0/0          0/0  9:17793407:17793407  \n",
       "20554          0/0          0/0          0/0  9:17793437:17793437  \n",
       "20555          0/0          0/0          0/0  9:17793465:17793465  \n",
       "20556          0/0          0/0          0/0  9:17793474:17793474  \n",
       "20557          0/0          0/0          0/0  9:17793477:17793477  \n",
       "20558          0/0          0/0          0/0  9:17793481:17793481  \n",
       "20780          0/0          0/0          0/0  9:17795545:17795545  \n",
       "20781          0/0          0/0          0/0  9:17795552:17795552  \n",
       "20782          0/0          0/0          0/0  9:17795560:17795560  \n",
       "20783          0/0          0/0          0/0  9:17795584:17795584  \n",
       "20784          0/0          0/0          0/0  9:17795626:17795626  \n",
       "20785          0/0          0/0          0/0  9:17795643:17795643  \n",
       "20786          0/0          0/0          0/0  9:17795649:17795649  \n",
       "20787          0/0          0/0          0/0  9:17795682:17795682  \n",
       "20788          0/0          0/0          0/0  9:17795685:17795685  \n",
       "\n",
       "[49 rows x 244 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic variants for SAS\n",
    "coding_SAS = SH3GL2_SAS[SH3GL2_SAS['Func.refGene'] == 'exonic']\n",
    "coding_SAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr             32\n",
       "Start           32\n",
       "End             32\n",
       "Ref             32\n",
       "Alt             32\n",
       "                ..\n",
       "Otherinfo229    32\n",
       "Otherinfo230    32\n",
       "Otherinfo231    32\n",
       "Otherinfo232    32\n",
       "Chr_bp          32\n",
       "Length: 244, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Filter exonic and non-syn vars in SAS\n",
    "coding_nonsynonymous_SAS = SH3GL2_SAS[(SH3GL2_SAS['Func.refGene'] == 'exonic') & (SH3GL2_SAS['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_nonsynonymous_SAS.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo224</th>\n",
       "      <th>Otherinfo225</th>\n",
       "      <th>Otherinfo226</th>\n",
       "      <th>Otherinfo227</th>\n",
       "      <th>Otherinfo228</th>\n",
       "      <th>Otherinfo229</th>\n",
       "      <th>Otherinfo230</th>\n",
       "      <th>Otherinfo231</th>\n",
       "      <th>Otherinfo232</th>\n",
       "      <th>Chr_bp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>9</td>\n",
       "      <td>17579258</td>\n",
       "      <td>17579258</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon1:c.C16T:p.L6F</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17579258:17579258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16099</th>\n",
       "      <td>9</td>\n",
       "      <td>17747115</td>\n",
       "      <td>17747115</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon2:c.A95C:p.D32A</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17747115:17747115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17523</th>\n",
       "      <td>9</td>\n",
       "      <td>17761500</td>\n",
       "      <td>17761500</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon3:c.C178G:p.P60A</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17761500:17761500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19900</th>\n",
       "      <td>9</td>\n",
       "      <td>17786465</td>\n",
       "      <td>17786465</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon4:c.C272T:p.A91V</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17786465:17786465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19901</th>\n",
       "      <td>9</td>\n",
       "      <td>17786478</td>\n",
       "      <td>17786478</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SH3GL2:NM_003026:exon4:c.G285C:p.E95D</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>9:17786478:17786478</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 244 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chr     Start       End Ref Alt Func.refGene Gene.refGene  \\\n",
       "43       9  17579258  17579258   C   T       exonic       SH3GL2   \n",
       "16099    9  17747115  17747115   A   C       exonic       SH3GL2   \n",
       "17523    9  17761500  17761500   C   G       exonic       SH3GL2   \n",
       "19900    9  17786465  17786465   C   T       exonic       SH3GL2   \n",
       "19901    9  17786478  17786478   G   C       exonic       SH3GL2   \n",
       "\n",
       "      GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "43                     .  nonsynonymous SNV   \n",
       "16099                  .  nonsynonymous SNV   \n",
       "17523                  .  nonsynonymous SNV   \n",
       "19900                  .  nonsynonymous SNV   \n",
       "19901                  .  nonsynonymous SNV   \n",
       "\n",
       "                            AAChange.refGene  ... Otherinfo224  Otherinfo225  \\\n",
       "43       SH3GL2:NM_003026:exon1:c.C16T:p.L6F  ...          0/0           0/0   \n",
       "16099   SH3GL2:NM_003026:exon2:c.A95C:p.D32A  ...          0/0           0/0   \n",
       "17523  SH3GL2:NM_003026:exon3:c.C178G:p.P60A  ...          0/0           0/0   \n",
       "19900  SH3GL2:NM_003026:exon4:c.C272T:p.A91V  ...          0/0           0/0   \n",
       "19901  SH3GL2:NM_003026:exon4:c.G285C:p.E95D  ...          0/0           0/0   \n",
       "\n",
       "      Otherinfo226 Otherinfo227  Otherinfo228  Otherinfo229 Otherinfo230  \\\n",
       "43             0/0          0/0           0/0           0/0          0/0   \n",
       "16099          0/0          0/0           0/0           0/0          0/0   \n",
       "17523          0/0          0/0           0/0           0/0          0/0   \n",
       "19900          0/0          0/0           0/0           0/0          0/0   \n",
       "19901          0/0          0/0           0/0           0/0          0/0   \n",
       "\n",
       "      Otherinfo231 Otherinfo232               Chr_bp  \n",
       "43             0/0          0/0  9:17579258:17579258  \n",
       "16099          0/0          0/0  9:17747115:17747115  \n",
       "17523          0/0          0/0  9:17761500:17761500  \n",
       "19900          0/0          0/0  9:17786465:17786465  \n",
       "19901          0/0          0/0  9:17786478:17786478  \n",
       "\n",
       "[5 rows x 244 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_nonsynonymous_SAS.to_csv(f'{WORK_DIR}/coding_nonsynonymous.txt', sep = '\\t', index = False)\n",
    "coding_nonsynonymous_SAS.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculate freq - cases vs controls for SAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "reduced_coding_nonsynonymous_SAS = coding_nonsynonymous_SAS[[\"Chr\", \"Start\", \"End\", \"Gene.refGene\"]]\n",
    "reduced_coding_nonsynonymous_SAS.to_csv(f'{WORK_DIR}/reduced_coding_nonsynonymous_SAS.txt', sep = '\\t', index = False, header= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\t17579258\t17579258\tSH3GL2\n",
      "9\t17747115\t17747115\tSH3GL2\n",
      "9\t17761500\t17761500\tSH3GL2\n",
      "9\t17786465\t17786465\tSH3GL2\n",
      "9\t17786478\t17786478\tSH3GL2\n",
      "9\t17786519\t17786519\tSH3GL2\n",
      "9\t17787509\t17787509\tSH3GL2\n",
      "9\t17789425\t17789425\tSH3GL2\n",
      "9\t17789473\t17789473\tSH3GL2\n",
      "9\t17789476\t17789476\tSH3GL2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head reduced_coding_nonsynonymous_SAS.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_SH3GL2_SAS.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile pheno_SH3GL2_SAS\n",
      "  --ci 0.95\n",
      "  --extract range reduced_coding_nonsynonymous_SAS.txt\n",
      "  --out coding_nonsynonymous_pheno_SH3GL2_SAS\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "20913 variants loaded from .bim file.\n",
      "220 people (122 males, 98 females) loaded from .fam.\n",
      "215 phenotype values loaded from .fam.\n",
      "--extract range: 20881 variants excluded.\n",
      "--extract range: 32 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 220 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999574.\n",
      "32 variants and 220 people pass filters and QC.\n",
      "Among remaining phenotypes, 80 are cases and 135 are controls.  (5 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to coding_nonsynonymous_pheno_SH3GL2_SAS.assoc ...\n",
      "0%\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_SH3GL2_SAS --extract range reduced_coding_nonsynonymous_SAS.txt --assoc --ci 0.95 --out coding_nonsynonymous_pheno_SH3GL2_SAS --allow-no-sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17579258:C:T</td>\n",
       "      <td>17579258</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17747115:A:C</td>\n",
       "      <td>17747115</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17761500:C:G</td>\n",
       "      <td>17761500</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17786465:C:T</td>\n",
       "      <td>17786465</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17786478:G:C</td>\n",
       "      <td>17786478</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17786519:A:G</td>\n",
       "      <td>17786519</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17787509:T:C</td>\n",
       "      <td>17787509</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789425:G:T</td>\n",
       "      <td>17789425</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789473:C:G</td>\n",
       "      <td>17789473</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789476:C:T</td>\n",
       "      <td>17789476</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789477:G:A</td>\n",
       "      <td>17789477</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789485:C:G</td>\n",
       "      <td>17789485</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789499:T:G</td>\n",
       "      <td>17789499</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789544:G:T</td>\n",
       "      <td>17789544</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17789545:A:G</td>\n",
       "      <td>17789545</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17791272:G:C</td>\n",
       "      <td>17791272</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17791276:G:C</td>\n",
       "      <td>17791276</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17791308:G:C</td>\n",
       "      <td>17791308</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17791318:G:A</td>\n",
       "      <td>17791318</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793386:C:G</td>\n",
       "      <td>17793386</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793389:C:G</td>\n",
       "      <td>17793389</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793407:C:T</td>\n",
       "      <td>17793407</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793437:A:G</td>\n",
       "      <td>17793437</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793465:G:T</td>\n",
       "      <td>17793465</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793474:A:T</td>\n",
       "      <td>17793474</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17793477:C:T</td>\n",
       "      <td>17793477</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795552:A:G</td>\n",
       "      <td>17795552</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795560:G:T</td>\n",
       "      <td>17795560</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795643:C:G</td>\n",
       "      <td>17795643</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795649:A:G</td>\n",
       "      <td>17795649</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795682:A:G</td>\n",
       "      <td>17795682</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:17795685:G:A</td>\n",
       "      <td>17795685</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHR                SNP        BP A1  F_A  F_U A2  CHISQ   P  OR  SE  L95  \\\n",
       "0     9  chr9:17579258:C:T  17579258  T    0    0  C    NaN NaN NaN NaN  NaN   \n",
       "1     9  chr9:17747115:A:C  17747115  C    0    0  A    NaN NaN NaN NaN  NaN   \n",
       "2     9  chr9:17761500:C:G  17761500  G    0    0  C    NaN NaN NaN NaN  NaN   \n",
       "3     9  chr9:17786465:C:T  17786465  T    0    0  C    NaN NaN NaN NaN  NaN   \n",
       "4     9  chr9:17786478:G:C  17786478  C    0    0  G    NaN NaN NaN NaN  NaN   \n",
       "5     9  chr9:17786519:A:G  17786519  G    0    0  A    NaN NaN NaN NaN  NaN   \n",
       "6     9  chr9:17787509:T:C  17787509  C    0    0  T    NaN NaN NaN NaN  NaN   \n",
       "7     9  chr9:17789425:G:T  17789425  T    0    0  G    NaN NaN NaN NaN  NaN   \n",
       "8     9  chr9:17789473:C:G  17789473  G    0    0  C    NaN NaN NaN NaN  NaN   \n",
       "9     9  chr9:17789476:C:T  17789476  T    0    0  C    NaN NaN NaN NaN  NaN   \n",
       "10    9  chr9:17789477:G:A  17789477  A    0    0  G    NaN NaN NaN NaN  NaN   \n",
       "11    9  chr9:17789485:C:G  17789485  G    0    0  C    NaN NaN NaN NaN  NaN   \n",
       "12    9  chr9:17789499:T:G  17789499  G    0    0  T    NaN NaN NaN NaN  NaN   \n",
       "13    9  chr9:17789544:G:T  17789544  T    0    0  G    NaN NaN NaN NaN  NaN   \n",
       "14    9  chr9:17789545:A:G  17789545  G    0    0  A    NaN NaN NaN NaN  NaN   \n",
       "15    9  chr9:17791272:G:C  17791272  C    0    0  G    NaN NaN NaN NaN  NaN   \n",
       "16    9  chr9:17791276:G:C  17791276  C    0    0  G    NaN NaN NaN NaN  NaN   \n",
       "17    9  chr9:17791308:G:C  17791308  C    0    0  G    NaN NaN NaN NaN  NaN   \n",
       "18    9  chr9:17791318:G:A  17791318  A    0    0  G    NaN NaN NaN NaN  NaN   \n",
       "19    9  chr9:17793386:C:G  17793386  G    0    0  C    NaN NaN NaN NaN  NaN   \n",
       "20    9  chr9:17793389:C:G  17793389  G    0    0  C    NaN NaN NaN NaN  NaN   \n",
       "21    9  chr9:17793407:C:T  17793407  T    0    0  C    NaN NaN NaN NaN  NaN   \n",
       "22    9  chr9:17793437:A:G  17793437  G    0    0  A    NaN NaN NaN NaN  NaN   \n",
       "23    9  chr9:17793465:G:T  17793465  T    0    0  G    NaN NaN NaN NaN  NaN   \n",
       "24    9  chr9:17793474:A:T  17793474  T    0    0  A    NaN NaN NaN NaN  NaN   \n",
       "25    9  chr9:17793477:C:T  17793477  T    0    0  C    NaN NaN NaN NaN  NaN   \n",
       "26    9  chr9:17795552:A:G  17795552  G    0    0  A    NaN NaN NaN NaN  NaN   \n",
       "27    9  chr9:17795560:G:T  17795560  T    0    0  G    NaN NaN NaN NaN  NaN   \n",
       "28    9  chr9:17795643:C:G  17795643  G    0    0  C    NaN NaN NaN NaN  NaN   \n",
       "29    9  chr9:17795649:A:G  17795649  G    0    0  A    NaN NaN NaN NaN  NaN   \n",
       "30    9  chr9:17795682:A:G  17795682  G    0    0  A    NaN NaN NaN NaN  NaN   \n",
       "31    9  chr9:17795685:G:A  17795685  A    0    0  G    NaN NaN NaN NaN  NaN   \n",
       "\n",
       "    U95  \n",
       "0   NaN  \n",
       "1   NaN  \n",
       "2   NaN  \n",
       "3   NaN  \n",
       "4   NaN  \n",
       "5   NaN  \n",
       "6   NaN  \n",
       "7   NaN  \n",
       "8   NaN  \n",
       "9   NaN  \n",
       "10  NaN  \n",
       "11  NaN  \n",
       "12  NaN  \n",
       "13  NaN  \n",
       "14  NaN  \n",
       "15  NaN  \n",
       "16  NaN  \n",
       "17  NaN  \n",
       "18  NaN  \n",
       "19  NaN  \n",
       "20  NaN  \n",
       "21  NaN  \n",
       "22  NaN  \n",
       "23  NaN  \n",
       "24  NaN  \n",
       "25  NaN  \n",
       "26  NaN  \n",
       "27  NaN  \n",
       "28  NaN  \n",
       "29  NaN  \n",
       "30  NaN  \n",
       "31  NaN  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_SAS_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_SH3GL2_SAS.assoc', delim_whitespace=True)\n",
    "SH3GL2_SAS_freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_SH3GL2_SAS.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_SH3GL2_SAS\n",
      "  --extract range reduced_coding_nonsynonymous_SAS.txt\n",
      "  --out coding_nonsynonymous_pheno_SH3GL2_SAS\n",
      "  --recode A\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "20913 variants loaded from .bim file.\n",
      "220 people (122 males, 98 females) loaded from .fam.\n",
      "215 phenotype values loaded from .fam.\n",
      "--extract range: 20881 variants excluded.\n",
      "--extract range: 32 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 220 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999574.\n",
      "32 variants and 220 people pass filters and QC.\n",
      "Among remaining phenotypes, 80 are cases and 135 are controls.  (5 phenotypes\n",
      "are missing.)\n",
      "--recode A to coding_nonsynonymous_pheno_SH3GL2_SAS.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/SH3GL2_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_SH3GL2_SAS --extract range reduced_coding_nonsynonymous_SAS.txt --recode A --out coding_nonsynonymous_pheno_SH3GL2_SAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:17579258:C:T_T</th>\n",
       "      <th>chr9:17747115:A:C_C</th>\n",
       "      <th>chr9:17761500:C:G_G</th>\n",
       "      <th>chr9:17786465:C:T_T</th>\n",
       "      <th>...</th>\n",
       "      <th>chr9:17793437:A:G_G</th>\n",
       "      <th>chr9:17793465:G:T_T</th>\n",
       "      <th>chr9:17793474:A:T_T</th>\n",
       "      <th>chr9:17793477:C:T_T</th>\n",
       "      <th>chr9:17795552:A:G_G</th>\n",
       "      <th>chr9:17795560:G:T_T</th>\n",
       "      <th>chr9:17795643:C:G_G</th>\n",
       "      <th>chr9:17795649:A:G_G</th>\n",
       "      <th>chr9:17795682:A:G_G</th>\n",
       "      <th>chr9:17795685:G:A_A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000062_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000119_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000299_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000331_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000378_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   FID             IID  PAT  MAT  SEX  PHENOTYPE  chr9:17579258:C:T_T  \\\n",
       "0    0  APGS_000062_s1    0    0    1          2                    0   \n",
       "1    0  APGS_000119_s1    0    0    1          2                    0   \n",
       "2    0  APGS_000299_s1    0    0    1          2                    0   \n",
       "3    0  APGS_000331_s1    0    0    1          2                    0   \n",
       "4    0  APGS_000378_s1    0    0    2          2                    0   \n",
       "\n",
       "   chr9:17747115:A:C_C  chr9:17761500:C:G_G  chr9:17786465:C:T_T  ...  \\\n",
       "0                    0                    0                    0  ...   \n",
       "1                    0                    0                    0  ...   \n",
       "2                    0                    0                    0  ...   \n",
       "3                    0                    0                    0  ...   \n",
       "4                    0                    0                    0  ...   \n",
       "\n",
       "   chr9:17793437:A:G_G  chr9:17793465:G:T_T  chr9:17793474:A:T_T  \\\n",
       "0                    0                    0                    0   \n",
       "1                    0                    0                    0   \n",
       "2                    0                    0                    0   \n",
       "3                    0                    0                    0   \n",
       "4                    0                    0                    0   \n",
       "\n",
       "   chr9:17793477:C:T_T  chr9:17795552:A:G_G  chr9:17795560:G:T_T  \\\n",
       "0                    0                    0                    0   \n",
       "1                    0                    0                    0   \n",
       "2                    0                    0                    0   \n",
       "3                    0                    0                    0   \n",
       "4                    0                    0                    0   \n",
       "\n",
       "   chr9:17795643:C:G_G  chr9:17795649:A:G_G  chr9:17795682:A:G_G  \\\n",
       "0                    0                    0                    0   \n",
       "1                    0                    0                    0   \n",
       "2                    0                    0                    0   \n",
       "3                    0                    0                    0   \n",
       "4                    0                    0                    0   \n",
       "\n",
       "   chr9:17795685:G:A_A  \n",
       "0                    0  \n",
       "1                    0  \n",
       "2                    0  \n",
       "3                    0  \n",
       "4                    0  \n",
       "\n",
       "[5 rows x 38 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SH3GL2_SAS_recode = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_SH3GL2_SAS.raw', delim_whitespace=True)\n",
    "SH3GL2_SAS_recode.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save out results..!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -mu terra-18d8e41c cp -r /home/jupyter/SH3GL2_GP2/ gs://fc-f7a400c1-827e-48f8-b7b6-90c488a000a4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_EAS_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AJ_release5.pgen [Content-Type=application/octet-stream]...\n",
      "==> NOTE: You are uploading one or more large file(s), which would run          \n",
      "significantly faster if you enable parallel composite uploads. This\n",
      "feature can be enabled by editing the\n",
      "\"parallel_composite_upload_threshold\" value in your .boto\n",
      "configuration file. However, note that if you do this large files will\n",
      "be uploaded as `composite objects\n",
      "<https://cloud.google.com/storage/docs/composite-objects>`_,which\n",
      "means that any user who downloads such objects will need to have a\n",
      "compiled crcmod installed (see \"gsutil help crcmod\"). This is because\n",
      "without a compiled crcmod, computing checksums on composite objects is\n",
      "so slow that gsutil disables downloads of composite objects.\n",
      "\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EAS.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_SAS.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_EUR.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release4.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_AAC.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_EUR.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_SAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_AMR_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_AAC.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AJ_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_MDE_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_FIN_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EAS.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_EUR.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AMR_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EAS.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AAC_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_SAS.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_SAS.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_CAS.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_AAC_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_EUR_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/master_key_release4_final.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_MDE_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AJ_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_SAS.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EUR_release4.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_SAS_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EAS.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AMR_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_EUR_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_SAS.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EAS_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EUR_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/SEX.txt [Content-Type=text/plain]...    \n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_AMR.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EAS_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/master_key_release5_final.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_EAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/PHENO.txt [Content-Type=text/plain]...  \n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_EAS.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_AJ.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EUR_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_MDE_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AMR_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_FIN.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EAS.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EUR_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_FIN.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EUR_release4.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EAS.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_CAS_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_FIN_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_EUR_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_MDE.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EUR_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_SAS.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_MDE_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release4.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_SAS.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_SAS_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release4.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_CAS_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EAS.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EAS.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_MDE.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AMR_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_CAS_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_CAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_SAS_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_EUR_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_SAS.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_AMR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AAC_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AAC_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_CAS_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EAS_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AJ_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_SAS.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_SAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EUR_release4.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release4.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EUR_release4.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_FIN_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_FIN_release5.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_SAS_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AAC.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_FIN_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AJ.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_AJ.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_pheno_SH3GL2_EUR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_SAS.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_MDE.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AFR_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_MDE_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_AAC_release5.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_CAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EAS.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_CAS_release5.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/chr9_EAS_release5.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_EUR.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AMR.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/reduced_coding_nonsynonymous_EUR.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/coding_nonsynonymous_AJ_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_FIN.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/SH3GL2_GP2/pheno_SH3GL2_AFR.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "| [183/183 files][ 18.3 GiB/ 18.3 GiB] 100% Done  44.2 MiB/s ETA 00:00:00       \n",
      "Operation completed over 183 objects/18.3 GiB.                                   \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR} {WORKSPACE_BUCKET}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {
    "height": "248.972px",
    "width": "263.062px"
   },
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "268.372px"
   },
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
